---
title: "Evaluating Archaic Admixture Time Estimates"
author: Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary Anthropology,
  MPI EVA), Dr. Benjamin Marco Peter (MPI EVA, benjamin_peter@eva.mpg.de)
date: "`r Sys.Date()`"
output:
  pdf_document:
    keep_tex: yes
    latex_engine: xelatex
    fig_caption: yes
#  html_document:
#    code_folding: hide
#    toc: yes
#    toc_depth: 4
#    toc_float:
#      collapsed: no
#  md_document:
#    toc: yes
#  word_document:
#    toc: yes
#    toc_depth: '4'
#  github_document:
#    toc: yes
header-includes:
- \usepackage{setspace}
- \doublespacing
- \usepackage[none]{hyphenat}
- \usepackage{amsfonts}
- \usepackage{amssymb}
- \usepackage{graphicx}
- \usepackage{float}
- \floatplacement{figure}{H}

csl: References/plos-biology.csl
bibliography: References/My_Library_ATE.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r echo=FALSE}
#Make code wrap text so it doesn't go off the page when Knitting to PDF
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
```

<style>
body {
text-align: justify}
</style>

#Abstract


#Introduction

Detecting admixture, i.e. gene flow between previously isolated populations can shed light into the complex evolutionary history of populations. It is an important factor in speciation  events and is found in many sexually reproducing species of plants, invertebrates and vertebrates [@abbott_hybridization_2013;@payseur_genomic_2016;@harrison_hybridization_2014]. The sequencing of the Neandertal [@green_draft_2010] and Denisovan [@reich_genetic_2010] genome revealed admixture events between Neandertals and  modern humans outside of Africa  [@green_draft_2010; @prufer_complete_2013; @fu_early_2015; @fu_genome_2014; @sankararaman_genomic_2014;@prufer_high-coverage_2017], as well as a second admixture event between Denisovans and Asian populations [@reich_genetic_2010; @meyer_high-coverage_2012; @sankararaman_combined_2016; @vernot_excavating_2016]. 


*bp 19/02:this feels very thin. It's OK to motivate admixture as something that is widely present, but then you need to give a comprehensive introduction, not just a very incomplete list of papers. The alternative is to simply refer to a bunch of relevant reviews*

*li 27/02:I refere now to some hybridization reviews which point out to alot of admixture papers in different species*

##Admixture on the genomic level and basic idea how to date it
On the genomic level, admixture introduces highly divergent chromosomal segments into the admixing population. Over time, miotic recombination progressively breaks these introgressed segments down [@falush_inference_2003, @baird_etheridge_barton_2003]. 
Assuming that recombination events occur according to a Poisson process, the fragment length distribution can be approximated by an exponential distribution, whose parameter is informative about the number of generations since the fragments entered the admixing population [@moorjani_history_2011;@pool_inference_2009; @gravel_population_2012].
In particular, under a constant recombination rate, the length of introgressed segments should be inversely proportional to the time since the admixture event [@gravel2012, @liang_nielsen2014] .  Hence, the length distribution of introgressed chromosomal segments ('recombination clock') can be used to inferred the time since the admixture event [@moorjani_history_2011;@pugach_dating_2011; @sankararaman_date_2012;@loh_inferring_2013;@sankararaman_combined_2016;@pugach_gateway_2018;@jacobs_multiple_2019,@hellenthal_genetic_2014]  
*bp 19/02:add hellenthal 2014 science*

*bp 19/02: Maybe introduce here the term of admixture pulse*

To do this, the model usually assumes an admixture pulse, where admixture between two populations is a one generation long event followed by a period of isolation [@liang_lengths_2014]. The proportion of admixture of individuals in the admixed population is assumed to be identical, tracts are rare and inbreeding is not significant, such that admixture segments do not recombine with each other to form longer segments [@pool_inference_2009].  The effect of drift is assumed to be negligible [@loh_inferring_2013], the introgressed segments act neutral [@shchur_distribution_2019] and the recombination rate map is identical across populations and does not change over time [@gravel_population_2012].
<!-- Introduce model assumptions here -->
<!-- constant recombination rate -->
<!-- same ancestor -->


##The two approaches of obtaining the length of introgressed segments: ALD
The challenge is to correctly infer the length of the introgressed segments. There are two main approaches to date admixture [@chimusa_dating_nodate] (Figure \ref{fig:Intro_fig} A). The first approach uses the admixture-induced linkage disequilibrium (ALD) decay. 
Gene flow introduces highly divergent archaic segments into the human background. Hence, variants on these archaic segments are expected to be in high linkage disequilibrium to each other at the time of admixture [@chakraborty_admixture_1988; @stephens_mapping_1994; @wall_detecting_2000]. 
As recombination breaks the introgressed chromosomal segments apart each generation, the association between variants on these introgressed segments decays as genetic distance increases. In case of a recent admixture event, ALD is found over long genetic distances [@patterson_methods_2004] and is therefore easy distinguishable from short range non-ALD [@moorjani_history_2011]. For ancient admixture however, a pairwise LD between introgressed variants is calculated using an ascertainment scheme to filter for the introgressed variants in the admixed population, since LD form admixture and LD from other processes is in a similar length range [@sankararaman_date_2012]. The pairwise LD as a function of genetic distant is then fitted using an exponential distribution holding a point estimate for the time since the admixture event [@moorjani_history_2011; @loh_inferring_2013]. 


##The two approaches of obtaining the length of introgressed segments: searching for introgressd haplotypes
The other approach first tries to infer introgressed haplotypes directly, by partitioning an individuals genome into admixture segments [@gravel_population_2012; @price_sensitive_2009; @lawson_inference_2012]. Multiple methods are available focusing on different characteristics of introgressed haplotypes, such as LD patterns and density of derived variants, to infer them [@lawson_inference_2012; @racimo_signatures_2017; @seguin_orlando_paleogenomics_2014; @vernot_excavating_2016; @sankararaman_combined_2016; @skov_detecting_2018]. In the second step, the length distribution of these directly inferred introgressed haplotypes can be used for dating by fitting an exponential distribution to it.

##Dates of archaic introgression obtained so far

*bp 19/02: This paragraph is in quite good shape, but I think it is worth highlighting how the approaches used differ, and why accuracy was increased over time*

*li 27/02: Included how they differ but it goes into quit detail already especially the differences in Pryas method compared to Siriams where she uses a slightly differnt ascerteinment scheme (the one I am also testing) pluse the DECODE map for the corection of the genetic map*

Both approaches have been used to estimate the time of admixture between Neandertals and modern humans.  Using the decay of introgressed Neandetlas variants in modern day genomes, Sankararaman et al dated the neandertal-human admixture time to be 37,000–86,000 ya (years ago) [@sankararaman_date_2012]. Later, this data was refined to 40,510–54,454 ya (95% CI) using a differnt ascerteinment scheme combined with a differnt genetic map [@moorjani_genetic_2016]. A date was also obtained from an ancient genome to be 50,000 to 60,000 ya by adding the admixture date of the ancient individual, obteined by the decay of pairwise covariance between introgressed SNPs, to the speciments radiocarbon date [@fu_genome_2014]. 

An additional admixture event between Neandertals and East-Asian populations at the same time as the admixture between Neandertals and all non-Africans was suggested to explain the higher amount of Neandertal ancestry in present-day East Asians [@kim_selection_2015; @vernot_complex_2015]. The Denisovan human admixture was dated to be between 44,000–54,000 ya using the ALD approache on modern day genomes [@sankararaman_combined_2016]. An additional Denisovan admixture event was suggested for East-Asian populations by directly identifying Denisovan haplotype segments, which are more diverged from the Denisovan high-coverage genome then previously found segments [@browning_analysis_2018]. Comparing the mean length of the two groups of segments introgressed from divergent Denisovan populations did, however, not reveal significant differences, suggesting a lack of power to distinguish the two events by time [@browning_analysis_2018; @jacobs_multiple_2019]. Analysing genomes from Papuan individuals revealed two time separated admixture events with Denisovan, one in line with previous estimates at 45.7 kya (95% CI 31.9-60.7 kya) and one exclusive to Papuans dated to be around 29.8 kya (95% CI 14.4-50.4 kya) [@jacobs_multiple_2019].

<!-- [I would move the intro paragraph ] -->

<!-- What does admixture pulse mean? is it distingushable from a continuous admixtuer? -->
<!-- the time of the last GF is relevant for Neandertal extinction, hence the extrema of the distribition more important then mean time -->


*li 27/02: I think it makes sense to mention all the assumptions which might be problematic and in the end how we gonne do it (check the effec of continuous admixture, compare it to the other assumptions, try to explicitly model continuous admixture using the Lomax)*

##Assumptions on the data
Both approaches, the direct measure of the segment length and the indirect ALD based method, without first directly inferring introgressed segments such as the popular ALDER method [@loh_inferring_2013], rely on some fairly strong modelling assumptions [@pool_inference_2009; @gravel_population_2012; @liang_lengths_2014]. We want to investigate which of these assumptions can be relaxed without causing a strong bias in the estimate.
<!-- last sentence changed to be more positive by Ben's recommendationc -->

### Segments are is independent and identically exponentially distributed

First, it is assumed that the lengths of the introgressed segments are independent 
and identically exponentially distributed. One restriction this puts on the data is that segments of the same ancestor are 
assumed not to recombine into longer segments. This assumption is violated in particular 
when admixture is very old $(t \approx 2N)$ or very recent $t \approx \log(2N)$ [@liang_lengths_2014].

*li 27/02: from liang_lengths_2014: Tract length can be correlated due to the variance in migrant ancestors an individual has and due to inbreeding joining together migrant tracts which violates the iid assumption especially in the cases mentioned*

*bp 19/02: in this paragraph, you repeated yourself a lot. Try to be concise and present your argument in one logical sequence whenever possible* 

### Recombination Map

Second, it is assumed that the recombination rates are known. Since the recombination 
is used as a clock, misspecification of the recombinationn map will lead to over/or 
underestimates of tract lengths, and admixture times. Crucially, even if the
length estimation is unbiased, the admixture time estimates can be substantially
biased. Uncertainties in the genetic map were found to downward bias the time estimates by missing certain recombination events resulting in assuming introgressed segments longer than they actually are [@sankararaman_date_2012].  Sankararaman et al. 2012 suggested a method for accounting for uncertainty in the recombination map by estimating a correction parameter for a given recombination map by comparing the distances between a pair of markers in the map to the number of crossovers that span those markers as observed in a pedigree map. This however limits the applicability of the method, since it requires a population-specific map as well as a pedigree based map to estimate the error in the map [@sankararaman_date_2012; @fu_genome_2014; @sankararaman_combined_2016]. 

*bp 19/02:recap what Priya/Sriram found about the bias/correction*

*li 27/02: now I first explain  how it affects the estimate according to Siriam (Prya quotes him on this) and then what is the problem with the correction and then what is the problem with genetic maps are*

There are different recombination maps used to assign the genetic length between introgressed segments/variants available[@coop_high-resolution_2008; @kong_fine-scale_2010;@hinch_landscape_2011; @wegmann_recombination_2011;@mcvean_fine-scale_2004; @myers_fine-scale_2005; @hapmap_consortium_2007]. 
They either focuse on short term information by using parent-offspring data to directly observe recombination events [@coop_high-resolution_2008; @kong_fine-scale_2010], searching for admixture tracts breakpoints [@hinch_landscape_2011; @wegmann_recombination_2011] or long term population averages of recombination inferred by LD [@mcvean_fine-scale_2004; @myers_fine-scale_2005; @hapmap_consortium_2007]. 
Different genetic maps are highly correlated on a megabase scale but reveal significant population differences on a finer scale [@kong_fine-scale_2010; @hinch_landscape_2011]. The average length of Neandertal introgressed segments is approximately 50 kb [@sankararaman_date_2012], and so fine-scale recombination maps are essential. Furthermore, recombination maps can also vary over time [@auton_fine-scale_2012]. Highly precise maps are not available for every population and the change through time is unknown.

### Demography
*bp 19/02:this section (and frequently your writing in general) could need more structure. One simple approach is the _context-content-conclusion_ system For this intro section, the basic idea is that for each paragraph you need to make clear that you start by*

*1. explaining why we're interested in demography,* 
*2. what precisely the issue is*
*3. explain what you're gonna do about it*


A third assumption is that the demographic history of the studied population is
known. While the recombination clock is relatively robust to population size 
changes *bp 19/02: ref Gravel2012? li 27/02: No they didn't check for fluctuations in Ne*, demographic processes can create similar genomic signals as 
admixture *how*. Genomic segments carrying these signals can falsely be included 
in the estimate of the length distribution of admixed segments, leading to a bias by considering them while fitting them to a distribution.  *how?*
Admixture segments can also get lost by genetic drift, making it harder to 
estimate the length distribution due to its potential deviation from an exponential distribution.*why?* 
Figure \ref{fig:Intro_fig} B shows two possible demographies for the Neandertal 
admixture, a simplified one without no population changes and a more complex one 
including structured populations, bottlenecks and additional gene flow. 
The extreme reduction in population sizes of the admixing population can create 
non-admixture LD [@schaper_linkage_2012]. Substructure in an ancestral human 
population, where human population in Africa are subdivided with a low migration 
rate, as shown in Figure \ref{fig:Intro_fig} B, can also introduce divergent 
chromosomal segment with variants in strong LD, mimicking Archaic introgressed 
segments [@nei_linkage_1973;@pfaff_population_2001]. Moreover additional gene flow after the Neandertal admixture event between the African and admixed non-African population was shown to potentially influence the detected amount of Neandertal ancestry [@petr_limits_2019] as well as the time estimates [@sankararaman_date_2012]. 
<!-- Not sure about the last sentence -->
<!-- Drift biasing admixture citation needed -->

### Methodological bias through ascertainment 
An additional bias might be introduced in the ALD based dating method when
ascerteinment of SNPs for admixture informative sites is deploid, wherby positions in the admixed population are considered where both source populations are diverged from eachother. This is usually done for archaic 
admixture to enrich the signal of ALD to not confalte it with background LD from demographic processes which is in the same range length [@sankararaman_date_2012; @fu_genome_2014]. 

*bp 19/02: explain briefly why this is necessary, and how exactly this is done*

The problem here is that  introgressed variants from the ancestral archaic 
population are ascertained using only a few sequenced archaic individuals as 
proxies for the actual introgressing archaic population. These individuals are 
potentially only distantly related to the introgressing population, which might 
result in missing introgressed variants. 

*li 27/02 What to cite for the last sentence?*

### The Pulse Assumption
Finally, the admixture is assumed to happened in only a single generation [@moorjani_history_2011]. This single pulse is mathematically convenient since all the introgressed segments entering 
in this single generation can be modeled using a single exponential distribution. 
This however limits the information gained form the dating since in case of a 
multi-generation long admixture the single-pulse model results in a mean admixture 
time between start and end of the admixture. Ideally one would like to obtain 
the start and end, hence the duration of admixture between populations. 

*bp 19/02: preceeding section could be merged with last intro paragraph*

*bp 19/02: this should be at the end of the intro*

*li 27/02: Now it is at the end and i merged it with the existing paragraph as well as the previous attempts of modeling multi-generation long admixture*

##What we want to do


While good estimates for the mean admixture time between Neandertals and modern humans
exist, little is known about the duration of the event. In particular, dates
when admixture ended are of crucial importance, as they are relevant to 
Neandertals extinction. Models were introduced to model multi-generation long admixture. If admixture is happening over $n$ generations, this results in a mixture of $n$ 
different exponential distribution, one for the segments entered in each generation 
[@pickrell_ancient_2014].  Previous studies tried to model this, using the ALD, 
by assuming not a continuous admixture but summarizing it into different 
independent events. In this case, the decay curve is comprised out of $n$ 
different exponential terms, each for one admixture event 
[@pickrell_ancient_2014; @zhou_inference_2017]. To model it as a continuous admixture Zhou et al. 2017
suggested using a polynomial function with the number of coefficients related to the number of generations of continuous admixture [@zhou_modeling_2017]. These models however work best with recent admixture, in the last 100 generations, with a substantial admixture proportion of around 0.3 [@zhou_modeling_2017]. 
<!-- zhou_modeling_2017-tested on simulations with no changes in Ne, Admixture proportion of 30% and uniform recombnation [@zhou_modeling_2017]-->
<!-- Fitting this polynomial functions is not trivial -->
This is outside of the range for scenarios of Neandertal admixture, were the admixture proportion is around 0.03 and the timing around 1500 generations ago [@sankararaman_date_2012;@prufer_complete_2013;@sankararaman_genomic_2014]. here we introduce a model for continuous admixture, where the migration rate over time is Gamma distributed with only two parameters, one for the mean time of continuous admixture and one for the duration of the admixture. In our simulation study, we first examining the effect of long continuous admixture on the admixture time estimates in comparison to the effects of the aforementioned model assumptions. Therefore, we quantitatively evaluate the effect of the admixture model, recombination, demography and analysis parameters on the inference. Second, we define the expectation of the resulting segment length distribution for continuous Gamma distributed admixture being Lomax distributed, holding a parameter for the duration of admixture. This expectation works for both methods to infer the segments length, either directly or by using the ALD decay. Using this model, we investigate under which scenarios the parameters of the Lomax-distribution can be  accurately  estimated and for which parameters we can distinguish a pulse-like admixture event from a continuous event. We show that in many cases
pulses cannot be distinguished from more continuous admixture events, 
and so current methods are unsuitable to definitively answer when the contact between 
Neandertals and modern humans ended. 


```{r Intro_fig,message=FALSE, echo=FALSE,warning=FALSE,fig.align="center",out.width="95%",fig.pos="H",fig.cap="\\label{fig:Into_fig} A) Chromosomal sections with introgressed segments (grey). Introgressed variants (green circles) are in high LD compared to the background (stars). The ALD approach estimates linkage between the introgressed variants, wheres the haplotype  approach tries to estimate the segment directly. B) Two possible demographies of humans splitting into African population (AFR) and non-African (non-AFR) with admixture from Neandertals (NEA) into non-Africans. The simple demographic model with constant population sizes and the complex with rapid population size changes (1), substructure in Africa, where after an initial earlier split and isolation the structured population exchange migrants till the out-of-Africa event (2) and additional gene flow between Africans and non-Africans after the Neandertal admixture (3)."}
library(png)
#Intro_Fig <- readPNG('Paper_Graphics.png',native = T,info = T)
include_graphics(path = 'Paper_Graphics_2.png',auto_pdf = T)

```


#Methods

We conducted various simulations to assess the effect of continuous admixture compared to a pulse under ideal circumstances. We changed recombination and demographic parameters to simulate more realistic models. We compared the effect of these parameters together with analysis parameters to the effect of continuous admixture on the estimates. After assessing these effects we evaluate the possible parameter range for using a Lomax distribution to fit the ALD decay, enabling to obtain a duration of continuous admixture.
<!-- Only draft paragraph!!!! -->

##Simulations

We used the msprime coalescent simulator [@kelleher_efficient_2016] for simulations with sample sizes chosen to reflect presently available data: We simulate 176  diploid African individuals and  170 diploid non-Africans, corresponding to the number of haploid Yoruba (YRI) and Central Europeans from Utah (CEU) sequences in the 1000 Genomes project data [@the_1000_genomes_project_consortium_global_2015]. Since three high coverage Neandertal sequences are available [@prufer_complete_2013;@prufer_high-coverage_2017] we choose to simulate three diploid genomes. For each individual we simulated 20 chromosomes with a length of 150 Mb each. The mutation rate was set for all simulations to $2*10^{-8}$ per base per generation. The recombination rate was set to $1*10^{-8}$  per base pair per generation unless specified otherwise. The demographic parameters are based on previous studies dating Neandertal admixture [@sankararaman_date_2012;@fu_genome_2014;@moorjani_genetic_2016]. In the “simple” model (Figure \ref{fig:Intro_fig} B), the  effective population size is assumed constant at Ne=10000 for all populations, the split time between modern humans and Neandertals is 10000 generations ago and a split time between Africans and non-Africans is 2550 generations ago. The migration rate from Neandertals into non-Africans was set to zero before the split from Africans, to ensure no Neandertal ancestry in Africans. Each simulation was repeated 100 times.


###Gene Flow

<!-- Maybe one should harmonize the equations with Ben's -->
In the admixture pulse model, gene flow is a one generation long resulting in an exponentially distributed admixture-induced linkage disequilibrium (ALD) decay curve (Eq. \ref{eq:1}), with $\lambda$ as the rate parameter of the exponential distribution holding the inverse of time since the admixture event $t$ as a function of genetic distance $d$, 

\begin{equation}
\begin{split}
\label{eq:1}
x_i \sim exp(\lambda) \\
\mathbb{E}[x] = \frac{1}{\lambda}
\end{split}
\end{equation}

Continuous gene flow over time was modeled as a gamma distribution  (Eq. \ref{eq:2})

\begin{equation}
\label{eq:2}
m_i \sim \Gamma(k+1,\frac{1}{\theta})
\end{equation}

Where k is the shape and $\theta$ the rate parameter. The parameter values are chosen such that the mean length $\frac{1}{\lambda}$ of the exponentially distributed ALD decay curve resulting from the one generation admixture pulse, is equal to the mean length of a ALD decay curve, as a result of continuous migration with the same total amount of migrants, modeled using a Lomax distribution (Eq. \ref{eq:3})

\begin{equation}
\begin{split}
\label{eq:3}
x_i \sim Lomax(k,\theta) \\
\mathbb{E}[x] = \frac{1}{\lambda} = \frac{\theta}{k}
\end{split}
\end{equation}

\begin{equation}
\label{eq:4}
\lambda=\frac{k}{\theta}
\end{equation}

\begin{equation*}
\nonumber
where \qquad k=\mathbb{E}[x] \, \theta \qquad and \qquad \theta=\frac{\mathbb{E}[x]}{Var[x]}
\end{equation*}

Equation (Eq. \ref{eq:4}) shows the relationships between the distribution parameters such that the resulting decay mean length are equal. Here $x$ is in generations.

<!-- Here E[x] is mean admixture time in generations and Var[x] is one fourth of the length of admixture time in generations squared, with x as the number of generations of admixture between two populations. -->

###Recombination map

Uncertainties in the recombination map were previously shown to bias admixture time estimates.
To investigate the effect of more realistic recombination rate variation we simulated using a recombination map. We either used the African-American-Map [@hinch_landscape_2011] or the HapMap phase 3 [@noauthor_second_2007] for simulations under a variable recombination rate, for simplicity, we used the same recombination map (150 Mb of chromosome 1, excluding the first 10 Mb) for all simulated chromosomes.  The mean recombination rate was calculated from the 150 Mb map ($1.843 \, \frac{cM}{Mb}$ AAMap and $1.549 \, \frac{cM}{Mb}$ HapMap). To emulate uncertainties in the genetic map we either used the mean recombination rate from the respective map to calculate the genetic position from the physical position for each SNP, used linear interpolation based on the other map (e.g. AAMap used for the msprime simulation and HapMap used to assign genetic distances) or used the same map for simulation and assigning genetic distances. 


###Inferred demography

Demography such as population size changes are known to influence LD patterns and hence create false admixture signals.
To test the impact of demographic history on admixture time estimates, we simulate a more realistic and complex demographic history based on effective population sizes and split times estimated from Neandertal and present-day human genomes. MSMC estimates from YRI as representatives for Africans and CEU for non-Africans from Schiffles & Durbin 2014 [@schiffels_inferring_2014] were used together with PSMC [@li_inference_2011] inferred demographic model for Neandertals based on the Vindija33.19 high-coverage genome [@prufer_high-coverage_2017].
In order to use the effective population size estimates at a given time point for modern humans from Schiffels & Durbin 2014 (Figure 4 Excel Table) for our simulations, we first transformed the time points given in years back to generations by using 30 years for one generation, as assumed in the original study. Second, since the original estimates are based on a different mutation rates ($1.25*10^{-8} \frac{bp}{gen}$), we corrected all estimates for the mutation rate used in the simulations ($2*10^{-8} \frac{bp}{gen}$). The split times between Neandertals and modern humans as well as between Africans and non-Africans were kept the same as in the simple simulations (1000  and 2550 generations ago, respectively). The population size of the ancestor of Neandertals and humans before the split was set to 1000. To simulate branch shortening caused by the extinction of Neandertals, Neandertals were sampled 750 generations before the Africans and non-Africans.

##Admixture time estimates

###Ascertainment scheme

Ascertainment schemes are used to select certain variable positions of interest in a genome. Ascertainment schemes can be used to enrich for Neandertal informative sites in the test population to remove noise and amplify the ALD signal [@sankararaman_date_2012]. Two ascertainment schemes were tested to enrich for Neandertal informative sites, which were used previously [@sankararaman_date_2012,@fu_genome_2014]. The lower-enrichment ascertainment scheme filters for SNPs fixed for the ancestral state in Africans and polymorphic in Neandertals. The higher-enrichment ascertainment scheme restricts the analysis on SNPs fixed for the ancestral state in Africans, polymorphic in Neandertals and polymorphic in non-Africans.

###ALD calculation and curve fitting

The pairwise weighted LD between the ascertained SNPs a certain genetic distance $d$ apart was calculated using ALDER [@loh_inferring_2013]. A minimal genetic distance $d_0$ between SNPs is set either to 0.02 cM and 0.05 cM. This minimal distance cutoff removes extreme short range LD likely confounded by non-ALD, facilitating the fitting procedure. To obtain the mean time estimates the data is fitted with an exponential distribution shown in equation \ref{eq:5}, using a non-linear least-square optimization algorithm implemented in R [@R_Core_Team_2019]. Where *A* is the intercept, $t$ the time since the admixture event in generations, *d* the genetic distance in cM and *c* is a constant modeling background LD. The model was fitted following Moorjani et al 2016 [@moorjani_genetic_2016]. The duration of continuous admixture is modeled using the Lomax fit shown in Eq. \ref{eq:6}. We used the notation from Kozubowski et al. 2008 [@Kozubowski_2008_TestingEV]. The starting value of $t$ is taken from the exponential fit to ensure convergence of the model. Model validity diagnostics like distribution of residuals, residuals plotted against fitted values and model stability diagnostics like dfbetas were checked for selected simulations and showed no obvious deviations from the model assumptions [@quinn_experimental_2002]. To compare the two nested models we used Akaike's information criterion (AIC) measuring the goodness of fit while penalizing the addition of new parameters $p$ and thus controlling for under- and overfitting (Eq. \ref{eq:7}).

\begin{equation}
\label{eq:5}
ALD \sim\ A\,e^{-td}+c
\end{equation}

\begin{equation}
\label{eq:6}
ALD \sim\ A\,\left( \frac{1}{1 + \frac{1}{k}td}\right) ^k+c
\end{equation}

\begin{equation}
\label{eq:7}
AIC = 2p - 2\ln(\hat{L})
\end{equation}

###Modeling parameter effect sizes

To model and compare parameter effect sizes we  simulated 100 replications  for each  combination of the previously introduced parameters: ascertainment scheme (LES/HES), minimal genetic distance ($d_{0}=0.02\,cM$/$d_{0}=0.05\,cM$), demography (simple/inferred), recombination rate (constant/variable), gene flow model (pulse/continuous). Results of simulations where  the nls-optimization to fit the ALD decay curve did  not converge were removed (11 out of 3200). We used a Gaussian linear  least-squares model to estimate the effect size of the different parameters. The deviation between the estimated admixture time and the true admixture time, the error in the estimate, was used as the response to the non interacting parameters as model predictors assuming normal distributed and homogeneous residuals. Calculation of variance inflation factors of the predictors where not indicative for colinearity between predictors. Model stability assessment using dfbetas showed stable model predictors and residuals and plotting residuals against fitted values revealed slight deviations  from a normal distribution mostly driven by few extreme values of simulations under the variable recombination rate, however, no overall obvious deviation from the assumptions.



#Results

##Theoretical framework for continuous admixture

First we want to establish a model of continuous admixture and an expectation of the tract length distribution under this model in an ideal-case from perfect data.
<!-- The goal here is to estimate the ideal-case limits of which admixture scenarios can be distinguished from perfect data.  -->
For this purpose, we assume that the lengths of introgressed tracts are perfectly known. In this case, under some models, the distribution of introgressed tract lengths $L_i$ can be written as

\begin{equation}
x_i \sim exp({\lambda})
\end{equation}

where $t$ is the time when the fragment entered the population, and $\lambda$ is a parameter that depends on the model assumptions and recombination rate $r$ [@liang_lengths_2014]. E.g under the SMC, $\lambda = (1-m)r$, and under the SMC' allowing for back coalescence, $\lambda = 2N(1-m)(1-exp^{-t/2N})r$. For Neandertal admixture where $m$, the admixture fraction, is typically low, the exponential assumption is satisfied [@liang_lengths_2014]. For scenarios where the length of admixture tracts is not exponential, e.g. because admixture is recent or very old, our results do not apply. 

<!-- SMC' lambda under the assumption that 2N fixed and T -> inf otherwise it is 2N(1-m)(1-e^(-t/2N)) -->
<!-- m is the admixture proportion and r is the genetic distance -->
	
It is widely assumed that Neandertal ancestry entered the modern human population over a very short period. 
As an alternative model, we need to consider $t$ not as a single point in time, but as a random variable itself that follows a mixture distribution $\mathcal{D}_t$. The most widely studied is a small number of discrete ``pulses'' of admixture, in which case $\mathcal{D}_t$ is categorical. Here, we instead assume a continuous $\mathcal{D}_t$; more precisely we assume $\mathcal{D}_t$ follows a $\Gamma(k+1, \lambda t)$-distribution. This has a number of advantages:
\begin{itemize}
	\item We just need one additional parameter $k$, that can be interpreted as the duration of gene flow, instead of the minimal of two additional parameters required for the pulse model.
	\item The tract length distribution $L_i$ follows a Lomax-distribution, i.e. has the analytically density $Pr(L=l) = \frac{k \lambda t}{(1 + \lambda l t)^{k+1}}$
	\item The cdf is $Pr(L > l) = (1+\lambda l t )^{-k}$
	\item The mean tract-length is  $\frac{1}{\lambda t}$ for all $k > 0$, and undefined otherwise. 
	\item As $k$ approaches infinity, we recover the exponential distribution. Thus, if tract length are directly inferred, one can use a likelihood-ratio test to distinguish continuous from discrete gene flow. As the special case of exponentially lies on the boundary of the parameter space, the test-statistic does not follow a $\chi^2$-distribution (https://pdfs.semanticscholar.org/e6a3/ae271354701ecca576cf94821869f6069e9e.pdf). This is however not possible when tract length are indirectly inferred by ALD.
\end{itemize}
	
<!-- Using this model, we are interested in calculating whether i) the parameters of the Lomax-distribution can be estimated accurately for the parameter region relevant for Neandertal admixture and ii) for which parameters we can distinguish a pulse-like admixture event (resulting in exponentially-distributed track lengths) from a continuous event (resulting in Lomax-distributed track lengths). -->

Using this model we i) examining the effect of continuous admixture on the admixture time estimates calculated using the exponential model assuming a pulse like admixture. ii) comparing this effect to the effects by demography, recombination rate and analysis parameters used for the indirect inference of admixture tract length using ALD, namely the ascertainment scheme and the minimal distance between SNPs. iii) are interested under which conditions the parameters of the Lomax-distribution can be estimated accurately for a scenario of Neandertal admixture and if it is possible to distinguish a pulse-like admixture event (resulting in exponentially-distributed track lengths) from a continuous event (resulting in Lomax-distributed track lengths).

```{r message=FALSE, echo=FALSE,warning=FALSE}
suppressPackageStartupMessages({
  library(VGAM)
  library(tidyverse)
  library(ggplot2)
  library(reshape)
  library(viridis)
  library(ggpubr)
  library(dplyr)
})

Results_Table <- function(Table_Path) {
  header_for_Result=c("A","m","c","RSS_Expo","error","Scenario","GF_Start","GF_Stop","Ascertained","min_dist","Gene_Flow_Model")
  Raw_results <-  read.table(Table_Path, header = F,col.names = header_for_Result)
  Raw_results$mean.t.GF <- rowMeans(Raw_results[c('GF_Start', 'GF_Stop')], na.rm=TRUE)
  Raw_results$length.t.GF <- Raw_results$GF_Stop - Raw_results$GF_Start
  Raw_results$GF[Raw_results$length.t.GF== 1]="0_Pulse"
  Raw_results$GF[Raw_results$length.t.GF > 1]="Continous"
  Raw_results$Ascertainment[Raw_results$Ascertained== 0]="0_LES"
  Raw_results$Ascertainment[Raw_results$Ascertained == 1]="HES"
  Raw_results$min_dist <- as.factor(Raw_results$min_dist)
  Raw_results$GF[Raw_results$Gene_Flow_Model== "GF_Model_I"]="0_Pulse"
  return(Raw_results)
}

cbPalette <- c( "#56B4E9", "#E69F00", "#009E73", "#F0E442", "#0072B2", "#D55E00")
cbPalette_viridis <- viridis(6,option = "D")

Get_Data_Table <- function(Data){
  xx=as.data.frame(table(round(Data$m,digits = 0), paste(Data$GF,Data$mean.t.GF, sep="_"),Data$mean.t.GF,Data$GF,Data$mean.t.GF,Data$length.t.GF))
  xx=subset(xx,Freq>0)
  xx=xx[order(xx$Var3),]
  return(xx)
}

Figure_1_C_1 <- function(input,log,Colour_P){
  Get_points <- function(Raw_data,log){
    # Read input file
    data <- read.table(Raw_data, header = F)
    
    
    # set dist and wcorr
    col=2
    dist <- data[,1]
    wcorr <- data[,col]
    ndist <- length(dist)  ## number of rows in dataset
    lval=0.05
    hval=0.6
    
    # check x lower value and y lower value
    data.sub <- data
    if ((lval > dist[1]) || (hval < dist[ndist])) {
      data.sub <- subset(data, ((dist <= hval) & (dist >= lval)))
    } 
    dist <- data.sub[,1]		# updated x values
    wcorr <- data.sub[,col]		# updated y values
    if(log==T){
      xx <- cbind(dist,log(wcorr))
      xx <- xx[complete.cases(xx),]
      wcorr <- xx[,2]
      wcorr <-c(wcorr,rep(NA,length(data.sub[,1])-length(wcorr)))
      dist <- xx[,1]
      dist <-c(dist,rep(NA,length(data.sub[,1])-length(dist)))
    }
    
    return(data.frame(dist,wcorr))
  }
  
  Pulse <- Get_points(input[1],log)
  Continous <- Get_points(input[2],log)
  P_C_Data <- data.frame(Pulse,Continous)
  P_C_Data <- P_C_Data[P_C_Data$wcorr > -13,]
  
  Px <-  ggplot(data=P_C_Data,aes(y=wcorr,x=dist))+
    geom_point(aes(x=dist,y=wcorr),color=Colour_P[1],pch=4)+
    geom_point(aes(x=dist.1,y=wcorr.1),color=Colour_P[6],pch=18)+
    labs(y = "log weighted LD")+
    labs(x = "Genetic Distance in cM")
  #coord_cartesian(ylim = c(-13,-8),xlim=c(0,0.4),expand = 0)
  
  return(Px)
  
}

Figure_All_Real_Data <- function(input,log,GF_length,Colour_P){
  Get_points <- function(Raw_data,log){
    # Read input file
    data <- read.table(Raw_data, header = F)
    
    
    # set dist and wcorr
    col=2
    dist <- data[,1]
    wcorr <- data[,col]
    ndist <- length(dist)  ## number of rows in dataset
    lval=0.05
    hval=0.5
    
    # check x lower value and y lower value
    data.sub <- data
    if ((lval > dist[1]) || (hval < dist[ndist])) {
      data.sub <- subset(data, ((dist <= hval) & (dist >= lval)))
    } 
    dist <- data.sub[,1]		# updated x values
    wcorr <- data.sub[,col]		# updated y values
    if(log==T){
      xx <- cbind(dist,log(wcorr))
      xx <- xx[complete.cases(xx),]
      wcorr <- xx[,2]
      wcorr <-c(wcorr,rep(NA,length(data.sub[,1])-length(wcorr)))
      dist <- xx[,1]
      dist <-c(dist,rep(NA,length(data.sub[,1])-length(dist)))
    }
    
    return(data.frame(dist,wcorr))
  }
  
  Real_Data <- c()
  for(i in input){
    xx <- Get_points(i,log)
    Real_Data <- c(Real_Data,xx)
  }
  Real_Data <- as.data.frame(Real_Data)
  Real_Data_names <- c()
  for(i in GF_length){
    Real_Data_names <- c(Real_Data_names,c('Genetic_Distance',i))
  }
  colnames(Real_Data) <- Real_Data_names
  Real_Data.melted <- melt(Real_Data, id = c("Genetic_Distance"))
  
  Px <-  ggplot(data =Real_Data.melted, aes(x = Genetic_Distance, y = value,color=variable)) +
    geom_point(pch=18)+
    coord_cartesian(ylim = c(-8,-14),expand = 0,xlim = c(0,0.5))+
    #geom_vline(xintercept=Split_time,linetype=3,color='black')+
    scale_color_manual(values = Colour_P)+
    labs(x = "Genetic distance")+
    labs(y = "log weighted LD")+
    labs(color="Gene Flow Length")
  return(Px)
  
}

Figure_1_C_2 <- function(Timespan_GF_Models,time_l,Split_time,Mean_Time,Colour_P){
  
  n_GF_Models=length(Timespan_GF_Models)
  time=seq(1,time_l,1)
  
  Gamma_fun <- function(GF_Length,time_l,Split_time,Mean_Time){
    time=seq(1,time_l,1)
    EX= Mean_Time
    VarX= (GF_Length/4)**2
    b= EX/VarX
    a=EX*b
    
    GF_gamma <- dgamma(x=time,shape = a+1,scale = 1/b)
    GF_gamma[GF_gamma < 1e-6] = 0
    GF_gamma <- GF_gamma[1:Split_time]
    m2 <- c()
    for (i in GF_gamma){
      x <- i*(0.03/sum(GF_gamma))
      m2 <- c(m2,x)
    }
    m2 <- c(m2,rep(0,time_l-Split_time))
    cutoff_in_Percent=(sum(m2[2550:5001])/0.03)*100
    
    GF <- c(m2)
    return(GF)
  }
  GF <- c(seq(1,time_l,1))
  for(i in 1:n_GF_Models){
    Timespan <- Timespan_GF_Models[i]
    GF_x <- Gamma_fun(Timespan,time_l,Split_time,Mean_Time)
    GF <- cbind(GF,GF_x)
  }
  GF <- as.data.frame(GF)
  colnames(GF) <- c('Time',Timespan_GF_Models)
  GF.melted <- melt(GF, id = "Time")
  Px <-  ggplot(data =GF.melted, aes(x = Time, y = value,color=variable)) +
    geom_line(show.legend = F)+
    coord_cartesian(ylim = c(0,max(GF$'2')),expand = 0)+
    geom_vline(xintercept=Split_time,linetype=3,color='black')+
    scale_color_manual(values = Colour_P)+
    labs(x = "Time in Generations")+
    labs(y = "Migration Rate")+
    labs(color="Gene Flow Length")
  return(Px)
}

Theoratical_Lomax <- function(Timespan_GF_Models,max_Genetic_distance,Split_time,Mean_Time,Intercept,Colour_P){
  n_GF_Models=length(Timespan_GF_Models)
  
  Lomax_fun <- function(GF_Length,max_Genetic_distance,Split_time,Mean_Time,Intercept,log=T){
    Genetic_length=seq(0.01,max_Genetic_distance,0.01)
    EX= Mean_Time
    VarX= (GF_Length/4)**2
    b= EX/VarX
    a=EX*b
    Theta=b
    k=a
    Lomax_normal <- function(dist,k,theta,A) A*(1 + ((dist/100) /  theta))^-(k)
    Lomax_log <- function(dist,k,theta,A) -k* log(1 + ((dist/100) /  theta)) + log(A)
    if(log==T){
      ALD <-  Lomax_log(dist=Genetic_length,k = k ,theta = Theta, A=Intercept)
    }
    else{
      ALD <-  Lomax_normal(dist=Genetic_length,k = k ,theta = Theta, A=Intercept)
    }
    Lomax_Result <- cbind(Genetic_length,ALD,k,Theta)
    return(Lomax_Result)
  }
  Lomax_Result <- c()
  for(i in Timespan_GF_Models){
    xx=Lomax_fun(i,max_Genetic_distance,Split_time,Mean_Time,Intercept,log=T)
    Lomax_Result <- cbind(Lomax_Result,xx)
  }
  Lomax_Result <- as.data.frame(Lomax_Result)
  colnames_Lomax_Result <- c()
  for(time in Timespan_GF_Models){
    colnames_Lomax_Result <- c(colnames_Lomax_Result,c('Genetic_Distance',time,'k','Theta'))
  }
  colnames(Lomax_Result) <- colnames_Lomax_Result
  Lomax_Result.melted <- melt(Lomax_Result, id = c("Genetic_Distance",'k','Theta'))
  Px <-  ggplot(data =Lomax_Result.melted, aes(x = Genetic_Distance, y = value,color=variable)) +
    geom_line()+
    #coord_cartesian(ylim = c(0,max(ALD$'2')),expand = 0)+
    coord_cartesian(ylim = c(-8,-14),expand = 0,xlim = c(0,0.5))+
    #geom_vline(xintercept=Split_time,linetype=3,color='black')+
    scale_color_manual(values = Colour_P)+
    labs(x = "Genetic distance")+
    labs(y = "log weighted LD")+
    labs(color="Gene Flow Length")
  return(Px)
}


######################## GG PLotting ###########################

## ggplot Figure 1 A
Plot_Fig_1_A <- function(Data,Colour_P){
  Px <-  ggplot(Data,aes(x=Var4,y=as.numeric(as.character(Var1)),colour=factor(Var4)))+
    geom_point(aes(size = Freq),show.legend = F)+
    geom_boxplot()+
    facet_grid(~as.factor(Var3),switch = "x")+
    geom_hline( aes(yintercept = as.numeric(as.character(Data$Var3)) ))+
    labs(x = "True Admixture Time")+
    labs(y = "Estimated Admixture Time")+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    coord_cartesian(ylim = c(0,2250), expand = 0)+
    scale_color_manual("Gene Flow Model",
                       values = c(Colour_P[1],Colour_P[4]),
                       labels = c("Pulse","Continuous"))
  return(Px)
  
}


### Figure 1 ggplot ###
Plot_Fig_1_B <- function(Data,Colour_P){
  Px <-  ggplot(Data,aes(x=Var7,y=as.numeric(as.character(Var1)),colour=factor(Var6)))+
    geom_point(aes(size = Freq),show.legend = F)+
    geom_boxplot()+
    facet_grid(~as.factor(Var6),switch = "x")+
    geom_hline( aes(yintercept = 1500 ))+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    labs(x = "Pulse length")+
    labs(y = "Estimated Admixture Time")+
    scale_color_manual("Gene Flow Length",
                       values = Colour_P  
                       
    )
  return(Px)
  
}

```

```{r fig1,message=FALSE, echo=FALSE,warning=FALSE,fig1.pos="H",fig.width=9,fig.height=3,fig.cap="\\label{fig:fig1} A) Migration rate per generation modeled using a Gamma distribution for different gene flow length, dotted line indicates maximum time of gene flow. B) The expected LD decay modeled as a Lomax distribution for the different length. C) The observed LD decay from msprime simulations" }

## Visualization of gamma theoretical lomax LD and real data LD
ALL_Continuous_Data <- c('Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_11_-run0-GF_Model_IV-ascertainment-0.txt',
                         'Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_12_-run0-GF_Model_IV-ascertainment-0.txt',
                         'Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_13_-run0-GF_Model_IV-ascertainment-0.txt',
                         'Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_14_-run0-GF_Model_IV-ascertainment-0.txt',
                         'Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_15_-run0-GF_Model_IV-ascertainment-0.txt',
                         'Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_16_-run0-GF_Model_IV-ascertainment-0.txt')

P_Gamma <- Figure_1_C_2(c(1,200,400,800,1000,1500),3000,2550,1500,Colour_P = cbPalette_viridis)

P_Lomax <-Theoratical_Lomax(c(1,200,400,800,1000,1500),0.5,2550,1500,0.0004,Colour_P = cbPalette_viridis)

P_real <- Figure_All_Real_Data(ALL_Continuous_Data,log=T,c(1,200,400,800,1000,1500),Colour_P = cbPalette_viridis)

ggarrange(P_Gamma,P_Lomax,P_real,
          labels = c("A","B", "C"),
          ncol = 3, nrow = 1,common.legend = T,legend = 'bottom')
```


##The effect of continuous admixture on the admixture time estimates

```{r message=FALSE, echo=FALSE,warning=FALSE}
#Data
Figure_1_A_Data <- Results_Table("Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_1_A_new-GF_Model_I-min_dist_Fit-0.05-ascertainment-0.txt")
Figure_1_A_Data <- Figure_1_A_Data[Figure_1_A_Data$error=="no_error",] 
Figure_1_B_Data <- Results_Table("Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_1_B-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
Figure_1_C_1_Data <- c('Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_11_-run0-GF_Model_IV-ascertainment-0.txt','Results_Fig_1_2/Differnt_GF_length_Data/Raw_ALDER_output-_Scenario_16_-run0-GF_Model_IV-ascertainment-0.txt')

Fig_A_1 <- Get_Data_Table(Figure_1_A_Data)
Fig_A_1$Var1 <- as.numeric(as.character(Fig_A_1$Var1))
Fig_A_1$Var3 <- as.numeric(as.character(Fig_A_1$Var3))
Fig_A_1_Means <- aggregate(Fig_A_1[,c(1,3)],list(Fig_A_1$Var2), mean)
Fig_A_1_Means_diff <- data.frame(sqrt((Fig_A_1_Means$Var1-Fig_A_1_Means$Var3)^2)/Fig_A_1_Means$Var3)
Fig_A_1_Means_diff_pulse <- round(range(Fig_A_1_Means_diff[1:5,]*100))
Fig_A_1_Means_diff_continuous <- round(range(Fig_A_1_Means_diff[6:10,]*100))
```
To asses the effect of continuous ancient admixture on the admixture time estimates calculated from modern human populations, we compared two models of admixture between Neandertals and non-Africans using coalescent simulations. The simplified model of a pulse like gene flow and a continuous admixture event with several generations of gene flow. The total amount of gene flow $m$ from Neandertals into non-Africans in the two models is equal. 
Gene flow is modeled using a Gamma distribution  (Eq. \ref{eq:2}) holding the migration rate per generation for different length of continuous gene flow (Figure \ref{fig:fig1} A). The shape and scale parameter of the Gamma distribution are chosen such that the resulting weighted LD decay curves as functions of genetic distance share the same mean for a pulse (Eq. \ref{eq:1}) and a continuous admixture (Eq. \ref{eq:3}).  Sites informative for Neandertal introgression into non-Africans where enriched  using the lower-enrichment ascertainment scheme filtering for SNPs ancestral in all Africans and polymorphic in Neandertals. The pairwise weighted LD between the ascertained SNPs was computed using the ALDER program. Figure \ref{fig:fig1} C shows the resulting weighted LD for different length of continuous admixture ranging from a one generation pulse to 1500 generations.

Comparing estimates for different mean admixture times ranging from 250 generations ago to 2000 generations ago, simulated either with a pulse or a continuous admixture with a length of 50 % of the mean admixture time, reveals minor deviations between the two scenarios and the true admixture time of `r Fig_A_1_Means_diff_pulse[1]`% to `r Fig_A_1_Means_diff_pulse[2]`% for the pulse and `r Fig_A_1_Means_diff_continuous[1]`% to `r Fig_A_1_Means_diff_continuous[2]`% for the continuous (Figure \ref{fig:fig2} A). Estimates for mean admixture times older than 1000 generations show a slight overestimation, which is lesser for the estimates of simulations under a continuous gene flow . The slight overestimation is consistent with the findings of previous studies estimating ancient admixture times using modern admixed populations [@sankararaman_date_2012;@fu_genome_2014;@moorjani_genetic_2016]. To further investigate the effect of  pulse and continuous gene flow on the admixture time estimates, we compared different durations of continuous gene flow simulated under a fixed mean time of admixture of 1500 generations ago, displayed in Figure \ref{fig:fig2} B. Estimates between pulse and continuous gene flow start to deviate for 800 generations of continuous admixture, with increasingly lower estimates for simulations under continuous gene flow compared to simulations under a pulse like gene flow per increase of gene flow duration. This bias is likely caused by the differences in LD between sites entered in the tails of the gamma distribution. LD between sites arising from early admixture events, simulated by the right tail of the gamma distribution, is not detected anymore, while LD between sites from late admixture is still present, biasing the estimate towards younger dates. However, deviations in estimates between the two scenarios of ~100 generations in the most extreme case are moderate compared to the mean admixture time of 1500 generations, making admixture time estimates for long continuous gene flow compatible with the used method.


```{r fig2,message=FALSE, echo=FALSE,warning=FALSE,fig2.pos="H",fig.width=8,fig.height=8,fig.cap="\\label{fig:fig2} A) Comparison of mean admixture time estimates between pulse and continuous gene flow for different admixture times. The length of continuous gene flow corresponds to 50% of the mean admixture time, black line indicates true mean admixture time. B) Comparison of mean admixture time estimates for simulations with a  mean time of admixture of 1500 generations ago, at a varying length of gene flow. Boxplot created from 100 simulation replicates, respectively."}

### Plot in one window

P1_1 <- Plot_Fig_1_A(Fig_A_1,Colour_P = cbPalette_viridis)

Fig_B_1 <- Get_Data_Table(Figure_1_B_Data)
Fig_B_1$Var7 <- rep('xx',length(Fig_B_1$Var1))
P1_2 <- Plot_Fig_1_B(Fig_B_1,Colour_P = cbPalette_viridis)
#P1_3 <- Figure_1_C_1(Figure_1_C_1_Data,log=T,Colour_P = cbPalette_viridis)

#P1_4 <- Figure_1_C_2(c(1,200,400,800,1000,1500),3000,2550,1500,Colour_P = cbPalette_viridis)

#ggarrange(P1_4,P1_3,P1_1,P1_2,
#          labels = c("A","B", "C","D"),
#          ncol = 2, nrow = 2,common.legend = F,legend = 'bottom')

ggarrange(P1_1,P1_2,
          labels = c("A","B"),
          ncol = 2, nrow = 2,common.legend = F,legend = 'bottom', align = "h")
```


##Comparing effect sizes

```{r message=FALSE, echo=FALSE,warning=FALSE}

##### GLM for the bais on the admixture dates #####


### Read in data ###

Simple_D_Sim_1 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_A-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
Simple_D_Sim_1$Demography <- "0_Simple"
Simple_D_Sim_1$Recomb.rate <- "0_constant"
Simple_D_Sim_2 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_A-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
Simple_D_Sim_2$Demography <- "0_Simple"
Simple_D_Sim_2$Recomb.rate <- "0_constant"
Inferred_D_Sim_1 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_B_complex-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
#Inferred_D_Sim_1 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_B-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
Inferred_D_Sim_1$Demography <- "Inferred"
Inferred_D_Sim_1$Recomb.rate <- "0_constant"
Inferred_D_Sim_2 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_B_complex-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
#Inferred_D_Sim_2 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_B-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
Inferred_D_Sim_2$Demography <- "Inferred"
Inferred_D_Sim_2$Recomb.rate <- "0_constant"
Recom_Sim_1 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_C-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
Recom_Sim_1$Demography <- "0_Simple"
Recom_Sim_1$Recomb.rate <- "variable"
Recom_Sim_1$m <- Recom_Sim_1$m
Recom_Sim_2 <- Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_C-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
Recom_Sim_2$Demography <- "0_Simple"
Recom_Sim_2$Recomb.rate <- "variable"
Recom_Sim_2$m <- Recom_Sim_2$m
Recom_Sim_and_Inf_1 <-  Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_D_Complex_recomb_map_-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
#Recom_Sim_and_Inf_1 <-  Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_D-GF_Model_IV-min_dist_Fit-0.02-ascertainment-0.txt")
Recom_Sim_and_Inf_1$Demography <- "Inferred"
Recom_Sim_and_Inf_1$Recomb.rate <- "variable"
Recom_Sim_and_Inf_1$m <- Recom_Sim_and_Inf_1$m
Recom_Sim_and_Inf_2 <-  Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_D_Complex_recomb_map_-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
#Recom_Sim_and_Inf_2 <-  Results_Table("~/Desktop/ATE_Paper/Admixture_Time_Inference_Paper/Results_Fig_1_2/Result_file_SIM_Raw_ALDER-Fit-Fig_2_D-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0.txt")
Recom_Sim_and_Inf_2$Demography <- "Inferred"
Recom_Sim_and_Inf_2$Recomb.rate <- "variable"
Recom_Sim_and_Inf_2$m <- Recom_Sim_and_Inf_2$m
# first build model only with Simple and Inferred demography
#xdata <- rbind(Simple_D_Sim_1,Simple_D_Sim_2,Inferred_D_Sim_1,Inferred_D_Sim_2)

######## model with response beeing the difference between the estimated Admixture time and the true one #####
# full model with Ascertainement, min dist, Demographi and recombination rate as predictore but no interactions
xdata.M.3 <- rbind(Simple_D_Sim_1,Simple_D_Sim_2,Inferred_D_Sim_1,Inferred_D_Sim_2,Recom_Sim_1,Recom_Sim_2,Recom_Sim_and_Inf_1,Recom_Sim_and_Inf_2)
xdata.M.3$Diff <- xdata.M.3$m - xdata.M.3$mean.t.GF
# remove all estimates where nls reported an error
xdata.M.3_no_error <- xdata.M.3[xdata.M.3$error=="no_error",]

# Check response
#hist(xdata.M.3_no_error$Diff)
#min(xdata.M.3_no_error$Diff)
#max(xdata.M.3_no_error$Diff)
# check combinations of predictors do they have equal numbers of observations
Combination_of_predictors <- table(xdata.M.3_no_error$Ascertainment,xdata.M.3_no_error$min_dist,xdata.M.3_no_error$Demography,xdata.M.3_no_error$GF,xdata.M.3_no_error$Recomb.rate)

# Fit linear model while assuming normal distributed residuals
res.M.3=lm(Diff~Ascertainment+min_dist+Demography+Recomb.rate+GF,data = xdata.M.3_no_error)

### check assumptions

# check colinearity
#vif(res.M.3)

#check normal distribution 
#diagnostics.plot(res.M.3)

# check model stability
#m.stab.plot(lm_check_dfbetas(res.M.3))

### check result
res.M.3.sum=summary(res.M.3)$coefficients

# get confidence intervals
m.coefs=cbind(org=coefficients(res.M.3),confint(res.M.3))

Summary_Table_res.M.3 <- cbind(m.coefs,res.M.3.sum[,-1])

```

<!-- Having established that the duration of continuous admixture under ideal circumstances is only marginally influential on admixture time estimates, we deploy more realistic simulation scenarios.
Specifically, we investigated the influence of two analysis parameter by changing the ascertainment scheme to a higher-enrichment scheme requiring additional to the LES non-Africans to be polymorphic at a SNP, and applying a larger minimal distance of 0.05 cM between SNPs used for the fitting of the exponential distribution. Additionally, we changed the demographic model from a simple to a complex one based on inferred effective population sizes and time points taken from MSMC estimates for Africans and non-Africans and PSMC estimates for the simulated Neandertals. Split times between the population where unchanged and Neandertals were sampled 750 generations before the modern humans to simulate branch shortening. We further applied the African-American-Map to simulate varying recombination rates across the 150 Mb long chromosomes, instead of a constant rate. We simulated every combination of these parameter sets resulting in 32 different sets with 100 replications each (Supp. Fig. \ref{fig:figS1}). A Gaussian linear model was applied to estimate effect sizes of the four predictors being ascertainment scheme, minimal distance, demography and recombination on the bias of admixture estimates (Supplement Table \ref{tab:tableS1}). Figure \ref{fig:fig3} shows the comparison of the bias on admixture time estimates between the previously used model (ascertainment = LES, $d_{0} = 0.02 cM$, demography = simple and recombination = constant) further refereed to as the standard model and a model with one of the four parameter changed, respectively and the corresponding model prediction. The previously observed overestimation of the standard model was estimated to be 122 (+7/-7) generations. Every parameter change results in lower estimates compared to the standard model, with the biggest difference between a constant and a varying recombination (-863 +5/-6 generations) and the smallest differences between a pulse and continuous gene flow model (-40 +5/-6 generations). Most accurate estimates are achieved using the LES ascertainment scheme in combination with a minimal distance of 0.05 cM. These analysis parameter combination shows more robustness towards changes in the demography (Supplement Figure 1). -->

Having established that the duration of continuous admixture under ideal circumstances is only marginally influential on admixture time estimates, we deploy more realistic simulation scenarios.
Specifically, we investigated the influence of two analysis parameter by changing the ascertainment scheme to a higher-enrichment scheme requiring additional to the LES non-Africans to be polymorphic at a SNP, and applying a larger minimal distance of 0.05 cM between SNPs used for the fitting of the exponential distribution. Additionally, we changed the demographic model from a simple to a complex one based on inferred effective population sizes. Addidtinaly we included deep structure in the ancestral AFrican-non-African population and a gene flow event between Africans and non-Africans after the Neandertal admixture event (Figure \ref{fig:Intro_fig} B).  We further applied the African-American-Map to simulate varying recombination rates across the 150 Mb long chromosomes, instead of a constant rate. We simulated every combination of these parameter sets resulting in 32 different sets with 100 replications each (Supp. Fig. \ref{fig:figS1}). A Gaussian linear model was applied to estimate effect sizes of the four predictors being ascertainment scheme, minimal distance, demography and recombination on the bias of admixture estimates (Supplement Table \ref{tab:tableS1}). Figure \ref{fig:fig3} shows the comparison of the bias on admixture time estimates between the previously used model (ascertainment = LES, $d_{0} = 0.02 cM$, demography = simple and recombination = constant) further refereed to as the standard model and a model with one of the four parameter changed, respectively and the corresponding model prediction. The previously observed overestimation of the standard model was estimated to be `r round(Summary_Table_res.M.3["(Intercept)","org"])` (`r round(Summary_Table_res.M.3["(Intercept)","2.5 %"])` - `r round(Summary_Table_res.M.3["(Intercept)","97.5 %"])` 95 % CI) generations. Every parameter change results in lower estimates compared to the standard model, with the biggest difference between a constant and a varying recombination (`r round(Summary_Table_res.M.3["Recomb.ratevariable","2.5 %"])` - `r round(Summary_Table_res.M.3["Recomb.ratevariable","97.5 %"])` generations) and the smallest differences between a pulse and continuous gene flow model (`r round(Summary_Table_res.M.3["GFContinous","2.5 %"])` - `r round(Summary_Table_res.M.3["GFContinous","97.5 %"])` generations). Most accurate estimates are achieved using the LES ascertainment scheme in combination with a minimal distance of 0.05 cM. The bias introduced by the demography seems to depend on the used ascerteinent scheme, since only under the HES a difference between the simple and complex demography is evident (Supplement Figure 1). Overall the bias introduced by the ascerteinment, minimal distance cutoff, demography and gene flow model are 10 % or less of the true mean admixture time.


```{r fig3,message=FALSE, echo=FALSE,warning=FALSE,fig4.pos="H",fig.width=9,fig.height=6,fig.cap="\\label{fig:fig3} Comparison of the bias on the admixture time estimates between each parameter and the standard model of 100 replications each. Dotted line represents the model prediction with the 2.5 % and 97.5 % compatibility intervals solid lines. Dotted horizontal line indicates unbiased admixture estimates."}
#### Ploting the model ####

### functions ###
Get_colour <- function(xx,col1,col2){
  is.even <- function(x) x %% 2 == 0
  for (i in 1:length(as.numeric(xx$Var2))) {
    if (is.even(i) == T) {
      xx$col[as.numeric(xx$Var2)==i] <- col2
    } else{
      xx$col[as.numeric(xx$Var2)==i] <- col1
    }
    
  }
  return(xx)
}

Plot_Fig_2_X <- function(Null_Case,Predictore_case,m.intercept,m.slope,x.lable,x.title,xlab=NULL,ylab=NULL,y.ticks=F){
  Data=rbind(Null_Case,Predictore_case)
  if(y.ticks==F){
    Px <-  ggplot(Data,aes(x=Var2,y=as.numeric(as.character(Var1)),colour=factor(Var3)))+
      geom_point(aes(size = Freq),show.legend = F)+
      geom_boxplot(show.legend = F)+
      geom_abline(intercept = m.intercept[1],slope = m.slope[1],linetype=2)+
      geom_abline(intercept = m.intercept[2],slope = m.slope[2])+
      geom_abline(intercept = m.intercept[3],slope = m.slope[3])+
      geom_abline(intercept = 0,slope = 0,linetype=2,aes(colour='grey'))+
      labs(x = xlab)+
      labs(y = ylab)+
      scale_x_discrete(labels= x.lable)+
      theme(axis.text.x = element_text(angle = 0,size = 12),
            axis.text.y=element_blank(),
            axis.ticks.y =element_blank())+
      coord_cartesian(ylim = c(-1100,300), expand = 0)+
      ggtitle(x.title)+
      theme(plot.title = element_text(hjust = 0.5,size = 12))+
      scale_color_manual("Gene Flow Model",
                         values = c(cbPalette_viridis[1],cbPalette_viridis[6]),
                         labels = c("Pulse","Continuous"))
  }
  if(y.ticks==T){
    Px <-  ggplot(Data,aes(x=Var2,y=as.numeric(as.character(Var1)),colour=factor(Var3)))+
      geom_point(aes(size = Freq),show.legend = F)+
      geom_boxplot(show.legend = F)+
      geom_abline(intercept = m.intercept[1],slope = m.slope[1],linetype=2)+
      geom_abline(intercept = m.intercept[2],slope = m.slope[2])+
      geom_abline(intercept = m.intercept[3],slope = m.slope[3])+
      geom_abline(intercept = 0,slope = 0,linetype=2,aes(colour='grey'))+
      labs(x = xlab)+
      labs(y = ylab)+
      scale_x_discrete(labels= x.lable)+
      theme(axis.text.x = element_text(angle = 0,size = 12))+
      coord_cartesian(ylim = c(-1100,300), expand = 0)+
      ggtitle(x.title)+
      theme(plot.title = element_text(hjust = 0.5,size = 12))+
      scale_color_manual("Gene Flow Model",
                         values = c(cbPalette_viridis[1],cbPalette_viridis[6]),
                         labels = c("Pulse","Continuous"))
  }

  return(Px)
  
}

### Execute plot functions ###

Model_3=as.data.frame(table(round(xdata.M.3_no_error$Diff,digits = 0), paste(xdata.M.3_no_error$GF,xdata.M.3_no_error$Ascertainment,xdata.M.3_no_error$min_dist,xdata.M.3_no_error$Demography,
                                                                        xdata.M.3_no_error$Recomb.rate, sep="_"),xdata.M.3_no_error$GF))
Model_3=subset(Model_3,Freq>0)
xx=Get_colour(xx=Model_3,col1 ="#56B4E9" ,col2 = "#D55E00")

### Plot the simple case against the model prediction where one of the other parametres is different respectively ###
Simplest_case=xx[xx$Var2=="0_Pulse_0_LES_0.02_0_Simple_0_constant",]

PX1 <- Plot_Fig_2_X(Simplest_case,xx[xx$Var2=="0_Pulse_HES_0.02_0_Simple_0_constant",],
             m.coefs["(Intercept)",]-m.coefs["AscertainmentHES",],m.coefs["AscertainmentHES",],x.lable = c("LES","HES"),"Ascertainment\nScheme",y.ticks = T)

PX2 <- Plot_Fig_2_X(Simplest_case,xx[xx$Var2=="0_Pulse_0_LES_0.05_0_Simple_0_constant",],
             m.coefs["(Intercept)",]-m.coefs["min_dist0.05",],m.coefs["min_dist0.05",],x.lable = c("d0 = 0.02","d0 = 0.05"),"Minimal Distance\n")

PX3 <- Plot_Fig_2_X(Simplest_case,xx[xx$Var2=="0_Pulse_0_LES_0.02_Inferred_0_constant",],
             m.coefs["(Intercept)",]-m.coefs["DemographyInferred",],m.coefs["DemographyInferred",],x.lable = c("Simple","Complex"),"Demography\n")

PX4 <- Plot_Fig_2_X(Simplest_case,xx[xx$Var2=="0_Pulse_0_LES_0.02_0_Simple_variable",],
             m.coefs["(Intercept)",]-m.coefs["Recomb.ratevariable",],m.coefs["Recomb.ratevariable",],x.lable = c("Constant","Variable"),"Recombination Map\n")

PX5 <- Plot_Fig_2_X(Simplest_case,xx[xx$Var2=="Continous_0_LES_0.02_0_Simple_0_constant",],
             m.coefs["(Intercept)",]-m.coefs["GFContinous",],m.coefs["GFContinous",],x.lable = c("Pulse","Continous"),"Gene Flow\n")

Figure_2_X <- ggarrange(PX1,PX2,PX3,PX4,PX5,
          ncol = 5, nrow = 1)
annotate_figure(Figure_2_X,bottom = text_grob("Predictor combination"),left = text_grob("Bias on admixture time estimate",rot=90))

```

##Estimating the Lomax-parameters under different conditions

```{r message=FALSE, echo=FALSE,warning=FALSE}
#### New Figure for Paper using recent sampling (50 gen): constant RR, RM no correction, RM diff RM correction, RM correction same RM ####

library(ggplot2)
library(reshape2)
library(viridis)
library(ggpubr)
cbPalette_viridis <- viridis(6,option = "D")

Filter_Result_Table <- function(Result.Table){
  Result.Table=read.table(Result.Table,header=F,sep = " ")
  Result.names <- c('A.exp', 's.exp', 'c.exp', 'RSS_Expo','AIC_Expo', 'A.lomax', 's.lomax', 'w.lomax','c.lomax', 'RSS_Lomax','AIC_Lomax', 'F_Test','Scenario.name','GF.Start','GF.End','AS','minDist','GF.Model')
  colnames(Result.Table) <- Result.names
  #Result.Table <- Result.Table[(1/Result.Table$s.exp)>0,]
  #Result.Table <- Result.Table[(1/Result.Table$s.exp)<5000,]
  #Result.Table <- Result.Table[Result.Table$RSS_Lomax<1e-2,]
  return(Result.Table)
}

#### True values
True_params <- function(Result.Table){
  True_params_Calc <- function(True_GF_length,True_mean_GF){
    EX= True_mean_GF
    GF_Len <-True_GF_length
    VarX= ((GF_Len)/4)**2
    b= EX/VarX
    a=EX*b
    #a=a+1
    True_W=1/a
    True_S=b/(1/True_W)
    xx=c(True_W,True_S)
    return(xx)
  }
  True.params <- c()
  for (i in 1:length(Result.Table$F_Test)) {
    xx=True_params_Calc(Result.Table$True_GF_length[i],Result.Table$True_mean_GF[i])
    True.params <- rbind(True.params,xx)
  }
  True.params <- as.data.frame(True.params)
  return(True.params)
}

Result.Table.fn <- function(Result.Table.path,Sampling.time.from.GF.End,name){
  Result.Table <- Filter_Result_Table(Result.Table.path)
  Result.Table$Name <- name
  Result.Table$Sample_Time <- Sampling.time.from.GF.End
  Result.Table$True_GF_length <- Result.Table$GF.End-Result.Table$GF.Start
  Result.Table$True_mean_GF <- ((Result.Table$GF.End+Result.Table$GF.Start)/2)-(Result.Table$GF.Start-Result.Table$Sample_Time)
  Result.Table$mean_GF_exp <- 1/Result.Table$s.exp
  Result.Table$mean_GF_lomax_s <- 1/Result.Table$s.lomax
  Result.Table.comparison <- True_params(Result.Table)
  Result.Table$True_W <- Result.Table.comparison$V1
  Result.Table$True_S <- Result.Table.comparison$V2
  Result.Table.comparison$length_GF <- (sqrt(Result.Table$mean_GF_lomax_s/(((1/Result.Table$w.lomax)+1)*Result.Table$s.lomax)))*4
  Result.Table$length_GF <- Result.Table.comparison$length_GF
  
  return(Result.Table)
}

Plot_sampling_50_gen_t_GF <- function(ggdata_t.GF,cbPalette_viridis){
  ggplot(ggdata_t.GF,aes(x=variable,y=as.numeric(as.character(value)),colour=factor(Name)))+
    geom_boxplot(show.legend = T)+
    facet_grid(~as.factor(True_GF_length),switch = "x")+
    geom_hline( aes(yintercept = True_mean_GF ))+
    ggtitle("All sampled 50 gen from GF end") +
    labs(x = "Gene Flow Length")+
    labs(y = "Estimated Admixture Time")+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    coord_cartesian(ylim = c(0,1000), expand = 0)+
    scale_color_manual("Simulations",
                       values = c(cbPalette_viridis[1],cbPalette_viridis[2],cbPalette_viridis[3],cbPalette_viridis[4]),
                       labels = c("Constant Recombination","HapMap AAMap corrected","HapMap HapMap corrected","HapMap not corrected"))
  
}

Plot_sampling_50_gen_GF_Length <- function(ggdata_l.GF,cbPalette_viridis){
  ggplot(ggdata_l.GF,aes(x=variable,y=as.numeric(as.character(value)),colour=factor(Name)))+
    geom_boxplot(show.legend = F)+
    facet_grid(~as.factor(True_GF_length),switch = "x")+
    geom_hline( aes(yintercept = True_GF_length ))+
    ggtitle(paste("All sampled 50 gen from GF end",sep = " ")) +
    labs(x = "Gene Flow Length")+
    labs(y = "Estimated Gene Flow Length")+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    coord_cartesian(ylim = c(0,4000), expand = 0)+
    scale_color_manual("Simulations",
                       values = c(cbPalette_viridis[1],cbPalette_viridis[2],cbPalette_viridis[3],cbPalette_viridis[4]),
                       labels = c("Constant Recombination","HapMap AAMap corrected","HapMap HapMap corrected","HapMap not corrected"))
}

Plot_sampling_varying_gen_t_GF <- function(ggdata_t.GF_varying,cbPalette_viridis){
  ggplot(ggdata_t.GF_varying,aes(x=variable,y=as.numeric(as.character(value)),colour=factor(Name)))+
    geom_boxplot()+
    facet_grid(~as.factor(Sample_Time),switch = "x")+
    geom_hline( aes(yintercept = True_mean_GF ))+
    ggtitle("All Gene Flow length = 800 gen") +
    labs(x = "Generations Sample After Gene Flow End")+
    labs(y = "Estimated Admixture Time")+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    coord_cartesian(ylim = c(0,2000), expand = 0)+
    scale_color_manual("Simulations",
                       values = c(cbPalette_viridis[1],cbPalette_viridis[2],cbPalette_viridis[3],cbPalette_viridis[4]),
                       labels = c("Constant Recombination","HapMap AAMap corrected","HapMap HapMap corrected","HapMap not corrected"))
  
}

Plot_sampling_varying_gen_GF_Length <- function(ggdata_l.GF_varying,cbPalette_viridis){
  ggplot(ggdata_l.GF_varying,aes(x=variable,y=as.numeric(as.character(value)),colour=factor(Name)))+
    #geom_point(aes(size = Freq),show.legend = F)+
    geom_boxplot(show.legend = F)+
    facet_grid(~as.factor(Sample_Time),switch = "x")+
    geom_hline( aes(yintercept = True_GF_length ))+
    ggtitle("All Gene Flow length = 800 gen") +
    labs(x = "Generations Sample After Gene Flow End")+
    labs(y = "Estimated Gene Flow  Length")+
    theme(
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank())+
    coord_cartesian(ylim = c(0,4000), expand = 0)+
    scale_color_manual("Simulations",
                       values = c(cbPalette_viridis[1],cbPalette_viridis[2],cbPalette_viridis[3],cbPalette_viridis[4]),
                       labels = c("Constant Recombination","HapMap AAMap corrected","HapMap HapMap corrected","HapMap not corrected"))
}

Result.Table.path_Recent_50 <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_no_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_AAMap_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-AAMap_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_HapMap_Recomb_HapMap_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-HapMap_correction.txt'

Result.Table.path_Recent_50_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_no_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_AAMap_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-AAMap_correction.txt'
Result.Table.path_Recent_50_Recomb_Map_HapMap_Recomb_HapMap_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Close_to_GF_End_Recent_GF_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-HapMap_correction.txt'

Result.Table.path_Recent_Varying <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_no_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_AAMap_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-AAMap_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_HapMap_Recomb_HapMap_correction <-'Result_Files_new_Plot/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-HapMap_correction.txt'

Result.Table.path_Recent_Varying_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_no_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-No_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_AAMap_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-AAMap_correction.txt'
Result.Table.path_Recent_Varying_Recomb_Map_HapMap_Recomb_HapMap_correction_fixed_Lambda <-'Result_Files_new_Plot/Fixed_Lambda/Result_file_SIM_Raw_ALDER-Fit-Variyng_Time_of_Recent_sampling_GF_l_fixed_Recomn_Map_Hap_Map-GF_Model_IV-min_dist_Fit-0.05-ascertainment-0-HapMap_correction.txt'


Plot.data_50<-rbind(
  Recent_50_constant<- Result.Table.fn(Result.Table.path_Recent_50,50,'Recent_50_constant'),
  Recent_50_HapMap_no_correction <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_no_correction,50,'Recent_50_HapMap_no_correction'),
  Recent_50_HapMap_AAMap_correction <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_AAMap_correction,50,'Recent_50_HapMap_AAMap_correction'),
  Recent_50_HapMap_HapMap_correction <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_HapMap_Recomb_HapMap_correction,50,'Recent_50_HapMap_HapMap_correction')
)

ggdata_t.GF_50 <- melt(Plot.data_50,measure.vars =  c('mean_GF_exp'),id.vars = c('True_mean_GF','True_GF_length','Name'))
ggdata_l.GF_50 <- melt(Plot.data_50,measure.vars =  c('length_GF'),id.vars = c('True_mean_GF','True_GF_length','Name'))

Plot.data_50_fixed_Lambda<-rbind(
  Recent_50_constant_fixed_Lambda<- Result.Table.fn(Result.Table.path_Recent_50_fixed_Lambda,50,'Recent_50_constant'),
  Recent_50_HapMap_no_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_no_correction_fixed_Lambda,50,'Recent_50_HapMap_no_correction'),
  Recent_50_HapMap_AAMap_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_AAMap_correction_fixed_Lambda,50,'Recent_50_HapMap_AAMap_correction'),
  Recent_50_HapMap_HapMap_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_50_Recomb_Map_HapMap_Recomb_HapMap_correction_fixed_Lambda,50,'Recent_50_HapMap_HapMap_correction')
)

ggdata_t.GF_50_fixed_Lambda <- melt(Plot.data_50_fixed_Lambda,measure.vars =  c('mean_GF_lomax_s'),id.vars = c('True_mean_GF','True_GF_length','Name'))
ggdata_l.GF_50_fixed_Lambda <- melt(Plot.data_50_fixed_Lambda,measure.vars =  c('length_GF'),id.vars = c('True_mean_GF','True_GF_length','Name'))

Plot.data_varying<-rbind(
  Recent_varying_constant<- Result.Table.fn(Result.Table.path_Recent_Varying,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_constant'),
  Recent_varying_HapMap_no_correction <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_no_correction,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_no_correction'),
  Recent_varying_HapMap_AAMap_correction <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_AAMap_correction,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_AAMap_correction'),
  Recent_varying_HapMap_HapMap_correction <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_HapMap_Recomb_HapMap_correction,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_HapMap_correction')
)

ggdata_t.GF_varying <- melt(Plot.data_varying,measure.vars =  c('mean_GF_exp'),id.vars = c('True_mean_GF','True_GF_length','Name','Sample_Time'))
ggdata_l.GF_varying <- melt(Plot.data_varying,measure.vars =  c('length_GF'),id.vars = c('True_mean_GF','True_GF_length','Name','Sample_Time'))

Plot.data_varying_fixed_Lambda<-rbind(
  Recent_varying_constant_fixed_Lambda<- Result.Table.fn(Result.Table.path_Recent_Varying_fixed_Lambda,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_constant'),
  Recent_varying_HapMap_no_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_no_correction_fixed_Lambda,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_no_correction'),
  Recent_varying_HapMap_AAMap_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_AAMap_correction_fixed_Lambda,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_AAMap_correction'),
  Recent_varying_HapMap_HapMap_correction_fixed_Lambda <- Result.Table.fn(Result.Table.path_Recent_Varying_Recomb_Map_HapMap_Recomb_HapMap_correction_fixed_Lambda,c(rep(50,100),rep(100,100),rep(200,100),rep(400,100),rep(800,100),rep(1000,100)),'Recent_50_HapMap_HapMap_correction')
)

ggdata_t.GF_varying_fixed_Lambda <- melt(Plot.data_varying_fixed_Lambda,measure.vars =  c('mean_GF_lomax_s'),id.vars = c('True_mean_GF','True_GF_length','Name','Sample_Time'))
ggdata_l.GF_varying_fixed_Lambda <- melt(Plot.data_varying_fixed_Lambda,measure.vars =  c('length_GF'),id.vars = c('True_mean_GF','True_GF_length','Name','Sample_Time'))


New.P1 <- Plot_sampling_50_gen_t_GF(ggdata_t.GF_50,cbPalette_viridis)
New.P2 <- Plot_sampling_50_gen_GF_Length(ggdata_l.GF_50,cbPalette_viridis)
New.P3 <- Plot_sampling_varying_gen_t_GF(ggdata_t.GF_varying,cbPalette_viridis)
New.P4 <- Plot_sampling_varying_gen_GF_Length(ggdata_l.GF_varying,cbPalette_viridis)

New.P1_fixed_Lambda <- Plot_sampling_50_gen_t_GF(ggdata_t.GF_50_fixed_Lambda,cbPalette_viridis)
New.P2_fixed_Lambda <- Plot_sampling_50_gen_GF_Length(ggdata_l.GF_50_fixed_Lambda,cbPalette_viridis)
New.P3_fixed_Lambda <- Plot_sampling_varying_gen_t_GF(ggdata_t.GF_varying_fixed_Lambda,cbPalette_viridis)
New.P4_fixed_Lambda <- Plot_sampling_varying_gen_GF_Length(ggdata_l.GF_varying_fixed_Lambda,cbPalette_viridis)
```



We investigated the effect of continuous admixture in comparison to other parameters on the admixture time estimates of a model only considering a single generation pulse. The continuous admixture revealed to have the smallest effect, whereas the recombination rate shows a sever underestimation of the admixture time. Building on the previous result we want to find out the conditions to retrieve the Lomax parameters i.e. the duration of gene flow. We simulated an admixture scenario under a simple demography with varying duration of continuous admixture and sampled the population at different time points since the end of the gen flow. Doing so allows us to examine how close one has to sample form the end of a continuous admixture to still accurately estimate its duration. Moreover, since the recombination map is highly influential we test the duration estimates under a constant and variable recombination using the HapMap genetic map, whereby we correct the genetic distance by: assuming a constant rate, using the AAMap or the HapMap itself. We used the LES ascertainment scheme in combination with a minimal distance of 0.05 cM. To fit the Lomax we can take advantage of the fact that the bias of the mean time estimates using the simplified one generation pulse model is pretty accurate. We can thus estimate lambda first by fitting an exponential and in a second step estimating k using a Lomax distribution with a starting parameter for lambda received form the exponential fit. Figure \ref{fig:fig4} A and C show the mean time estimates received from the Exponential fit. A shows estimates for different durations of continuous admixture with all populations sampled 50 generations after the end of gene flow. C shows estimates for a 800 generation long continuous admixture sampled after different times after the end of the gene flow. In both scenarios, for simulations under a constant recombination the mean time can be estimated confidently with only slight underestimation for long continuous gene flow of 1500 generations. Estimates for simulation under a recombination map only assuming a constant rate when calculating the LD results in severely underestimation of the mean time estimates. Using the AAMap yields results closer to the true value and less downwards biased, however only using the exact same genetic map gives unbiased results. Figure \ref{fig:fig4} B and D shows the corresponding duration estimates by the Lomax fit. Accurate estimates can be obtained throughout the different gene flow length under a constant recombination rate, when sampled recent from the end of the continuous admixture. The further away one samples form the end of the admixture event the less accurate the estimates. All simulations using a recombination map show a much higher variance in the estimates, especially for a gene flow length longer then 800 generations or sampled later then 50 generations away from the end of gene flow the estimates are not reliably. If no precise genetic map is used to inferred the genetic distance between SNPs, no accurate duration estimates can be obtained. The mean duration over all replicates for the simulation corrected by the exact same map seems relatively unbiased for a recent gene flow with a duration shorter 1000 generations. Under a realistic scenario with a varying recombination rate, the duration can only be accurately estimated with a highly precise map for recent continuous admixture events not longer then 400 generations. With regard to scenarios of Neandertal admixture, accurately estimating the duration of possible continuous admixture from present day human genomes even under a constant rate is probably is underpowered. The time since the end of the gene flow when using present day human genome is to far away such that the signal of continuous gene flow, even under simple simulation scenarios using a constant recombination rate, is not strong enough. 



```{r message=FALSE, echo=FALSE,warning=FALSE}
Get_model_selection <- function(Data,GF_length,Sim_Scenario){
  if (GF_length==1){
    xx <- Data[Data$True_GF_length==GF_length,]
  } else {
    xx <- Data[Data$True_GF_length>=GF_length,]
  }

  xx <- xx[xx$Name==Sim_Scenario,]
  
  
  Expo_prefered <- xx[xx$AIC_Expo<xx$AIC_Lomax,]
  
  Lomax_prefered <- length(xx$A.exp)-length(Expo_prefered$A.exp)
  
  Model_select <- cbind(length(Expo_prefered$A.exp),Lomax_prefered)
  colnames(Model_select) <- c("Expo_prefered","Lomax_prefered")

  return(Model_select)
}

Pulse_model_selction <- Get_model_selection(Plot.data_50,1,'Recent_50_constant')
Continuous_model_selection <- Get_model_selection(Plot.data_50,2,'Recent_50_constant')
```


To distinguish a pulse like admixture event resulting in an exponentially distributed ALD decay from a Lomax distributed ALD decay associated with a continuous admixture, we compared the two model via goodness-of-fit to the ALD data. Therefore we used Akaike's information criterion comparing the exponential model nested in the Lomax. We expect that, since the AIC is penalizing extra parameter, that a Lomax fit to a pulse like simulation scenario should be rejected, wheres for all the other scenarios the Lomax would be preferred. However, for the simplified simulation scenarios under a constant recombination rate were we obtained the best results the Lomax displays a lower AIC in `r Pulse_model_selction[,2]` fits out of 100 replications. Model comparison for scenarios of true continuous admixture always rested in the Lomax having a lower AIC. The AIC reject the null hypothesis to often and thus seem to be a to liberal test.
<!-- does on call a a statistical test rejecting the H0 to often liberal? -->





<!-- To get a duration under realistic scenario on needs a highly precise map and genflow has to be shorter then 400 generation and recently in the past. -->

<!-- Maybe call it inferring the genetic distance rather than correcting -->

``` {r fig4, message=FALSE, echo=FALSE,warning=FALSE,fig4.pos="H",fig.width=9,fig.height=6,fig.cap="\\label{fig:fig4} ll"}


New.Fig <- ggarrange(New.P1,New.P2,New.P3,New.P4,
                   labels = c("A","B","C","D"),
                   ncol = 2, nrow = 2,common.legend = T,legend = 'bottom')

New.Fig_fixed_Lambda <- ggarrange(New.P1_fixed_Lambda,New.P2_fixed_Lambda,New.P3_fixed_Lambda,New.P4_fixed_Lambda,
                     labels = c("A","B","C","D"),
                     ncol = 2, nrow = 2,common.legend = T,legend = 'bottom')
New.Fig

#New.Fig_fixed_Lambda
```


#Discussion

*bp 19/02: this should go in the discussion*

Neandertals and modern human populations overlapped in some regions for a long period of time [@higham_timing_2014], potentially resulting in continuous admixture over hundreds of generations. We are especially interested in the extrema of a continuous admixture distribution, since the start and end point can inform us about the first contact between humans and Neandertals and the time of extinction of Neandertals.

Conclusions:
\begin{itemize}
	\item The mean time of gene flow can be reliably estimated with a pulse model even if the gene flow is actually continuous.
  \item Technical model variables (ascertainment scheme, minimal distance, demography and recombination map) outweigh the effect of continuous gene flow.
  \item The Lomax model is under powered to inferring continuous admixture for parameters relevant for Neandertal admixture.
  \item Estimating the the mean time and especially the duration of gene flow is only possible with a highly precise genetic map.
  \item The signal of continuous gene flow is hard to detect if the gene flow is not long enough or to far in the past.
  \item Reliably distinguishing the models is difficult.
\end{itemize}

#Refereces

<div id="refs"></div>


\pagebreak
\setcounter{figure}{0}
\renewcommand{\figurename}{Fig. S}
\renewcommand{\tablename}{Tab. S}


#Supplement

```{r figS1, message=FALSE, echo=FALSE,warning=FALSE,fig.width=12,fig.height=6,fig.cap="\\label{fig:figS1}Comparison of the bias on the admixture time estimates in generations between all combinations the parameters: ascertainment scheme = LES/HES,  $d_{0}$ = 0.02/0.05 cM, demography = simple/complex, recombination = constant/variable and the gene flow model = pulse/continuous.  Dotted horizontal line indicates unbiased admixture estimates."}


### Supplement Figures ? Tables

Plot_Fig_2_A <- function(Data,x.axes.lables){
  Px <-  ggplot(Data,aes(x=Var2,y=as.numeric(as.character(Var1)),colour=factor(Var3)))+
    geom_point(aes(size = Freq),show.legend = F)+
    geom_boxplot()+
    geom_abline(intercept = 0,slope = 0,linetype=2,aes(colour='grey'))+
    coord_cartesian(ylim = c(-1500,1100),expand = 0)+
    labs(x = "Predictor combination")+
    labs(y = "Bias on admixture estimate")+
    theme(axis.text.x = element_text(angle = 0))+
    scale_x_discrete(labels= x.axes.lables)+
    scale_color_manual("Gene Flow Model",
                       values = c(cbPalette_viridis[1],cbPalette_viridis[6]),
                       labels = c("Pulse","Continuous"))
  return(Px)
  
}

#x.axes.lables=c("LES\n0.02\nSim\nCon","LES\n0.02\nSim\nVar","LES\n0.02\nInf\nCon","LES\n0.02\nInf\nVar"
#                ,"LES\n0.05\nSim\nCon","LES\n0.05\nSim\nVar","LES\n0.05\nInf\nCon","LES\n0.05\nInf\nVar"
#                ,"HES\n0.02\nSim\nCon","HES\n0.02\nSim\nVar","HES\n0.02\nInf\nCon","HES\n0.02\nInf\nVar"
#                ,"HES\n0.05\nSim\nCon","HES\n0.05\nSim\nVar","HES\n0.05\nInf\nCon","HES\n0.05\nInf\nVar"
#                ,"LES\n0.02\nSim\nCon","LES\n0.02\nSim\nVar","LES\n0.02\nInf\nCon","LES\n0.02\nInf\nVar"
#                ,"LES\n0.05\nSim\nCon","LES\n0.05\nSim\nVar","LES\n0.05\nInf\nCon","LES\n0.05\nInf\nVar"
#                ,"HES\n0.02\nSim\nCon","HES\n0.02\nSim\nVar","HES\n0.02\nInf\nCon","HES\n0.02\nInf\nVar"
#                ,"HES\n0.05\nSim\nCon","HES\n0.05\nSim\nVar","HES\n0.05\nInf\nCon","HES\n0.05\nInf\nVar")

x.axes.lables=c("LES\n0.02\nSim\nCon","LES\n0.02\nSim\nVar","LES\n0.02\nCom\nCon","LES\n0.02\nCom\nVar"
                ,"LES\n0.05\nSim\nCon","LES\n0.05\nSim\nVar","LES\n0.05\nCom\nCon","LES\n0.05\nCom\nVar"
                ,"HES\n0.02\nSim\nCon","HES\n0.02\nSim\nVar","HES\n0.02\nCom\nCon","HES\n0.02\nCom\nVar"
                ,"HES\n0.05\nSim\nCon","HES\n0.05\nSim\nVar","HES\n0.05\nCom\nCon","HES\n0.05\nCom\nVar"
                ,"LES\n0.02\nSim\nCon","LES\n0.02\nSim\nVar","LES\n0.02\nCom\nCon","LES\n0.02\nCom\nVar"
                ,"LES\n0.05\nSim\nCon","LES\n0.05\nSim\nVar","LES\n0.05\nCom\nCon","LES\n0.05\nCom\nVar"
                ,"HES\n0.02\nSim\nCon","HES\n0.02\nSim\nVar","HES\n0.02\nCom\nCon","HES\n0.02\nCom\nVar"
                ,"HES\n0.05\nSim\nCon","HES\n0.05\nSim\nVar","HES\n0.05\nCom\nCon","HES\n0.05\nCom\nVar")

Plot_Fig_2_A(xx,x.axes.lables)
```


```{r tableS1, echo=FALSE,results='asis' }
kable(Summary_Table_res.M.3,
      caption = "\\label{tab:tableS1}Estimates for the predictor from the Gaussian linear model with 2.5/95.5 % compatibility intervals in generations, standard error, t and p-values.")
```

