\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fontshttps://de.overleaf.com/project/5e85b0680d0bed00011ea790
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={The dating of the Human-Neandertal introgression event estimated from present-day human genomes is compatible with a multitude of admixture durations},
            pdfauthor={Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary Anthropology, MPI EVA), Dr.~Benjamin Marco Peter (MPI EVA, benjamin\_peter@eva.mpg.de)},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{plainnat}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{The dating of the Human-Neandertal introgression event estimated from present-day human genomes is compatible with a multitude of admixture durations}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary
Anthropology, MPI EVA), Dr.~Benjamin Marco Peter (MPI EVA,
\href{mailto:benjamin_peter@eva.mpg.de}{\nolinkurl{benjamin\_peter@eva.mpg.de}})}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2020-03-24}

\usepackage{setspace}
\doublespacing
\usepackage[none]{hyphenat}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{float}
\usepackage{xcolor}
\floatplacement{figure}{H}

\begin{document}
\maketitle

\section{Abstract}\label{abstract}

\section{Introduction}\label{introduction}



\subsection{(Archaic) Interbreeding, genomic consequences and why is it interesting}\label{(Archaic) Interbreeding, genomic consequences and why is it interesting}

In recent years, the sequencing of Neandertal \citep{green_draft_2010,prufer_complete_2013,prufer_high-coverage_2017, mafessoni_high_coverage_2020} and Denisovan genomes \citep{reich_genetic_2010, meyer_high-coverage_2012} revealed that modern humans outside of Africa were in contact and interbred with other hominin populations \citep{vernot_resurrecting_2014,fu_genome_2014,fu_early_2015,sankararaman_genomic_2014,sankararaman_combined_2016,vernot_excavating_2016,malaspinas_genomic_2016}. On a genomic level, the interbreeding with a migrant population (admixture) introduces divergent chromosomes into the mixing population. Over time, meiotic recombination progressively breaks these divergent chromosomes down into admixture segments, whose size decrease with time \citep{falush_inference_2003}. Assuming that recombination events are independent from eachother, the length distribution of the admixture segments is roughly inversely proportional to the number of generations since the interbreeding happened. Hence, using this 'recombination clock', the length distribution of introgressed chromosomal segments can be used to infer the time of an
admixture event \citep{moorjani_history_2011,pugach_dating_2011,sankararaman_date_2012,loh_inferring_2013,sankararaman_combined_2016,pugach_gateway_2018,jacobs_multiple_2019,hellenthal_genetic_2014}. \citep{pool_inference_2009,moorjani_history_2011,gravel_population_2012,liang_lengths_2014}. One particular event of interest is the time and duration of interbreeding between Neandertals and modern humans , i.e. the time between start and end of sympatry of Neandertals and modern humans outside of Africa. This could give a clue to the debated questions of the time of the first modern humans leaving Africa and the extinction of Neandertals. The earliest modern human remains outside of Africa are directly dated to around 188 thousand years ago (kya)  using a combination of uranium-thorium, uranium-series and electron spin resonance dating techniques  \citep{stringer_when_2018,hershkovitz_earliest_2018} and a possible extinction of Neandertals is suggested around 37 kya inferred by radiocarbon and luminescent dating of a Neandertal site in Spain \citep{zilhao_precise_2017} and 39 kya from radiocarbon dates of Neandertal sites across Europa \citep{higham_timing_2014} at the end of the Mousterian. This leaves a potential time frame for Neandertal and modern human interaction of approximately 140,000 years. Detailed genetic dating of Neandertal and modern human admixture could unveil more evidence of the timing and especially duration of this interaction. 

\subsection{Admixture models}\label{Admixture models}

The accuracy and information content of admixture date estimates depend on the model used and its assumptions. The most widely used admixture model is the Isolation-Migration model (IM), which models divergence of two populations from an ancestral one by jointly estimating the divergence time and migration rate \citep{nielsen_distinguishing_2001,hey_multilocus_2004}. In the IM model however, migration has no time component and is assumed constant i.e. in our case from the divergence time of Neandertals and modern humans till today. Dates of Admixture events can be estimated by modeling the admixture segment length distribution. For this purpose, most models assume that admixture segments are rare and inbreeding is not significant, such that admixture segments are unlikely to recombine with eachother \citep{pool_inference_2009,liang_lengths_2014}. The segments act neutral \citep{shchur_distribution_2019} and the recombination clock is constant over time and populations \citep{gravel_population_2012}. In addition, it is generally assumed that the admixture happens over a very short time period, in a single \textit{admixture pulse} \citep{moorjani_history_2011}, usually modelled as a single generation of interbreeding.


\subsection{The two approaches and their application to find archaic admixture dates}\label{the-two-approaches-and-their-application-to-find-archaic-admixture-dates}

The first step in dating admixture events from genetic data is therefore estimating the length distribution of admixture segments.  There are two main approaches for this; a first approach is to use patterns of linkage along a chromosome to estimate the length distribution, without explicitly inferring the genomic location of these segments. In contrast, a second set of methods first aims to identify all admixture segments over a certain length, and then use these segments for inference. 
(\citep{chimusa_dating_2018}) (Figure \ref{fig:fig1} A).

The first approach uses the admixture-induced linkage disequilibrium (ALD) decay. Variants on introgressed archaic segments are expected to be in high linkage disequilibrium to each other at the time of admixture \citep{chakraborty_admixture_1988,stephens_mapping_1994,wall_detecting_2000}. The extent of linkage between introgressed variants decreases over generations. Hence, in case of a recent admixture event a few tens of generations ago, ALD stretches  over long genetic distances
\citep{patterson_methods_2004} and is therefore easily distinguishable from short range LD due to inheritance of chromosomal segments from the ancestral population as well as LD caused by bottlenecks and genetic drift after the split of the populations \citep{moorjani_history_2011,sankararaman_date_2012}. For ancient admixture events however, ALD is quite similar to the genomic background \citep{sankararaman_date_2012}. To circumvent this issue for dating the Neandertal-human admixture time, an ascertainment scheme was used to calculate LD only for markers that are nearly differentially fixed between the two taxa. In this case, the presence of apparent Neandertal alleles in close-range LD is a signature of a locally introgressed locus
\citep{sankararaman_date_2012}. Typically, estimation of admixture dates proceeds by fitting a decay curve of pairwise LD as a function of genetic distance, using an exponential distribution whose parameters are informative for the time of an admixture pulse \citep{moorjani_history_2011,loh_inferring_2013}. Using this approach Sankararaman et al. dated the Neandertal-human admixture pulse to be  between 37,000--86,000 ya (years ago) \citep{sankararaman_date_2012}. Later, this date was refined to 40,510--54,454 ya (95\% CI) using a different ascertainment scheme combined with a different genetic map \citep{moorjani_genetic_2016}. A date was also obtained from an ancient genome to be 50,000 - 60,000 ya by adding the time since the admixture obtained from the ancient individual, by the decay of pairwise covariance between
introgressed SNPs, to the specimens radiocarbon date \citep{fu_genome_2014}.

Using the second set of approaches, a more recent interbreeding event around 37 - 42 kya from an early modern human remain found in Romania with a recent Neandertal ancestor not more than 200 years ago, was found \citep{fu_early_2015}. The identification of segments is largely independent from the later dating, and can be done using a variety of methods \citep{racimo_signatures_2017,seguin_orlando_paleogenomics_2014,vernot_excavating_2016,sankararaman_combined_2016,skov_detecting_2018}. The length distribution of the obtained fragments is then used to estimate the time of the admixture pulse, typically using an exponential distribution.

The Denisovan modern human admixture time point was estimated to lie in the interval between 44,000--54,000 ya using the ALD
approach on modern day genomes \citep{sankararaman_combined_2016}. 
Direct inference of archaic segments in genomes from present day Southeast-Asians revealed a proportion of previously unknown Denisovan ancestry private to these populations. The segments from this ancestry are more diverged from the high-coverage Denisovan genome then previously found ones. This suggests an additional admixture event from a different population of Denisovans \citep{browning_analysis_2018}. 
Comparing the mean length of the formerly known and newly identified Denisovan segments did, however, not reveal significant differences, suggesting a lack of power to distinguish the
two events by time \citep{browning_analysis_2018,jacobs_multiple_2019}. Analysing genomes from Papuan individuals revealed two time separated admixture events with Denisovans, one in line with previous estimates at 45.7 kya (95\% CI 31.9-60.7 kya) and one exclusive to Papuans dated to
be around 29.8 kya (95\% CI 14.4-50.4 kya) \citep{jacobs_multiple_2019}.


\subsection{Why can't we us the pulse model}\label{Why can't we us the pulse model}

While convenient for inference, one cannot use the admixture pulse model to distinguish an extend period of admixture from an admixture pulse \citep{pickrell_toward_2014}. Here, we are chiefly interested in the duration of the interbreeding between Neandertals and modern humans, especially in the start and end point.  Under an admixture pulse model, which assumes that these two times coincide, hypotheses about the impact of modern human colonization of Eurasia on Neandertal extinction cannot be evaluated. Therefore, we introduce the \textit{extended admixture pulse} model, allowing for multigeneration continuous admixture. 

\subsection{Previous attempts of a general admixture model and ours}\label{Previous attempts of a general admixture model and ours}

The extended pulse model is a generalization of the simple admixture pulse model. The simplest generalization of this model is to assume two or more discrete pulses. In such a case,  each segment can be thought of having entered in one of the pulses, and the resulting admixture tracts will be a discrete mixture of the constituent distributions \citep{pickrell_ancient_2014}, weighted by the relative migration rates. Zhou et al. 2017 \citep{zhou_modeling_2017} showed that this model, in principle, can be used for continuous mixtures as well, using a polynomial function as a mixture density. However, they found that even for relatively short admixture events, the large number of parameters led to an underestimate of admixture duration \citep{zhou_inference_2017}, and the beginning and end of admixture were not well inferred
\citep{zhou_modeling_2017,zhou_inference_2017}. 

Here we use a simpler model of continuous admixture with just two parameters, and one less than the two-pulse model of Pickrell et al. 2014 \citep{pickrell_ancient_2014}. One parameter reflects the mean admixture time, and the other the duration of the admixture event; letting this parameter go to zero thus recovers the (nested) simple pulse model. 
This model is particularly simple if we assume that migration rate over time is Gamma distributed, in which case the distribution of admixture segment lengths has a closed form (Figure \ref{fig:fig1} B & C).

Nevertheless, this is a difficult problem, as it requires deconvoluting an exponential mixture, which is notoriously hard \citep{dasgupta_mixture_2008}. Thus, we have to carefully evaluate potential sources of biases known from previous studies, namely the demography of the admixed population, the accuracy of the recombination map and the ascertainment scheme \citep{sankararaman_date_2012,fu_genome_2014,moorjani_genetic_2016}.
We compare the extended pulse with the simple pulse model, to examine the implications of the one-generation assumption in relation to the aforementioned potential biases. We test when the simple pulse model can be contrasted for the more general model of an extended admixture pulse.



\subsection{What we want to do}\label{what-we-want-to-do}

In our study, we first define the new extended admixture pulse model describing the expectation of the resulting segment length distribution for Gamma distributed admixture being Lomax distributed, holding a parameter for the duration of admixture. This model works for
both methods to infer the segments length, either directly or by using
the ALD decay. Second, we examine the effect of ongoing admixture on the admixture time estimates on methods that assume a one generation pulse. Using the extended pulse model, we investigate under which scenarios the parameters of the Lomax-distribution can be accurately estimated and for which parameters we can distinguish a pulse-like admixture event from a continuous event. We show that in many cases of ancient admixture, pulses cannot be distinguished from more continuous admixture events. Using the 1000 Genomes data, ALD inferred admixture times from Europeans are consistent with a multitude of duration times.
We conclude that current methods are unsuitable to definitively estimate the duration of admixture and thus answer when the contact between Neandertals and modern humans started or ended.

\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig1-1.pdf}
\caption{\label{fig:fig1} A) Neandertal introgression into non-Africans with a multitude of potential admixture durations. B) The time and duration of admixture results in different length distributions of introgressed chromosomal segments (grey) containing  Neandertal variants (green circles)  in high LD to each other
compared to the background . The ALD approach estimates linkage
between the introgressed variants (green circles), wheres the haplotype approach tries
to estimate the segment directly (grey area). C) Migration rate per generation
modeled using a Gamma distribution for different admixture durations,
dotted line indicates maximum time of gene flow. D) The expected LD
decay under the extended pulse model.}
\end{figure}

\section{Methods}\label{methods}

We conducted simulations to assess the effect of continuous
admixture compared to a pulse under ideal circumstances. 
We also evaluate the impact of different recombination and demographic scenarios and the effect of different analysis parameters for estimates under an instantaneous pulse model.
After assessing these effects we evaluated the possible
parameter range for using the extended pulse model on ALD data,
enabling to obtain a duration of continuous admixture. We used both admixture models on real data. We fitted several admixture scenarios with different durations to ALD data from the 1000 Genomes Project \citep{the_1000_genomes_project_consortium_global_2015} with 3 high coverage Neandertals \citep{prufer_complete_2013,prufer_high-coverage_2017,mafessoni_high_coverage_2020}.

\subsection{Simulations}\label{simulations}

We used the msprime coalescent simulator
\citep{kelleher_efficient_2016} for simulations with sample sizes
chosen to reflect presently available data. For ALD simulations we simulate 176 diploid
African individuals and 170 diploid non-Africans, corresponding to the
number of Yoruba (YRI) and Central Europeans from Utah (CEU)
sequences in the 1000 Genomes project data \citep{the_1000_genomes_project_consortium_global_2015}. Since three
high coverage Neandertal sequences are available \citep{prufer_complete_2013,prufer_high-coverage_2017,mafessoni_high_coverage_2020} we choose to
simulate three diploid genomes. For each individual we simulated 20
chromosomes with a length of 150 Mb each. The mutation rate was set for
all simulations to \(2*10^{-8}\) per base per generation. The
recombination rate was set to \(1*10^{-8}\) per base pair per generation
unless specified otherwise. The demographic parameters are based on
previous studies dating Neandertal admixture
\citep{sankararaman_date_2012,fu_genome_2014,moorjani_genetic_2016}. In
the ``simple'' model (S \ref{fig:figS1} A), the effective
population size is assumed constant at \textit{Ne}=10000 for all populations, the
split time between modern humans and Neandertals is 10000 generations
ago and the split between Africans and non-Africans is 2550
generations ago. The migration rate from Neandertals into non-Africans
was set to zero before the split from Africans, to ensure no Neandertal
ancestry in Africans. Each simulation was repeated 100 times. 

\subsubsection{Admixture Models}\label{admixture models}



\subsubsection{Recombination map}\label{recombination map}

Uncertainties in the recombination map were previously shown to bias
admixture time estimates \citep{sankararaman_date_2012,sankararaman_combined_2016,fu_genome_2014}. To investigate the effect of more realistic
recombination rate variation, we simulated samples using a recombination map. For ALD simulations we
either used the African-American-Map \citep{hinch_landscape_2011} or
the HapMap phase 3 \citep{HapMapConsortium_second_2007} for simulations
under a variable recombination rate, for simplicity, we used the same
recombination map (150 Mb of chromosome 1, excluding the first 10 Mb)
for all simulated chromosomes. The mean recombination rate was
calculated from the 150 Mb map (\(1.017 \, \frac{cM}{Mb}\) AAMap and
\(0.992 \, \frac{cM}{Mb}\) HapMap). To examine the effect of uncertainties in the
genetic map we either: used the mean recombination rate from the
respective map to calculate the genetic distance from the physical
distance for each SNP (assuming recombination to be constant), used the other map to assign distances
(e.g.~AAMap used for the msprime simulation and HapMap used to assign
genetic distances, emulating uncertainties in the recombination map) or used the same map for simulation and assigning
genetic distances (population specific map). 

\subsubsection{Complex demography}\label{inferred demography}

Demography such as population size changes are known to influence LD
patterns \citep{gravel_population_2012,liang_lengths_2014}. To test the impact of
demographic history on admixture time estimates, we contsrast our standart model with
realistic and complex demographic histories. This includes substructure in the ancestral human population and additional gene flow between Africans and non-Africans after the Neandertal admixture (S \ref{fig:figS1} B). The 
effective population
sizes are based on estimates from Neandertal and present-day human
genomes. MSMC estimates from YRI as representatives for Africans and CEU
for non-Africans from Schiffles \& Durbin 2014
\citep{schiffels_inferring_2014} were used together with PSMC
\citep{li_inference_2011} inferred demographic model for Neandertals
based on the Vindija33.19 high-coverage genome
\citep{mafessoni_high_coverage_2020}. In order to use the effective
population size estimates at a given time point for modern humans from
Schiffels \& Durbin 2014 for our simulations, we
first transformed the time points given in years back to generations by
using 30 years for one generation, as assumed in Schiffels \& Durbin.
Second, since the original estimates are based on a different mutation
rates (\(1.25*10^{-8} \frac{bp}{gen}\)), we corrected all estimates for
the mutation rate used in the simulations
(\(2*10^{-8} \frac{bp}{gen}\)). The split times between Neandertals and
modern humans was kept the
same as in the simple simulations (10000 generations ago) (Sup. Fig. \ref{fig:figS1} A). In the complex model, we simulated an earlier initial split between populations in Africa 3550 generations ago. With substructure starting from 3200 till the final split at 2550 generations ago with a per generation migration rate between the two subpopulations of 0.001. The population size of the ancestor of Neandertals and
humans before the split was set to 18296 (taken from the MSMC results). To simulate branch shortening
caused by the extinction of Neandertals, Neandertals were sampled 750
generations before the Africans and non-Africans. We simulated an additional admixture event between African and non-African populations with a total migration rate of 0.01 starting from 200 till 50 generations ago.

\subsection{Admixture time estimates}\label{admixture time estimates}

\subsubsection{Ascertainment scheme}\label{asceteinment scheme}

For the ALD method, SNPs need to be ascertained to enrich for
Neandertal informative sites in the test population to remove noise and
amplify the ALD signal \citep{sankararaman_date_2012}. 
We evaluate the impact of the ascertainment scheme by contrasting two distinct schemes \citep{sankararaman_date_2012,fu_genome_2014}. The lower-enrichment
ascertainment scheme (LES) filters for SNPs fixed for the ancestral state in
Africans and polymorphic in Neandertals. The higher-enrichment
ascertainment scheme (HES) restricts the analysis on SNPs fixed for the
ancestral state in Africans, polymorphic in Neandertals and polymorphic
in non-Africans.

\subsubsection{ALD calculation and curve fitting}\label{ALD calculation and curve fitting}

The pairwise weighted LD between the ascertained SNPs a certain genetic
distance \(d\) apart was calculated using ALDER
\citep{loh_inferring_2013}. A minimal genetic distance \(d_0\) between
SNPs was set either to 0.02 cM and 0.05 cM. This minimal distance cutoff
removes extremely short-range LD, which might be due to incomplete lineage sorting (ILS). To obtain the mean time estimates
the data is fitted with an Exponential distribution (one generation pulse model) shown in equation
\ref{eq:5}, using a non-linear least-square optimization algorithm
implemented in R \citep{R_Core_Team_2019}. Where \emph{A} is the
intercept, $t_m$ the mean time since the admixture event in generations,
\emph{d} the genetic distance in cM and \emph{c} is a constant modeling
background LD. The model was fitted following Moorjani et al 2016
\citep{moorjani_genetic_2016}. The duration of continuous admixture is
modeled using the Lomax fit shown in Eq. \ref{eq:6}. We used the
notation from Kozubowski et al. 2008 \citep{Kozubowski_Testing_2008}. Setting $\frac{1}{\theta} = \frac{t_m}{k}$.
The starting value of $t_m$  is taken from the Exponential fit.


\begin{equation}
\label{eq:5}
ALD \sim\ A\,e^{-t_m \:d}+c
\end{equation}

\begin{equation}
\label{eq:6}
ALD \sim\ A\,\left( \frac{1}{1 + \frac{1}{k}t_m \:d}\right) ^k+c
\end{equation}


\subsection{Modeling parameter effect sizes}\label{modeling prameter effect sizes}

To model and compare parameter effect sizes we simulated 100
replicates for each combination of
parameters: ascertainment scheme ($A_i$ = LES/HES), minimal genetic distance
($M_i$ = \(d_{0}=0.02\,cM\)/\(d_{0}=0.05\,cM\)), demography ($D_i$ = simple/complex),
recombination rate ($R_i$ = constant/variable), gene flow model
($G_i$ = pulse/continuous). Results of simulations where the nls-optimization to
fit the ALD decay curve did not converge were removed (10 out of 3200).
We used a Generalized Linear Model (GLM)  to estimate the effect
size of the different parameters (Eq.
\ref{eq:8}). The difference between the estimated
admixture time and the true admixture time, the error in the estimate,
was standardized using the true time as the mean. The standardized error in this model is the response to the parameters as model
predictors. We choose the Normal distribution as the likelihood function being the maximum entropy distribution in our case. The mean of the Normal distribution is then described by a fixed effect size model of the predictors (factors) coded as indicator variables. We obtained the posterior probability using a Hamiltonian Monte Carlo MCMC algorithm, as implemented in STAN \citep{carpenter_stan_2017} using an R interface \citep{stan_development_team_rstan_2018,mcelreath_statistical_2020}. The Markov chains converged to the target distribution (Rhat = 1) and efficiently sampled from the posterior (S Table \ref{tab:tableS1}).  

\begin{equation}\label{eq:8}
\begin{split}
E_i &\sim Normal(\mu_i,\sigma) \\
\mu_i &= \alpha + \beta_aA_i + \beta_mM_i + \beta_dD_i + \beta_rR_i + \beta_gG_i \\
\alpha &\sim Normal(0,2) \\
\beta_a,\beta_m,\beta_d,\beta_r,\beta_g &\sim Normal(0,2) \\
\sigma &\sim Exponential(1)
\end{split}
\end{equation}

\subsection{Estimating admixture time from real data}\label{Estimating admixture time from real data}

We apply the extended pulse model on real data to see if we can reject any admixture durations (including a one generation pulse), by comparing the model fit to the data for different duration parameters.
We used the 1000 Genomes phase 3 data together with the Altai, Vindija and Chagyrskaya high coverage Neandertals.  We filtered for all unrelated individuals from the YRI as representatives of unadmixed Africans and all CEU as admixed Europeans. We only considered biallelic sites. The sites were polarized relative to the Chimpanzee base (panTro4). We used the CEU specific fine-scale recombination map \citep{spence_inference_2019} to convert the physical distance between sites into genetic distance. 

\section{Results}\label{results}

\subsection{Introduction to result}\label{introduction to result}

We used coalescent simulations of Neandertal admixture into non-African humans to investigate multi-generation continuous admixture, using a Gamma distribution describing the per generation migration rate. Focusing on the ALD approach, the ALDER program was used to infer the decay of LD between ascertained sites informative for Neandertal introgression in non-Africans. The ascertained ALD was fitted with an exponential distribution assuming a one-generation admixture pulse.  We simulated this scenario using different conditions of recombination and demography as well as deploying different ascertainment schemes and LD cutoffs. A Generalized Linear Model (GLM) was used to compare all these factors with the effect of multi-generation continuous admixture on the pulse assumption. To get the duration of the admixture we introduce a new fitting approach deploying the Lomax distribution. We test this new fitting with the most inferential factors from the previous analysis on simulated and real data using the ALD.


\subsection{The effect of continuous admixture on the admixture time estimates}\label{the effect of continuous admixture on the admixture time estimates}


To assess the effect of estimating admixture time assuming a one-generation pulse when the real gene flow is actually continuous over multiple generations,
we compared the mean average deviation from the true mean time of admixture for simulations under a one-generation pulse with the one of multi-generation continuous gene flow with the same mean time.
The total amount of gene flow \(m\) from Neandertals into non-Africans
in the two models is equal. 
The shape and scale parameter of the Gamma distribution modeling the migration rate are chosen such
that the resulting weighted LD decay curves as functions of genetic
distance share the same mean for a pulse (Eq. \ref{eq:1}) and a
continuous admixture (Eq. \ref{eq:3}). We used the LES ascertainment scheme and a minimal distance between SNPs of 0.05 cM. 

First, we evaluate the effect of the admixture time on the accuracy of both a single pulse and an extended pulse model. The mean admixture times ranges
from 250 generations ago to 2000 generations ago.  The duration
of interbreeding under the extended pulse model is 50\% of the mean time (if the
mean time is 250 generations ago gene flow ranges from 325 to 175
generations ago). Assuming only a one generation pulse to estimate the
mean time, revealed only minor deviations between the two scenarios and
the true admixture time of 3\% to 6\% for the pulse and 0\% to 5\% for
the continuous (Figure \ref{fig:fig2} A).



To
further investigate the effect of pulse and continuous gene flow on the
admixture time estimates, we compared different durations of continuous
admixture with a fixed mean time of admixture of 1500
generations ago, displayed in Figure \ref{fig:fig2} B. Estimates between
pulse and continuous admixture start to deviate for 800 generations of
continuous admixture, with increasingly lower estimates for simulations
under continuous admixture compared to simulations under a pulse scenario per increase of admixture duration.

This bias is likely caused
by the differences in LD between sites entered in the tails of the gamma
distribution. LD between sites arising from early admixture events,
simulated by the right tail of the gamma distribution, is not detected
anymore, while LD between sites from late admixture is still present,
biasing the estimate towards younger dates. 

However, deviations in
estimates between the two scenarios of \textasciitilde{}120 generations
in the most extreme case are moderate compared to the mean admixture
time of 1500 generations.

In summary, the differences in mean time estimates between a pulse and a continuous admixture assuming a one-generation pulse are moderate, even for very long continuous admixture scenarios. This makes point estimates for mean admixture time compatible with long continuous admixture.


\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig2-1.pdf}
\caption{\label{fig:fig2} A) Comparison of mean admixture time estimates
between pulse and continuous gene flow for different admixture times.
The length of continuous gene flow corresponds to 50\% of the mean
admixture time, black line indicates true mean admixture time. B)
Comparison of mean admixture time estimates for simulations with a mean
time of admixture of 1500 generations ago, at a varying length of gene
flow. Boxplot created from 100 simulation replicates, respectively.}
\end{figure}
\subsection{Comparing effect sizes}\label{comparing effect sizes}

Having shown that the duration of continuous admixture under ideal
circumstances only marginally influences admixture time
estimates, we want to compare its effect in more realistic simulation scenarios. Two common assumptions, namely knowing the recombination pattern and knowing the demographic history, were implicitly used in the previous simulations.
Previous studies reported that violations of these assumptions can have an effect on the admixture time estimates.
Hence, we used the African-American genetic map and a more complex demography with substructure in the ancestral human population and additional gene flow between Africans and non-Africans after the Neadertal admixture ended (Supp. Fig. \ref{fig:figS1} B), for simulations under more realistic parameters. The genetic distance was assigned using the average recombination rate of the African-American genetic map (assuming a constant recombination rate).
Furthermore, we altered the used scheme for ascertainment of SNPs for admixture informative sites. We compared the LES where only
positions in the admixed population are considered where both source
populations are diverged from each other, to an even stricter scheme. The Higher-Enrichment Scheme (HES) requiring additional
to the LES non-Africans to be polymorphic at a SNP. Additionally, we altered the minimal distance between pairs of SNPs for which the LD is calculated from 0.05 cM to 0.02 cM.



We simulated every combination of these parameter sets resulting in 32
different sets with 100 replications each (Supp. Fig. \ref{fig:figS2}).
A GLM was applied to estimate effect sizes of the four predictors being
ascertainment scheme, minimal distance, demography and recombination on
the bias of admixture estimates (Supplement Table \ref{tab:tableS1}).
Figure \ref{fig:fig3} shows the comparison of the standardized
difference between true and estimated time between the previously used
model (ascertainment = LES, \(d_{0} = 0.05 cM\), demography = simple and
recombination = constant) further referred to as the standard model and
a model with one of the four parameter changed, respectively. The
previously observed overestimation of the standard model was estimated
to be 0.12 (0.08 - 0.16 95 \% CI) standard deviation from the true
admixture time.

Every parameter change, except using \(d_{0} = 0.02 cM\), results in lower estimates compared to the
standard model, with the biggest difference between a constant and a
varying recombination (-1.76 - -1.7 95 \% CI). The
underestimation caused by the continuous gene flow model (-0.34 -
-0.28 95 \% CI) is in the range of biases arising from the ascertainment scheme (-0.35 -
-0.29 95 \% CI) and only slightly more then caused by a more complex demography (-0.25 -
-0.19 95 \% CI). Most accurate estimates are achieved using the LES
ascertainment scheme in combination with a minimal distance of 0.05 cM.
Additionally, the bias introduced by the demography seems to depend on
the used ascertainment scheme, since only under the HES a difference
between the simple and complex demography is evident (Supp. Fig.
\ref{fig:figS2}).

Overall the bias introduced by the ascertainment, minimal distance
cutoff, demography and admixture model are  around +/-0.25 standard deviations or less. The major uncertainties in the admixture time
estimate arise from assuming a constant recombination rate. The
admixture time estimates for a pulse or a multi-generation admixture are similarly effected by the other modeling parameters.

\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig3-1.pdf}
\caption{\label{fig:fig3} Effect size estimates for each parameter on the standardized difference between true and estimated admixture time. Dotted horizontal line indicates unbiased admixture estimates.}
\end{figure}

\subsection{Estimating the Lomax-parameters under different conditions}\label{estimating the Lomax-parameters under different conditions}

Building on the previous
result we wanted to find out the conditions to retrieve the Lomax
parameters i.e.~the duration of the admixture. We simulated an admixture
scenario under a simple demography with varying durations of continuous
admixture and sampled the population at different time points since the
end of the gene flow. Doing so allows us to examine how close one has to
sample form the end of a continuous admixture to still accurately
estimate its duration. Since the recombination map was shown to be highly
influential, we tested the duration estimates under a constant and variable
recombination using the HapMap genetic map. Here, we correct the
genetic distance by: assuming a constant rate, using the AAMap (uncertainty in the recombination map) or the
HapMap itself (having a  population specific recombination map). We used the LES ascertainment scheme in combination with
a minimal distance of 0.05 cM. To fit the Lomax we can take advantage of
the fact that the bias of the mean time estimates using the simplified
one generation pulse model is relatively accurate. We can thus estimate
the mean time first by fitting an exponential and in a second step estimating k
using a Lomax distribution with a starting parameter for the mean time received
from the Exponential fit, to archive better convergent. Figure \ref{fig:fig4} A and C show the mean
time estimates received from the Lomax fit. For simulations under a
constant recombination the mean time can be estimated confidently for different durations and different sampling times after the end of the admixture event. Estimates for simulation under a recombination map only
assuming a constant rate when calculating the LD results in high uncertainty and overestimation. This is different for estimates under an Exponential distribution (Supp. Fig.  \ref{fig:figS2}). Using the AAMap yields
results closer to the true value and less biased, however only
using the exact same genetic map gives unbiased results. Figure
\ref{fig:fig4} B and D shows the corresponding admixture duration estimates by the
Lomax fit. Accurate estimates can be obtained throughout the different
admixture durations under a constant recombination rate, when sampled
recently from the end of the continuous admixture. Sampling further away from the end of the admixture results in less accurate estimates.  All simulations using a recombination map show a much higher
variance in the estimates. Especially for admixture durations longer then
800 generations or sampled later then 50 generations away from the end
of the admixture, the estimates are not reliably. If no precise genetic map
is used to inferred the genetic distance between SNPs, no accurate
duration estimates can be obtained. Even for estimates using the population specific map, the variation is considerable for admixture durations longer then 800 generations or sampled later then 200 generations after the end of the gene flow.

Under a realistic scenario with a varying recombination
rate, the duration can only be accurately estimated with a highly
precise map for recent continuous admixture events not longer then 400
generations. With regard to scenarios of Neandertal admixture,
accurately estimating the duration of possible continuous admixture from
present day human genomes even under a constant rate seems
under powered. 


\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig4-1.pdf}
\caption{\label{fig:fig4} Lomax parameter estimate for different sampling and
admixture duration times using different methods for assigning genetic
length A) Mean time estimates for different gene
flow length all sampled 50 generations after the gene flow ended. B)
Duration estimate of the scenario C) Mean time estimates for different sampling times after the end of 800
generations of gene flow. D) Duration estimate of the scenario}
\end{figure}


We can also demonstrate this using real data.
Figure \ref{fig:fig5} (Sup. table \ref{tab:tableS2}) shows multiple gene flow duration scenarios ranging from a one generation pulse to 2500 generations of continuous admixture. All scenarios are compatible with the ALD data from 1000 Genome's CEU computed using YRI and the Altai, Vindija and Chagyrskaya high coverage as reference populations and a CEU specific fine-scale genetic map \citep{spence_inference_2019}.



\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig5-1.pdf}
\caption{\label{fig:fig5} Different admixture duration models ranging
from a one generation pulse to 2500 generations of gene flow for Neandertal interbreeding using all 1k Genome CEU individuals as the admixed population
with all YRI and 3 high coverage Neandertals as reference populations. A) Weighted LD normal scaled B) Weighted LD log scaled.}
\end{figure}

\section{Discussion}\label{discussion}

How, where and when Neandertals and early modern humans interacted remains contentious.  
Archaeological evidence bounds the timing of interactions to times when their range overlapped; from the first out-of-Africa migrations of modern humans around 188 k years ago \citep{stringer_when_2018,hershkovitz_earliest_2018} to the extinction of Neandertals around 37 - 39 kya \citep{higham_timing_2014,zilhao_precise_2017}, leaving a time-span of roughly 140ky in which interbreeding between modern humans outside of Africa and Neandertals might be feasible.

Genetic data has revealed that gene flow between Neandertals and modern humans did occur \citep{green_draft_2010}; and it's mean age is estimated to 47,000–65,000 ya (\citep{sankararaman_date_2012}), under a single pulse model, but later interbreeding has been documented \citep{fu_early_2015}.

Here, we contrasted this pulse model with scenarios involving more long-standing gene flow.  Our model, assumes that migration rates  follow a Gamma distribution, which results in a simple closed-form expression for the introgressed segment lengths or ALD  distributions. 

In our analyses using the extended model, we find that mean admixture time estimates vary little between models. However, we also show that even gene flow lasting thousands of generations may yield data that are practically indistinguishable from a pulse, and that other modelling and method assumptions have an impact on estimates that are of a similar magnitude or much higher. Particularly assumptions on the recombination rate are highly impactfull, and will lead to sever underestimations of admixture times. Therefore reliable mean time estimates can only be obtained using population specific recombination maps.

The major implication of our result is that interpreting results of genetic dating is difficult.  Estimating the mean time of gene flow and its confidence intervals are not indicative of the actual duration of the interbreeding.  This is of great practical importance as it might be tempting to link the genetic admixture date estimates with biogeographical events \citep{sankararaman_date_2012,lazaridis_genomic_2016,jacobs_multiple_2019,vyas_analyses_2019,douka_age_2019}. For example, the lower bounds on admixture time estimates might be used as dates of Neandertal extinction, and likewise use the earliest dates of gene flow as evidence when the out-of-Africa migration happened \citep{sankararaman_date_2012}. Our results show that both of these interpretations might be misleading. However, using the extended pulse model relaxing the one generation pulse assumptions used for earlier estimates, also indicates that most of the gene flow was around 45,000 to 65,000 years ago, consistent with previous estimates.


While our results show that drawing concrete conclusions from admixture time estimates can be misleading, other avenues for differentiating different gene flow events remain fruitful.
In particular, measures based on population differentiation (e.g \citep{browning_analysis_2018,wall_higher_2013,villanea_multiple_2019}) are very promising to understand the different events that contributed to archaic ancestry in present-day humans. 

The other major avenue are ancient DNA samples from early modern humans that lived around or immediately after gene flow between archaic and modern humans occurred. The \textit{Oase 1} \citep{fu_genome_2014} genome is a prime example, as it directly demonstrates that some gene flow must have occurred in Europe, around 40,000 years ago, concurrently with when \textit{Oase 1} lived. More generally, detailed analyses of ancient genomes from the initial upper paleolithic should provide more accurate mean dates, as they are much closer to the gene flow, and may give better evidence about the duration of the gene flow. The drawback is that some of the gene flow these individuals experience may be private to their population, as has e.g. been suggested for \textit{Oase 1}. Larger numbers of early modern human genomes, as well as methods to accurately infer admixture tracts or ALD in low-coverage data will likely enable this line of research. 


\hypertarget{refs}{}

\bibliography{References/MyLibraryATE}


\pagebreak
\setcounter{figure}{0} \renewcommand{\figurename}{Fig. S}
\renewcommand{\tablename}{Tab. S}

\section{Supplement}\label{supplement}

\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/figS1-1.pdf}
\caption{\label{fig:figS1} Demographic models of Neandertal admixture with non-Africans used for the simulations. A) Simple demographic model used for ALD simulations with constant population sizes. B) Complex demographic model with substructure in Africa, where after an initial earlier split and isolation the structured population exchange migrants till the final split and additional gene flow between Africans and non-Africans after the Neandertal admixture. The  population sizes after the (final) split are taken fome MSMC/PSMC estimates for the respective populations. C) Demographic model for the direct inference of Neandertal segments in non-Africans taken and modified from Skov et al. 2018. (\citep{skov_detecting_2018})}
\end{figure}

\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/figS2-1.pdf}
\caption{\label{fig:figS2} Comparison of the standardized difference between true and estimated admixture time for all combinations of parameters: ascertainment scheme = LES/HES,  $d_{0}$ = 0.02/0.05 cM, demography = simple/complex, recombination = constant/variable and the gene flow model = pulse/continuous, with 100 replicates respectively. Dotted horizontal line indicates no difference between true and estimated time.}
\end{figure}

\begin{table}[H]

\caption{\label{tab:tableS1}\label{tab:tableS1} Mean, standart deviation, 5.5/94.5 compatibility interval of the posterior distribution for every parameter effect on the standardized difference between true and estimated admixture time.}
\centering
\begin{tabular}{l|r|r|r|r|r|r}
\hline
  & mean & sd & 5.5\% & 94.5\% & n\_eff & Rhat\\
\hline
a & 0.12 & 0.02 & 0.08 & 0.16 & 1220.94 & 1\\
\cline{1-7}
bG & -0.31 & 0.02 & -0.34 & -0.28 & 1559.84 & 1\\
\cline{1-7}
bR & -1.73 & 0.02 & -1.76 & -1.70 & 2112.69 & 1\\
\cline{1-7}
bD & -0.22 & 0.02 & -0.25 & -0.19 & 2036.66 & 1\\
\cline{1-7}
bm & 0.13 & 0.02 & 0.10 & 0.17 & 1657.66 & 1\\
\cline{1-7}
bA & -0.32 & 0.02 & -0.35 & -0.29 & 1721.62 & 1\\
\cline{1-7}
sigma & 0.46 & 0.01 & 0.44 & 0.47 & 2239.47 & 1\\
\hline
\end{tabular}
\end{table}

\begin{figure}
\centering
\includegraphics{Admixture_Time_Inference_Paper_Draft_files/figure-latex/figS3-1.pdf}
\caption{\label{fig:figS2} Mean time estimates of the Exponential model A) Mean time estimates for different 
admixture durations all sampled 50 generations after the admixture ended. B) Mean time estimates for different sampling times after the end of 800
generations of continuous admixture.}
\end{figure}

\begin{table}[H]
\caption{\label{tab:tableS2}\label{tab:tableS2} Mean and 2.5/97.5 compatibility interval for every parameter estimated in the model fit}
\centering
\begin{tabular}{l|l|l|l|l}
\hline
Model & Parameter & Estimate & 2.5 \% & 97.5 \%\\
\hline
Exponential & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Exponential & mean time & 1540.51741259424 & 1486.87051449324 & 1598.18041766271\\
\hline
Exponential & c & 1.04922402241663e-05 & 6.11947485562607e-06 & 1.48650055927066e-05\\
\hline
Lomax (td=1) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=1) & mean time & 1540.54690341371 & 1486.89897832594 & 1598.21101235108\\
\hline
Lomax (td=1) & c & 1.04949518630901e-05 & 6.12218649454983e-06 & 1.48677172316303e-05\\
\hline
Lomax (td=100) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=100) & mean time & 1541.21552745233 & 1487.54431823588 & 1598.90466361173\\
\hline
Lomax (td=100) & c & 1.04778971116211e-05 & 6.10513174308087e-06 & 1.48506624801614e-05\\
\hline
Lomax (td=200) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=200) & mean time & 1543.30241011905 & 1489.55852740936 & 1601.0696602451\\
\hline
Lomax (td=200) & c & 1.04341296352963e-05 & 6.06136426675609e-06 & 1.48068950038366e-05\\
\hline
Lomax (td=400) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=400) & mean time & 1551.70459843938 & 1497.66811836146 & 1609.78635031902\\
\hline
Lomax (td=400) & c & 1.02583870973029e-05 & 5.88562172876263e-06 & 1.46311524658431e-05\\
\hline
Lomax (td=800) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=800) & mean time & 1586.19945700954 & 1530.96172977427 & 1645.57238365183\\
\hline
Lomax (td=800) & c & 9.54350178205991e-06 & 5.17073641351966e-06 & 1.39162671506002e-05\\
\hline
Lomax (td=1000) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=1000) & mean time & 1613.04056713235 & 1556.86812647683 & 1673.41818158693\\
\hline
Lomax (td=1000) & c & 8.99447037087496e-06 & 4.62170500233471e-06 & 1.33672357394152e-05\\
\hline
Lomax (td=1500) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=1500) & mean time & 1713.23411309364 & 1653.57253761458 & 1777.36206551997\\
\hline
Lomax (td=1500) & c & 6.99221511836012e-06 & 2.61944974981987e-06 & 1.13649804869004e-05\\
\hline
Lomax (td=2000) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=2000) & mean time & 1874.70574615156 & 1809.42109093705 & 1944.87773256239\\
\hline
Lomax (td=2000) & c & 3.90606333903974e-06 & -4.66702029500504e-07 & 8.27882870757999e-06\\
\hline
Lomax (td=2500) & A & 0.0177951442439937 & 0.0177907714786252 & 0.0177995170093623\\
\hline
Lomax (td=2500) & mean time & 2128.77896518946 & 2054.64647743473 & 2208.46114940615\\
\hline
Lomax (td=2500) & c & -6.57497391994452e-07 & -5.0302627605347e-06 & 3.7152679765458e-06\\
\hline
\end{tabular}
\end{table}



\end{document}