



\title{An extended admixture pulse model reveals the limits to the dating of Human-Neandertal introgression}

\author[1,2]{Iasi, Leonardo N M}
\author[1,3]{Peter, Benjamin M}


\affil[1]{Department of Evloutionary Genetics, \\ 
        Max Planck Institute for Evolutionary Anthropology, Leipzig, Germany}
\affil[2]{leonardo\_iasi@eva.mpg.de}
\affil[3]{benjamin\_peter@eva.mpg.de}
\begin{document}

Over short distances, ALD may be confounded by short-range LD caused by the inheritance of chromosomal segments from an ancestral population, or LD caused by bottlenecks and genetic drift after the split from the parental population \citep{moorjani_history_2011}.

For ancient admixture events however, ALD is quite similar to the genomic background. To partially circumvent this issue for dating the Neandertal-human admixture time, variants are ascertained such that only  markers that are (nearly) differentially fixed between the two groups are used 
\citep{sankararaman_date_2012}. 

This is largely independent of the later dating, which uses the length distribution of the inferred admixture fragments \cite{hellenthal_genetic_2014}.

 The simplest models assume that admixture segments are rare and inbreeding is not significant, such that admixture segments are unlikely to recombine with each other \citep{pool_inference_2009,liang_lengths_2014}. Further assumptions are that there is no selection \citep{shchur_distribution_2019} and the recombination rate is the same in different populations at all times \citep{gravel_population_2012}. In addition, 


Using D-statistics \citep{green_draft_2010} and by directly inferring introgressed segments, it was found that the amount of Neandertal ancestry is 24\% higher in present-day East-Asians compared to Europeans \citep{meyer_high-coverage_2012,  wall_higher_2013}. A second admixture event private to East-Asians around the same time as the interbreeding event between Neandertals and all non-Africans is suggested to explain the higher amount of ancestry \citep{kim_selection_2015,vernot_complex_2015}.

In comparison, the history of interbreeding, Denisovans and modern humans appears more complex. Among present-day populations, Papuans and Melanesians have the highest amount of Denisovan ancestry (6\%), much more than East-Asians (0.2\%) \citep{reich_genetic_2010,meyer_high-coverage_2012}.
However, the Denisovan ancestry segments in East Asians are made up of two distinct groups, one from a population only distantly related to the sequenced Denisovan genome, and another one that is more closely related \citep{browning_analysis_2018}. In contrast, Papuans and Melanesians only have segments from the first of this group, despite them having much more overall ancestry.
One explanation is that there has been a common admixture event into an ancestor of both populations, and a second event that only contributed to East Asians; however the admixture segment lengths of these two sets of populations are not significantly different from each other, suggesting either a lack of power to distinguish the two events by time, or that they occurred roughly at the same time \citep{browning_analysis_2018}. A third  Denisovan admixture event private to Papuans was suggested by \cite{jacobs_multiple_2019}.


Most Neandertal admixture segments have a similar divergence to the sequenced Neandertals \citep{browning_analysis_2018}. However, there were also multiple Neandertal gene flows. \textit{Oase 1}, an early modern human from Romania had a recent Neandertal ancestor less than 200 years before he lived (~37â€“42 kya), later than the age of the \textit{Ust'Ishim} modern human who already carries Neadertals ancestry. However, the population \textit{Oase 1}  belonged to did  not contribute substantially to present-day human populations  \citep{fu_genome_2014,fu_early_2015}.

Recently, \cite{hajdinjak_early_2021} showed that segments from a recent Neandertal ancestor found in  46 - 43 ka (cal. BP) old modern humans from Bulgaria have a higher allele sharing with the Neandertal segments of present-day East-Asians then those of present day Europeans, indicating that this local gene flow did contribute to modern human populations \citep{hajdinjak_early_2021}.  



\cite{ralph_geography_2013} used the distribution of shared identical by descent (IBD)  segments between pairs of individuals to infer the number and age of genetic common ancestors. By modeling over all ancestors and ages they fit more complex migration patterns. However, they found that a large set of migration patterns fit the IBD distribution equally well. To cope with this they introduced a regularization scheme to their likelihood function by adding a penalization term \citep{ralph_geography_2013}.

\subsection{Evaluation of the simulations}

\subsection{The effect of continuous admixture on the admixture time estimates}\label{the effect of continuous admixture on the admixture time estimates}

The total amount of gene flow $m$ from Neandertals into non-Africans
in the two models is equal, and we match the two models such that the mean admixture times are equal, with the only difference being the duration of gene flow $t_d$. All comparisons are based on the LES ascertainment scheme and $d_0 =  0.05 cM$. 



\subsubsection{Recombination rate}
 
The inference of admixture time depends on accurately estimating the recombination clock. Since there are not always fine-scale population specific recombination maps available, we examine three ways of assigning the genetic distance with various degrees of uncertainty. When simulating data under a recombination map, we either: use the mean recombination rate from the
respective map to calculate the genetic distance from the physical
distance for each SNP (assuming recombination to be constant with the average recombination rate of the respective map), use another genetic map to assign distances
(e.g.~AAMap  for the \texttt{msprime} simulation and HapMap  to assign genetic distances, emulating uncertainties in the recombination map) or use the same map for simulation and assigning
genetic distances (population specific map).

\subsubsection{Parameter estimation under the extended pulse model}
Inference is performed using the standard set of parameters defined above. In addition, we also  evaluate the effect of recombination rate variation. For this purpose, we simulate under both a constant recombination rate and using the empirical HapMap genetic map. Inference is performed using either  a constant recombination rate, using the HapMap (i.e. exact same map as in the simulations) or the AAMap recombination maps (to mimic uncertainty and population differences in estimated recombination maps).
\end{document}