\section{Discussion}\label{discussion}

How, where and when Neandertals and early modern humans interacted remains contentious.  
Archaeological evidence bounds the timing of interactions to times when their range overlapped; from the first out-of-Africa migrations of modern humans around ~188 years ago (\cite{stringer_when_2018,hershkovitz_earliest_2018}) to the extinction of Neandertals ~37 - 39 kya \cite{higham_timing_2014,zilhao_precise_2017}), leaving a time-span of roughly 140ky.

Genetic data has revealed that gene flow between Neandertals and modern humans did occur \cite{green_draft_2010}; and it's mean age is estimated to 47,000â€“65,000 ya \citep{sankararaman_date_2012}, assuming the interaction occurred at an instantaneous pulse.

Here, we contrasted this pulse model with scenarios involving more long-standing gene flow.  Our model, assumes that migration rates  follow a Gamma distribution, which results in an simple closed-form expressions for the introgressed segment lengths or ALD  distributions. 

In our analyses using the extended model, we find that most methods focus on the mean time of gene flow, and are mostly accurate in its inference. However, we also show that even gene flow lasting thousands of generations may yield data that are practically indistinguishable from a pulse, and that other modelling and method assumptions have an impact on estimates that are of a similar magnitude or much higher. Particularly the assumption of a constant recombination rate will lead to sever underestimations of admixture times, therefore reliable mean time estimates can only be obtained using population specific recombination maps.

The major implication of our result is that one has to be extremely cautious when interpreting results of genetic dating; the data are entirely consistent with gene flow occurring both earlier and later than the confidence intervals of the mean gene flow timing might indicate. This is of great practical importance as it might be tempting to link the genetic admixture date estimates with biogeographical events. For example, the lower bounds on admixture time estimates might be used as dates of Neandertal extinction, and likewise use the earliest dates of gene flow as evidence when the out-of-Africa migration happened \citep{sankararaman_date_2012}. Our results show that both of these analyses might be misleading.




Using the Gamma admixture model, we established that even for very long continuous admixture scenarios, estimates for the mean time are compatible with a pulse like admixture. Testing the influence of different parameter scenarios on the admixture mean time estimates, revealed that the error in the estimates is out weighted by all other tested model parameters. Assuming a constant recombination rate will lead to sever underestimations of admixture times, whereas the other parameters tested (demography and analysis parameters) are far less influential. Although this bias is known from previous studies, we notice it is underappreciated that reliable mean time estimates can only be obtained using population specific recombination maps, especially for archaic admixture events. Even though reliable estimates can be obtained using a population specific recombination map, they in turn do not hold any information on the boundaries of a potential continuous admixture scenario. This however could be highly informative about the first contact between humans and Neandertals and the potential time of Neandertal extinction.

Hence, by using our continuous admixture model and the information on influencing parameters, we could obtain admixture durations modeling the ALD using the Lomax distribution under ideal circumstances, where we exactly know the genetic distance between introgressed SNPs and the recombination process is constant. Relaxation of the latter assumption while still knowing the exact population specific recombination map  leads to  uncertainty in the estimates. This uncertainty depends on the duration and time of sampling from the end of the admixture event. Crucially we could show that there is no power to correctly infer admixture durations even for long admixture scenarios under ideal circumstances if the event is to long ago. Likewise, reliable distinction between a pulse like admixture model and a multi-generation continuous model for Human-Neandertal admixture scenarios is not definite. We demonstrated that a multitude of different duration scenarios are compatible for the Human-Neandertal admixture estimated from  present-day non-Africans genomes. Hence, we conclude that using present day non-African human genomes for inference is not conclusive to resolve the duration of Neandertal admixture. As a consequence we could neither obtain genetic indications of the first contact between Humans and Neandertals nor for the time of potential Neandertal extinction.

While our results show that drawing concrete conclusions from introgression time estimates, other avenues for differentiating different gene flow events remain fruitful.
In particular, measures based on population differentiation (e.g \cite{browning_analysis_2018}, \cite{wall2013_genetics}, \cite{villanea_schraiber 2018_nature_ee}, \cite{jacobs_multiple_2019}) are very promising to understand the different events that contributed to archaic ancestry in present-day humans. 

The other major avenue are ancient DNA samples from early modern humans that lived around or immediately after gene flow between archaic and modern humans occurred. The \textit{Oase 1} genome is a prime example, as it directly demonstrates that some gene flow must have occurred in Europe, around 40,000 years ago, concurrently with when \textit{Oase 1 lived}. More generally, detailed analyses of ancient genomes from the initial upper paleolithic should provide more accurate mean dates, as they are much closer to the gene flow, and may give better evidence about the duration of the gene flow. The drawback is that some of the gene flow these individuals experience may be private to their population, as has e.g. been suggested for \textit{Oase 1}(\cite{fu_genome_2014}). Larger numbers of early modern human genomes, as well as methods to accurately infer admixture tracts or ALD in low-coverage data will likely enable this line of research. 

 Another possibility is using ancient DNA from admixed individuals to sample closer from the end of the admixture event. This could potentially help in either deconvolute discrete admixture pulses, which has the disadvantage of fitting multiple parameters or using our suggested approach on the ancient data. With a sufficient amount of ancient DNA samples from the same time period one could infer a recombination map. With that our study suggest that it would be possible to estimate a potential multi-generation admixture duration. One could also think of combining both ideas. Allocate introgressed segments from ancient DNA samples into different ancestry clusters according to their genetic differentiation and separately inferring mean and duration of the admixture time on each ancestry cluster. 

Conclusions:

\begin{itemize}
    \item The mean time of gene flow can be reliably estimated with a pulse model even if the gene flow is actually continuous. (check)
  \item Technical model variables (ascertainment scheme, minimal distance, demography and recombination map) outweigh the effect of continuous gene flow. (check)
  \item The Lomax model is under powered to inferring continuous admixture for parameters relevant for Neandertal admixture. (check)
  \item Estimating  the mean time and especially the duration of gene flow is only possible with a highly precise genetic map. (check)
  \item The signal of continuous gene flow is hard to detect if the gene flow is not long enough or to far in the past. (check)
  \item Reliably distinguishing the models is difficult. (check)
\end{itemize}

Outlook:

\begin{itemize}
  \item Population differentiation can help identifying different admixture events across time (e.g. Browning et al. 2018) (check)
  \item sampling close to the end of the admixture e.g. using ancient genomes and using a accurate recombination map enables the inference of admixture duration (check)
  
\end{itemize}