\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fontshttps://de.overleaf.com/project/5e85b0680d0bed00011ea790
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={The dating of the Human-Neandertal introgression event estimated from present-day human genomes is compatible with a multitude of admixture durations},
            pdfauthor={Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary Anthropology, MPI EVA), Dr.~Benjamin Marco Peter (MPI EVA, benjamin\_peter@eva.mpg.de)},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{plainnat}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{The dating of the Human-Neandertal introgression event estimated from present-day human genomes is compatible with a multitude of admixture durations}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary
Anthropology, MPI EVA), Dr.~Benjamin Marco Peter (MPI EVA,
\href{mailto:benjamin_peter@eva.mpg.de}{\nolinkurl{benjamin\_peter@eva.mpg.de}})}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2020-03-24}

\usepackage{setspace}
\doublespacing
\usepackage[none]{hyphenat}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{float}
\usepackage{xcolor}
\floatplacement{figure}{H}

\begin{document}
\maketitle

\section{Abstract}\label{abstract}

\section{Introduction}\label{introduction}

\subsection{(Archaic) Interbreeding, genomic consequences and why is it interesting}\label{(Archaic) Interbreeding, genomic consequences and why is it interesting}

In recent years, the sequencing of Neandertal \citep{green_draft_2010,prufer_complete_2013,prufer_high-coverage_2017, mafessoni_high_coverage_2020} and Denisovan genomes \citep{reich_genetic_2010, meyer_high-coverage_2012} revealed that modern humans outside of Africa were in contact and interbred with other hominin populations \citep{vernot_resurrecting_2014,fu_early_2015,fu_genome_2014,sankararaman_genomic_2014,sankararaman_combined_2016,vernot_excavating_2016,malaspinas_genomic_2016}. On a genomic level, the interbreeding with a migrant population (admixture) introduces divergent chromosomes into the mixing population. Over time, meiotic recombination progressively breaks these divergent chromosomes down into admixture segments, whose size decrease with time \citep{falush_inference_2003}. Assuming that recombination events are independent from eachother, the length distribution of the admixture segments is roughly inversely proportional to the number of generations since the interbreeding happened. Hence, using this 'recombination clock', the length distribution of introgressed chromosomal segments can be used to infer the time of an
admixture event \citep{moorjani_history_2011,pugach_dating_2011,sankararaman_date_2012,loh_inferring_2013,sankararaman_combined_2016,pugach_gateway_2018,jacobs_multiple_2019,hellenthal_genetic_2014}. \citep{pool_inference_2009,moorjani_history_2011,gravel_population_2012,liang_lengths_2014}. One particular event of interest is the time and duration of interbreeding between Neandertals and modern humans , i.e. the time between start and end of sympatry of Neandertals and modern humans outside of Africa. This could give a clue to the debated questions of the time of the first modern humans leaving Africa and the extinction of Neandertals. The earliest modern human remains outside of Africa are directly dated to around 188 thousand years ago (kya)  using a combination of uranium-thorium, uranium-series and electron spin resonance dating techniques  \citep{stringer_when_2018,hershkovitz_earliest_2018} and a possible extinction of Neandertals is suggested around 37 kya inferred by radiocarbon and luminescent dating of a Neandertal site in Spain \citep{zilhao_precise_2017} and 39 kya from radiocarbon dates of Neandertal sites across Europa \citep{higham_timing_2014} at the end of the Mousterian. This leaves a potential time frame for Neandertal and modern human interaction of approximately 140,000 years. Detailed genetic dating of Neandertal and modern human admixture could unveil more evidence of the timing and especially duration of this interaction. 

\subsection{Admixture models}\label{Admixture models}

The accuracy and information content of admixture date estimates depend on the model used and its assumptions. The most widely used admixture model is the Isolation-Migration model (IM), which models divergence of two populations from an ancestral one by jointly estimating the divergence time and migration rate \citep{nielsen_distinguishing_2001,hey_multilocus_2004}. In the IM model however, migration has no time component and is assumed constant i.e. in our case from the divergence time of Neandertals and modern humans till today. Dates of Admixture events can be estimated by modeling the admixture segment length distribution. For this purpose, most models assume that admixture segments are rare and inbreeding is not significant, such that admixture segments are unlikely to recombine with eachother \citep{pool_inference_2009,liang_lengths_2014}. The segments act neutral \citep{shchur_distribution_2019} and the recombination clock is constant over time and populations \citep{gravel_population_2012}. In addition, it is generally assumed that the admixture happens over a very short time period, in a single \textit{admixture pulse} \citep{moorjani_history_2011}, usually modelled as a single generation of interbreeding.


\subsection{The two approaches and their application to find archaic admixture dates}\label{the-two-approaches-and-their-application-to-find-archaic-admixture-dates}

The first step in dating admixture events from genetic data is therefore estimating the length distribution of admixture segments.  There are two main approaches for this; a first approach is to use patterns of linkage along a chromosome to estimate the length distribution, without explicitly inferring the genomic location of these segments. In contrast, a second set of methods first aims to identify all admixture segments over a certain length, and then use these segments for inference. 
(\citep{chimusa_dating_2018}) (Figure \ref{fig:fig1} A).

The first approach uses the admixture-induced linkage disequilibrium (ALD) decay. Variants on introgressed archaic segments are expected to be in high linkage disequilibrium to each other at the time of admixture \citep{chakraborty_admixture_1988,stephens_mapping_1994,wall_detecting_2000}. The extent of linkage between introgressed variants decreases over generations. Hence, in case of a recent admixture event a few tens of generations ago, ALD stretches  over long genetic distances
\citep{patterson_methods_2004} and is therefore easily distinguishable from short range LD due to inheritance of chromosomal segments from the ancestral population as well as LD caused by bottlenecks and genetic drift after the split of the populations \citep{moorjani_history_2011,sankararaman_date_2012}. For ancient admixture events however, ALD is quite similar to the genomic background \citep{sankararaman_date_2012}. To circumvent this issue for dating the Neandertal-human admixture time, an ascertainment scheme was used to calculate LD only for markers that are nearly differentially fixed between the two taxa. In this case, the presence of apparent Neandertal alleles in close-range LD is a signature of a locally introgressed locus
\citep{sankararaman_date_2012}. Typically, estimation of admixture dates proceeds by fitting a decay curve of pairwise LD as a function of genetic distance, using an exponential distribution whose parameters are informative for the time of an admixture pulse \citep{moorjani_history_2011,loh_inferring_2013}. Using this approach Sankararaman et al. dated the Neandertal-human admixture pulse to be  between 37,000--86,000 ya (years ago) \citep{sankararaman_date_2012}. Later, this date was refined to 40,510--54,454 ya (95\% CI) using a different ascertainment scheme combined with a different genetic map \citep{moorjani_genetic_2016}. A date was also obtained from an ancient genome to be 50,000 - 60,000 ya by adding the time since the admixture obtained from the ancient individual, by the decay of pairwise covariance between
introgressed SNPs, to the specimens radiocarbon date \citep{fu_genome_2014}.

The Denisovan modern human admixture time point was estimated to lie in the interval between 44,000--54,000 ya using the ALD
approach on modern day genomes \citep{sankararaman_combined_2016}. 

Using the second set of approaches, archaic segments in genomes from present day Southeast-Asians revealed a proportion of previously unknown Denisovan ancestry private to these populations. The segments from this ancestry are more diverged from the high-coverage Denisovan genome then previously found ones. This suggests an additional admixture event from a different population of Denisovans \citep{browning_analysis_2018}. The identification of segments is largely independent from the later dating, and can be done using a variety of methods \citep{racimo_signatures_2017,seguin_orlando_paleogenomics_2014,vernot_excavating_2016,sankararaman_combined_2016,skov_detecting_2018}. The length distribution of the obtained fragments is then used to estimate the time of the admixture pulse, typically using an exponential model.
Comparing the mean length of the formerly known and newly identified Denisovan segments did, however, not reveal significant differences, suggesting a lack of power to distinguish the
two events by time \citep{browning_analysis_2018,jacobs_multiple_2019}. Analysing genomes from Papuan individuals revealed two time separated admixture events with Denisovans, one in line with previous estimates at 45.7 kya (95\% CI 31.9-60.7 kya) and one exclusive to Papuans dated to
be around 29.8 kya (95\% CI 14.4-50.4 kya) \citep{jacobs_multiple_2019}.


\subsection{Why can't we us the pulse model}\label{Why can't we us the pulse model}

While convenient for inference, one cannot use the admixture pulse model to distinguish an extend period of admixture from an admixture pulse \citep{pickrell_toward_2014}. Here, we are chiefly interested in the duration of the interbreeding between Neandertals and modern humans, especially in the start and end point.  Under an admixture pulse model, which assumes that these two times coincide, hypotheses about the impact of modern human colonization of Eurasia on Neandertal extinction cannot be evaluated. Therefore, we introduce the \textit{extended admixture pulse} model, allowing for multigeneration continuous admixture. 

\subsection{Previous attempts of a general admixture model and ours}\label{Previous attempts of a general admixture model and ours}

The extended pulse model is a generalization of the simple admixture pulse model. The simplest generalization of this model is to assume two or more discrete pulses. In such a case,  each segment can be thought of having entered in one of the pulses, and the resulting admixture tracts will be a discrete mixture of the constituent distributions \citep{pickrell_ancient_2014}, weighted by the relative migration rates. Zhou et al. 2017 \citep{zhou_modeling_2017} showed that this model, in principle, can be used for continuous mixtures as well, using a polynomial function as a mixture density. However, they found that even for relatively short admixture events, the large number of parameters led to an underestimate of admixture duration \citep{zhou_inference_2017}, and the beginning and end of admixture were not well inferred
\citep{zhou_modeling_2017,zhou_inference_2017}. 

Here we use a simpler model of continuous admixture with just two parameters, and one less than the two-pulse model of Pickrell et al. 2014 \citep{pickrell_ancient_2014}. One parameter reflects the mean admixture time, and the other the duration of the admixture event; letting this parameter go to zero thus recovers the (nested) simple pulse model. 
This model is particularly simple if we assume that migration rate over time is Gamma distributed, in which case the distribution of admixture segment lengths has a closed form (Figure \ref{fig:fig1} B & C).

Nevertheless, this is a difficult problem, as it requires deconvoluting an exponential mixture, which is notoriously hard \citep{dasgupta_mixture_2008}. Thus, we have to carefully evaluate potential sources of biases known from previous studies, namely the demography of the admixed population, the accuracy of the recombination map and the ascertainment scheme \citep{sankararaman_date_2012,fu_genome_2014,moorjani_genetic_2016}.
We compare the extended pulse with the simple pulse model, to examine the implications of the one-generation assumption in relation to the aforementioned potential biases. We test when the simple pulse model can be contrasted for the more general model of an extended admixture pulse.



\subsection{What we want to do}\label{what-we-want-to-do}

In our study, we first define the new extended admixture pulse model describing the expectation of the resulting segment length distribution for Gamma distributed admixture being Lomax distributed, holding a parameter for the duration of admixture. This model works for
both methods to infer the segments length, either directly or by using
the ALD decay. Second, we examine the effect of ongoing admixture on the admixture time estimates on methods that assume a one generation pulse. Using the extended pulse model, we investigate under which scenarios the parameters of the Lomax-distribution can be accurately estimated and for which parameters we can distinguish a pulse-like admixture event from a continuous event. We show that in many cases of ancient admixture, pulses cannot be distinguished from more continuous admixture events. Using the 1000 Genomes data, ALD inferred admixture times from Europeans are consistent with a multitude of duration times.
We conclude that current methods are unsuitable to definitively estimate the duration of admixture and thus answer when the contact between Neandertals and modern humans started or ended.

\hypertarget{refs}{}

\bibliography{References/MyLibraryATE}

\end{document}