\documentclass[11pt]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{hyperref}
\usepackage{ifxetex,ifluatex}
%\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fontshttps://de.overleaf.com/project/5e85b0680d0bed00011ea790
}{}
\usepackage[margin=1in]{geometry}

\usepackage{hyperref}
%\newcommand*\referencetitle[1]{\def\@currentlabelname{#1}}

\hypersetup{unicode=true,
            pdftitle={Title: An extended admixture pulse model reveals the limitations to Human-Neandertal introgression dating},
            pdfauthor={Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary Anthropology, MPI EVA), ~Harald  Ringbauer (MPI EVA),
            ~Benjamin Marco Peter (MPI EVA, benjamin\_peter@eva.mpg.de)},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
 
\usepackage{natbib}
%\bibliographystyle{References/my_abbrvnat}
\bibliographystyle{my_abbrvnatI can }
\setcitestyle{authoryear,open={(},close={)}}

\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
%%% Introduced by Harald
\newcommand*\diff{\mathop{}\!\mathrm{d}}

\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% HELPER CODE FOR DEALING WITH EXTERNAL REFERENCES
%\usepackage{xr}
%\makeatletter
%\newcommand*{\addFileDependency}[1]{
%  \typeout{(#1)}
%  \@addtofilelist{#1}
%  \IfFileExists{#1}{}{\typeout{No file #1.}}
%}
%\makeatother

%\newcommand*{\myexternaldocument}[1]{
%    \externaldocument{#1}
%    \addFileDependency{#1.tex}
%    \addFileDependency{#1.aux}
%}
%%% END HELPER CODE

%\myexternaldocument{Paper/Supplements_Review}

\usepackage{setspace}
\onehalfspacing
\usepackage[left]{lineno}
%\linenumbers
\usepackage[none]{hyphenat}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{float}
\usepackage{xcolor}
\floatplacement{figure}{H}

\begin{document}


\begin{titlepage}

    \begin{flushright}
        \large
        \textbf{Article (Methods)}
    \end{flushright}


        \vspace*{1cm}
    \begin{center}       
        \Huge
        \vspace{1cm}
        An extended admixture pulse model reveals the limitations to Human-Neandertal introgression dating
        
        \vspace{1.0cm}
        \large
        Iasi, Leonardo N. M. \textsuperscript{1,2}, Ringbauer, Harald  \textsuperscript{3,4}, and Peter, Benjamin M. \textsuperscript{1,5} \\ 
        
        \vspace{1.0cm}
        
        \textsuperscript{1}Department of Evloutionary Genetics, \\ 
        Max Planck Institute for Evolutionary Anthropology, Leipzig, Germany
        
        \textsuperscript{3}Department of Archaeogenetics, \\ 
        Max Planck Institute for Evolutionary Anthropology, Leipzig, Germany
        
        \vspace{1.0cm}
        \textsuperscript{2} leonardo\_iasi@eva.mpg.de \\
        \textsuperscript{4} 
        harald\_ringbauer@eva.mpg.de \\\textsuperscript{5} 
        benjamin\_peter@eva.mpg.de \\
        \vspace{1.0cm}
        \today
    \end{center}  
     

            

\end{titlepage}


\section{Abstract}\label{sec:abstract}

Neandertal DNA makes up 2-3\% of the genomes of all non-African individuals. The patterns of Neandertal ancestry in modern humans have been used to estimate that this is the result of gene flow that occurred during the expansion of modern humans into Eurasia, but the precise dates of this event remain largely unknown. Here, we introduce an extended admixture pulse model that allows joint estimation of the timing and duration of gene flow. This model leads to simple expressions for both the admixture segment distribution and the decay curve of ancestry linkage disequilibrium, and we show that these two statistics are closely related.  In simulations, we find that estimates of the mean time of admixture are largely robust to details in gene flow models, but that the duration of the gene flow can only be recovered if gene flow is very recent and the exact recombination map is known. These results imply that gene flow from Neandertals into modern humans could have happened over hundreds of generations. Ancient genomes from the time around the admixture event are thus likely required to resolve the question when, where, and for how long humans and Neandertals interacted.

\section{Introduction}\label{sec:introduction}

The sequencing of Neandertal  \citep{green_draft_2010,prufer_complete_2013,prufer_high-coverage_2017, mafessoni_high_coverage_2020} and Denisovan genomes \citep{reich_genetic_2010, meyer_high-coverage_2012} revealed that modern humans outside of Africa interacted, and received genes from these archaic hominins \citep{vernot_resurrecting_2014,fu_genome_2014,sankararaman_genomic_2014,fu_early_2015,malaspinas_genomic_2016,sankararaman_combined_2016,vernot_excavating_2016}. There are two major lines of evidence: First, Neandertals are genome-wide more similar to non-Africans than to Africans \citep{green_draft_2010}. This shift can be explained by 2-4\% of admixture from Neandertals into non-Africans \citep{green_draft_2010, prufer_complete_2013}. Similarly, East Asians, Southeast Asians and Papuans are more similar to Denisovans than other human groups, which is likely because of gene flow from Denisovans \citep{meyer_high-coverage_2012}. 

As a second line of evidence, all non-Africans carry genomic segments that are very similar to the sequenced archaic genomes. As these putative \emph{admixture segments} are up to several hundred kilobases (kb) long, it is unlikely that they were inherited from a common ancestor that predates the split of modern and archaic humans \citep{sankararaman_genomic_2014, vernot_resurrecting_2014}. Rather, they entered the modern human populations through later gene flow \citep{sankararaman_date_2012,sankararaman_genomic_2014, vernot_resurrecting_2014, sankararaman_combined_2016,vernot_excavating_2016}. 


However, substantial uncertainty remains about when, where, and over which period of time this gene flow happened. A better understanding of the location and timing of the gene flow would allow us to place constraints on the timing of movements of early humans, and the population genetic consequences of their interactions.

Archaeological evidence puts some temporal boundaries on the times when Neandertals and modern humans might have interacted. The earliest currently known modern human remains outside of Africa are dated to around 188 thousand years ago (kya)  \citep{hershkovitz_earliest_2018,stringer_when_2018} and the latest Neandertals are suggested to have lived between 37 kya and 39 kya old \citep{higham_timing_2014,zilhao_precise_2017}. Thus the time window where Neandertals and modern humans might have been in the same area stretches over more than 140,000 years. However, there is less direct evidence of modern humans and Neandertals in the same geographical location at the same time. In Europe, for example, Neandertals and modern humans likely overlapped only for less than 10,000 years \citep{bard_extended_2020}. 

\subsection{Genetic Dating of Gene Flow}\label{Admixture models}

A common approach to learn about admixture dates from genetic data uses a \emph{recombination clock} model: Conceptually, admixture segments are the result of the introduced chromosomes being broken down by recombination. The first generation offspring of an archaic and a modern human parent will have one whole chromosome each of either ancestry. Thus, the genomic markers in these individuals are in full ancestry linkage disequilibrium (ALD); all archaic variants are present on one DNA molecule, and all modern human variants on the other one.

If this individual has offspring in a largely modern human population, in each generation meiotic recombination will reshuffle the chromosomes, progressively breaking down the ancestral chromosome down into shorter segments of archaic ancestry \citep{falush_inference_2003, gravel_population_2012,liang_lengths_2014}, and ALD similarly decreases with each generation after gene flow \citep{chakraborty_admixture_1988,stephens_mapping_1994,wall_detecting_2000}.


This inverse relationship between admixture time and either segment length or ALD is commonly used to infer the timing of gene flow \citep{pool_inference_2009,moorjani_history_2011,pugach_dating_2011,gravel_population_2012,sankararaman_date_2012,loh_inferring_2013,hellenthal_genetic_2014,liang_lengths_2014,sankararaman_combined_2016,pugach_gateway_2018,jacobs_multiple_2019}. Most commonly, it is assumed that gene flow occurs over a very short duration, referred to as an \textit{admixture pulse}, which is typically modelled as a single generation of gene flow \citep[e.g][]{moorjani_history_2011}. This model has the advantage that both the length distribution of admixture segments and the decay of ALD with distance will follow an exponential distribution, whose parameter is directly informative about the time of gene flow \citep{pool_inference_2009, gravel_population_2012, liang_lengths_2014}.

In segment-based approaches, dating starts by identifying all admixture segments, which can be done using a variety of methods \citep{seguin_orlando_paleogenomics_2014,sankararaman_combined_2016,vernot_excavating_2016,racimo_signatures_2017,skov_detecting_2018}. The length distribution of inferred segments is then used as a summary for dating when gene flow happened.

Alternatively, ALD-based methods use linkage disequilibrium (LD) patterns, without explicitly inferring the segments \citep{chimusa_dating_2018} (Figure \ref{fig:fig1} B). Instead, admixture dates are estimated by fitting a decay curve of pairwise LD as a function of genetic distance, implicitly summing over all compatible segment lengths \citep{moorjani_history_2011,loh_inferring_2013}. 


\subsection{Neandertal Gene Flow Estimates}
Using this approach,   \cite{sankararaman_date_2012} dated the Neandertal-human admixture pulse to  between 37--86 kya. Later, \cite{moorjani_genetic_2016} refined this date to 41 -- 54 kya ($C.I._{95\%}$) using an updated method, a different marker ascertainment scheme and a refined genetic map for European populations. A date of 50 -- 60 kya was obtained from the analysis of the genome of \textit{Ust'-Ishim}, a 45,000-year-old modern human from western Siberia. The inferred Neandertal segments in \textit{Ust'-Ishim} are substantially longer than those in present-day humans, which makes their detection easier, and adds further evidence that gene flow between Neandertals and modern humans has happened relatively recently before \textit{Ust'-Ishim} lived \citep{fu_genome_2014}. In addition, we have direct evidence of gene flow from early modern humans from Oase \citep{fu_early_2015} and Bacho Kiro \citep{hajdinjak_initial_2021}, dated to 40ky and 45ky, respectively. In genomes from both sites, segments of recent Neandertal ancestry less than ten generations before hint at admixture histories with late gene flow in Europe.


\subsection{Limitations of the Pulse Model}\label{Why can't we us the pulse model}

The admixture pulse model assumes that gene flow occurs over a short time period; however it is currently unclear how long a time could still be consistent with the data. This makes admixture time estimates hard to interpret, as more complex admixture scenarios might be masked, and so gene flow could have happened tens of thousands of years before or after the estimated admixture time.

That admixture histories are often complicated has been shown in the context of Denisovan introgression into modern humans, where at least two distinct admixture events into East Asians and Papuans were proposed \citep{browning_analysis_2018, jacobs_multiple_2019,choin_genomic_2021}. While the length distributions of admixture segments are similar between populations, there are  differences in the genomic distribution of admixture segments, and their similarities to the sequenced high-coverage Denisovan \citep{browning_analysis_2018, massilani_denisovan_2020}. In contrast, all Neandertal admixture segments are most similar to the Vindija Neandertal  \citep{prufer_high-coverage_2017}, but Neandertal ancestry is slightly higher in East Asians than Western Eurasians \citep{meyer_high-coverage_2012, wall_higher_2013,kim_selection_2015,vernot_complex_2015,villanea_multiple_2019}.

One way to refine admixture time estimates is to include two or more distinct admixture pulses. The distribution of admixture segment lengths will then be a mixture of the segments introduced from each event. This is especially useful if the events are very distinct in time, e.g. if one event is only a few generations back, and the other pulse occurred hundreds of generations ago \citep{fu_genome_2014, fu_early_2015}. In this case, the admixture segments will be either very long if they are recent, or much shorter if they are older.

\cite{zhou_modeling_2017} extended this model to continuous mixtures, using a polynomial function as a mixture density. However, they found that even for relatively short admixture events, the large number of parameters led to an underestimate of admixture duration \citep{zhou_inference_2017}. 
\subsection{Extended Pulse Model}
One drawback of these approaches is that they introduce a large number of parameters. Even a discrete mixture of two pulses requires at least three parameters (two pulse times and the relative magnitude of the two events) \citep{pickrell_ancient_2014}, and the more complex models require regularization schemes for fitting \citep{zhou_inference_2017, ralph_geography_2013}.

Here, we propose an \emph{extended admixture pulse} model (Figure \ref{fig:fig1} A) to estimate the duration of an admixture event. It only adds one additional parameter, reflecting the duration of gene flow, while retaining much of the mathematical simplicity present in the simple pulse model. 
The extended pulse model assumes that the migration rate over time is Gamma distributed, so that the length distribution of admixture segments has a closed form (Figure \ref{fig:fig1} C \& D) with two parameters, the mean admixture time and duration.

Conceptually, identifying an extended pulse requires us to establish that the length distribution of admixture segments deviates from an exponential distribution. However, other sources of bias, such as the demography of the admixed population, the accuracy of the recombination map or details in the inference method parameters may also introduce similar biases. Thus, we have to carefully evaluate other potential sources of bias on whether they might lead to confounding signals. \citep{sankararaman_date_2012,fu_genome_2014,moorjani_genetic_2016}. 


Here, we first define the extended admixture pulse model and derive the resulting segment length and ALD distributions, and introduce inference schemes for either data. We then evaluate under which scenarios these two models can be distinguished.  We show that power to distinguish these scenario is higher for more recent events and longer pulses, but that accurate inference requires high-quality data. Based on these results, we use data from European genomes  \citep{the_1000_genomes_project_consortium_global_2015} and find that for the case of Neandertal admixture, a simple pulses cannot be distinguished from continuous admixture over an extended period of time, and the data are consistent with a multitude of durations, up to several tens of thousands of years. 

\begin{figure}
\centering
%\includegraphics[width=16cm,height=18cm,keepaspectratio]{Admixture_Time_Inference_Paper_Draft_files/figure-latex/fig1-1.pdf}
\includegraphics[width=16cm,height=18cm,keepaspectratio]{Fig_1_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:fig1} A) Neandertal introgression into non-Africans with a multitude of potential admixture durations. B) The time and duration of admixture results in different length distributions of introgressed chromosomal segments (grey) containing  Neandertal variants (green circles)  in high LD to each other
compared to the background (human variants white stars). The ALD approach estimates linkage
between the introgressed variants (green circles), whereas the haplotype approach tries
to estimate the segment directly (grey area). C) Migration rate per generation
modeled using the extended pulse model for different admixture durations (colored lines). The filled area under the curve indicates the boundaries of the discrete realization of the duration of gene flow $t_d$.
The dotted line indicates the oldest possible time of gene flow (as defined in the simulations). D) The expected LD decay under the extended pulse model.}
\end{figure}


\section{New Approaches}\label{new approaches}

In this section, we present the mathematical description of the admixture models we use in this paper, and introduce inference algorithms for estimating the admixture time and duration from both segment data and ALD. 


\subsection{Admixture Models and Inference}\label{admixture models}
	
We think of admixture as a series of ``foreign'' chromosomes introduced in a population (for a mechanistic model, see e.g. \cite{pool_inference_2009}. Throughout, we assume that alleles evolve neutrally, and that recombination is independent of local ancestry. The simple pulse model assumes that all admixture happens in the same generation, (\textit{i.e.} all chromosomes are introduced to the population at the same time). To extend this model, we allow chromosomes to enter at potentially many different time points, such that the migration rate at time $t$ in the past is given by the function $m(t)$ \citep{pool_inference_2009, ni_length_2016}. For simplicity, we assume that the total amount of introgressed material $\alpha=\int_0^\infty m(t)dt$ is small, so that segments do not interact, but we will discuss violations of this assumptions later. For archaic introgression, $\alpha \approx 0.03$, so this assumption is justified. 

Over time, recombination splits up the introgressed genome into smaller pieces, while by the neutrality assumption the expected amount of total ancestry remains approximately the same. Thus, if we measure the size of chromosomes in recombination units, a chromosome of size $G$ introduced at time $t$ gives rise to an expected number of $tG$ segments.


\subsection{Admixture Segment Lengths}
We enumerate the admixture segments in a sample $i=1\dots K$. We denote the length of the $i$-th segment as $L_i$ (measured in Morgan) and the time in the past when segment $i$ entered the population as $T_i$ (measured in generations). We assume that the $L_i$ and $T_i$ are both realizations from more general distributions $L$ and $T$ that reflect the overall segment length and segment age distributions, respectively. 

To relate $m(t)$ to $T$, we need to take into account that older fragments had more time to split up \citep[see e.g.][]{pool_inference_2009}. Hence

\begin{equation}
	P(T_i=t) = \frac{t  G m(t)}{\int_0^{\infty} t  G  m(t) dt}\label{eq:reweighting} \text{.}
\end{equation}

The denominator of the right-hand side term in equation \ref{eq:reweighting} is the expected number of admixture segments, $\mathbb{E}[K] = \int_0^{\infty} t  G  m(t) dt$.

Given $T_i$, the segment length $L_i$ is exponentially distributed with rate parameter $t$:

\begin{equation}
 \label{eq:generall_length_distribution}
    P(L_i=l|T_i=t) = t e^{-t l}  \text{.}
\end{equation}
	
Integrating over $T$ yields the unconditional distribution of admixture segment lengths:
	
\begin{align}
P(L_i=l) &= \int_{0}^{\infty} P(T_i=t) P(L_i=l | T_i=t) \ dt \text{,}\nonumber\\
&=\frac{G}{ \mathbb{E}[K]}\int_{0}^{\infty} t^2 m(t) e^{-tl}dt
    \label{eq:standard_likelihood_definintion}
\end{align}
	
	
and we can think of $L$ as an exponential mixture distribution with mixture density proportional to $tm(t)$ \citep{ralph_geography_2013, ni_length_2016, zhou_modeling_2017}.
	
\subsubsection{Ancestry Linkage Disequilibrium}
Alternatively, the impact of gene flow is often characterized using ALD, particularly when accurate identification of archaic segments is difficult. We follow \cite{loh_inferring_2013} and note that the ALD from gene flow in a single event at time $t$ generations in the past is

\begin{equation}
    D_{t} = m(1-m)\Delta_x \Delta_y \approx m A \text{,} \label{eq:ald_general}
\end{equation}

where $m$ is the fraction of immigrants and $\Delta_x, \Delta_y$ are the differences in allele frequencies between markers in the admixing populations. We assume that terms of the order of $m^2$ can be ignored and that migration is low enough that changes in the allele frequencies in the admixing populations can also be neglected (i.e. $A=\Delta_x\Delta_y$ remains a constant). 

At a later generation $t$, the expected LD between two markers a distance $l$ apart is
\begin{equation}
    D_s \approx D_t \exp(-l(s-t))\text{,}
\end{equation}
due to the decay of LD \cite[e.g.][]{sankararaman_date_2012}. If the  migration rate $m_f$ is a function of time, we can add up the LD introduced at each time $s$ in the past and approximate $D$ as 
\begin{equation}
    D_s = A\int_{-\infty}^s m_f(t)\exp(-l(s-t)) dt \text{.} \label{eq:ld_general_bwd}
\end{equation}

As we show in the Appendix (\nameref{Appendix_1}), equation \ref{eq:ld_general_bwd} satisfies the differential equation
\begin{align}
    \frac{dD_s}{ds} = -l D_s + A m_f(s)\text{,}
\end{align}
where the $-l D_s$-term models the exponential decay of LD due to recombination, and the $A m_f(s)$-term reflects the increase of LD due to admixture (Eq. \ref{eq:ald_general}).

To connect this equation more directly to the backward-in-time formulation used in the derivation of the admixture segment distribution, we set $s=0$ and invert the flow of time, such that $m(t) = m_f(-t)$. We obtain

\begin{equation}
\label{eq:ld_general_inverted}
    D(l) = A\int_0^{\infty} m(t)\exp(-lt) dt \text{.} 
\end{equation}

Thus, $D$ can be interpreted as the tail function of an exponential mixture with mixture density $m$. Alternatively, the integral in equation \ref{eq:ld_general_inverted} is also the (scaled) moment-generating function of $m$ with argument $-l$. 

The distribution of admixture segment lengths (equation \ref{eq:standard_likelihood_definintion}) and the ALD function (equation \ref{eq:ld_general_inverted}) are closely related -- in the Appendix (\nameref{app:connection}) we show that 

\begin{align}
D(l) &= \frac{\mathbb{E}(K)}{G} \int_l^\infty P(x) (x-l) d x \label{eq_fr1}\\
P(l)  &\propto D''(l).
\label{eq_fr2}
\end{align}

It follows that both functions uniquely determine each other. Consequently they contain identical information to estimate admixture dates.

Both for the segment and ALD models we use simplifying assumptions that ignore the effects of genetic drift,  the recombination between introgressed segments and the replacement of older introgressed material. In the Appendix, we discuss these approximations and show that particularly the replacement of admixed material can be accommodated by replacing $m$ with
\begin{equation}
    m_e(t) = m(t)\exp\left[-\int_0^t m(s)ds\right] \text{,} \label{eq:effective_migration}
\end{equation}
which can be interpreted as the probability of the event that migration happened at time $t$, and no more migration happened later on.



\subsubsection{The Simple Pulse Model}\label{The Simple Pulse Model}
	
	
Under the simple pulse model, all fragments enter the population at the same time $t_m$, and $T$ is a constant distribution. We can formalize this model by using a Dirac delta function which integrates to one if the integration interval includes $t_m$ and zero otherwise:

\begin{subequations}
\begin{equation}
\label{eq:RV_simple_pulse_1}
	m(t)=  \alpha \delta_{t_m}(T_i),
\end{equation} 
	
\begin{equation}
\label{eq:RV_simple_pulse_2}
	P(T_i)=\delta_{t_m}(T_i),
\end{equation} 
\end{subequations}


	
We obtain the exponential distribution of admixture fragments under this model \citep[e.g.][]{moorjani_history_2011}:

\begin{subequations}
\begin{align}
\label{eq:Likelihood_function_simple_pulse}
	P(L_i=l) &= t_me^{-t_m l}\\
	D(l) &\propto e^{-t_m l},
\end{align}
\end{subequations}	
where here and in the remainder of this section we omit the constant term from $D$, which is not relevant for fitting the LD decay. The expected segment length under a simple pulse model is given by
\begin{subequations}

\begin{equation}
\label{eq:Expected_l_simple_pulse}
\mathbb{E}[L]=\frac{1}{t_m}
\end{equation}
	
and the variance by
\begin{equation}
\label{eq:Expected_v_simple_pulse}
\text{Var}[L]=\frac{1}{t_m^2} \text{.}
\end{equation}
\end{subequations}	

\subsubsection{The Extended Pulse Model}\label{The Extended Pulse Model}

For the new extended pulse model, we assume that the migration rate $m(t)$ follows a rescaled Gamma distribution so that the total contribution of migrant alleles is $\alpha$.  It is convenient to parameterize the migration rate as $\Gamma(k,\frac{t_m}{k})$.
for $t \geq 0$ and $k \geq 1$. 

\begin{subequations}
Using this parameterization, the denominator of equation \ref{eq:reweighting} is $t_m \alpha G$ and

\begin{align}
    P(T_i=t) &= \frac{t}{t_m}m(t)\\
        &=\frac{1}{\Gamma(k)(\frac{t_m}{k})^k}t^{k-1}e^{-t\frac{k}{t_m}} \label{eq:Gamma_density}
\end{align}
\end{subequations}
for $t \geq 0$ and $k \geq 2$, which is is the density of a $\Gamma(k+1, \frac{t_m}{k})$-distribution with moments
	
\begin{equation}
\begin{split}
\label{eq:RV_extended_pulse_properties}
\mathbb{E}[T]&= \frac{k+1}{k}t_{m} \\
	Var[T]&=\frac{k+1}{k^2}t_{m}^2 =
\frac{k+1}{k}\bigg(\frac{t_d}{4} \bigg)^2 \text{.}
\end{split}
\end{equation}
	
Here, we define the admixture duration $t_d=4t_m k^{-\frac{1}{2}} $, as a convenient measure for the duration of gene flow. If $k$ is low, then $t_d$ will be large and gene flow extends over many generations. In contrast, if $k$ is large, then $t_d \approx 0$ and we recover the simple pulse model (Figure \ref{fig:fig1} C \& D). 
	
	
The distribution of segment length is calculated by plugging equation \ref{eq:Gamma_density} into  equation  \ref{eq:standard_likelihood_definintion} and integrating:

\begin{align}
\label{eq:Likelihood_function_extended_pulse}
    P(L=l | k, t_m) &= \int_{0}^{\infty} \frac{1}{\Gamma(k)(\frac{t_m}{k})^k}t^{k-1}e^{-t\frac{k}{t_m}}t e^{-tl}dt \nonumber\\
    &= t_{m}^{-k} \ \Bigg( \frac{k+1}{l +\frac{k}{t_{m}}}\Bigg)^{k+2}
	\text{.}
\end{align}	
	

The distribution  in equation \ref{eq:Likelihood_function_extended_pulse} is known as a \emph{Lomax} or \emph{Pareto-II} distribution, which is a heavier-tailed relative of the Exponential distribution. Under the extended pulse model, the expected segment length will be the same as under the simple pulse model (Eq. \ref{eq:Expected_l_simple_pulse}):
	
\begin{equation}
\label{eq:Expected_l_extended_pulse}
    \mathbb{E}[L] = \frac{k}{t_m}\frac{1}{(k+1)-1} = \frac{1}{t_{m}}
\end{equation}
	
	
but the variance is larger:
	

\begin{equation}
\label{eq:Var_l_extended_pulse}
	Var[L] = \frac{(k+1)}{(k-1)} \frac{1}{t_m^2}\text{.}
\end{equation}
	
We obtain the ALD-function from equation \label{eq:ld_general} using the  moment-generating function of $m(t)$:
\begin{equation}
\label{eq:extended_pulse_tail}
D_t(l) \propto  \left( 1 + \frac{t_m l}{k}\right)^{-k} \text{.}
\end{equation}

	
\paragraph{The Constant Migration Model}

The simple pulse model can be thought of as the extreme case of the extended pulse model when $k \to \infty$, i.e. the pulse gets infinitely short. In the  other extreme the extended pulse model approaches a model of constant migration. In this case, the last migration event at a particular location is exponentially distributed with rate $m$ (Eq. \ref{eq:effective_migration}), which is a model considered by \cite{pool_inference_2009}. 
Setting $t_m = \frac{2}{m}, k=2$, we obtain

\begin{subequations}
\begin{align}
    m(t) &= m \exp(-mt) \sim \Gamma(1, m)\\
    P(T_i=t) &\sim \Gamma\left(2, m\right)\\
	P(L_i=l) &= \frac{2m^2}{(m+l)^3}\label{eq:segments_continuous}\\
	D(l) &\propto \frac{m}{m + l}
\end{align}
\end{subequations}
	
Equation \ref{eq:segments_continuous} differs slightly from equation 6 in \cite{pool_inference_2009} because we approximate the expected number of segments with $n=Kt$, versus theirs $n=1+Kt$ (however they converge to each other for large $Kt$). 
	

\subsubsection{Estimation of Admixture Times}\label{admixture time estimates}
For inference, either the admixture segment lengths or ALD can be used. Assuming the admixture segment lengths are known, equation \ref{eq:Likelihood_function_extended_pulse} is the likelihood function and can be used for inference. For inference using ALD,  we follow \cite{moorjani_history_2011} and use the decay of ALD with genetic distance as a statistic. Following \cite{moorjani_genetic_2016}, we add an intercept $A$ and a constant $c$ modelling background LD:

\begin{equation}
\label{eq:simple_pulse_tail_inf}
ALD \sim\ A\,e^{-t_m \:l}+c
\end{equation}

\begin{equation}
\label{eq:extended_pulse_tail_inf}
ALD \sim\ A\,\left( 1 + \frac{t_m}{k} \:l\right) ^{-k}+c
\end{equation}

\section{Results}\label{results}

Here, we investigate under which scenarios we can distinguish the simple and extended pulse models, and when we can infer parameters under either model. 
We start with an idealized scenario of simulations under the model, and then continue with more realistic coalescent simulations using \texttt{msprime} \citep{kelleher_efficient_2016}. 

\subsection{Power Analysis under the Model}\label{Power Analysis}
In the easiest case, we assume that segments are  known and we simulate directly under the model (Eq. \ref{eq:Likelihood_function_extended_pulse}) and evaluate under which conditions we can tell the two models apart using likelihood-ratio tests on the simulated segments. For this purpose, we compare two scenarios, one where gene flow happened 1,500 generations ago, which reflects Neandertal gene flow inferred from present-day individuals. In the second scenario, which reflects inference from ancient modern human data, the samples are taken 50 generations after gene flow ended. We vary pulse durations from one to 2,500 generations, and sample between 100 and 100,000 unique segments. As the simple pulse model is an edge case of the extended pulse model with $k\to \infty$, standard likelihood theory does not apply, and we use empirical significance cutoffs \citep{Kozubowski_Testing_2008}.


The resulting log-likelihood ratios are given in Figure \ref{fig:fig1_1}. In general, we find that power to distinguish the model increases with pulse duration and the amount of data, and that it is easier to distinguish the models when gene flow had been more recent.  For example, with 10,000 unique segments we need an event lasting around 1,000 generations before we are able to confidently distinguish an extended from a simple pulse (Figure \ref{fig:fig1_1}) using present-day data. In contrast, by sampling closer to the admixture event we are able to distinguish an extended pulse already with a duration of 40-60 generations.

\begin{figure}[ht]
\centering
%\includegraphics[width=16cm,height=10cm,keepaspectratio]{ATE_Revisions_files/figure-latex/figR1-1.pdf}
\includegraphics[width=16cm,height=10cm,keepaspectratio]{Fig_2_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:fig1_1} Model comparisons on perfect data. Segments are either sampled at the present (purple) or 50 generations after the end of gene flow (turquoise). Log-likelihood ratios bigger than ten are rounded to ten.}
\end{figure}


\subsection{Population Genetic Model Comparisons}\label{Model comparison}
In the previous section, we have shown that we can distinguish long pulses from instantenous gene flow under idealized conditions. As a more realistic scenario, we perform population genetic simulations using  \texttt{msprime}  \citep{kelleher_efficient_2016}. Throughout, we simulate 3\% Neandertal admixture into non-Africans using a demographic model of archaic introgression (Supplementary Figure \ref{fig:figS1}B) with a mean admixture time of 1,500 generations ago and varying durations.  We simulate 20 chromosomes of length 150MB, using either a constant recombination map, or using the HapMap recombination map \citep{HapMapConsortium_second_2007}. This results in $\sim10,000$ introgressed segments. We then perform inference either using the simulated segments, segments inferred from the data \citep{skov_detecting_2018}, or ALD calculated using ALDER \citep{loh_inferring_2013}. We further vary recombination rate settings as i) inference and simulation under constant recombination rate (Constant/Constant), ii) simulation using the HapMap genetic map \citep{HapMapConsortium_second_2007}, and inference using no correction (HapMap/Constant), iii) simulation using HapMap, correction using a different map  (HapMap/AAMap) \citep{hinch_landscape_2011}, iv) and inference using the same map used for the simulations (HapMap/HapMap). 

Using these simulations, we perform  model comparisons (Figure \ref{fig:figResult2}A). For segments, we again use the likelihood-ratio and find that the results for the simulated segments closely match the simulations under the model (Figure \ref{fig:fig1_1}), showing that our model is a good approximation in the parameter range of interest. In contrast, we find that for inferred segments, results greatly depend on the recombination rate used: For a constant recombination rate, results are similar, but for the HapMap-recombination map, we do not have any power to distinguish these scenarios. As we fit ALD using non-linear least squares, no formal model-comparison framework exists. Qualitatively, we plot the normalized residual sum-of squares (RSS) and find that they increase with $t_d$ for both recombination scenarios, suggesting that the difference between the two models increase. 

Next, we evaluate parameter inference. In Figure \ref{fig:figResult2}B, we present estimates of the mean admixture times, admixture duration and the fitted segment and ALD distributions, respectively.  We find that the mean admixture times are reasonably accurately estimated in most scenarios, the exception being the inferred segments when using the variable (HapMap) recombination map. The admixture duration estimates are  often less accurate, and in most cases has very large variation between simulations.
 
We detect a slight, but consistent underestimate of the mean admixture times, which increases with $t_d$. For the segments, this underestimate is likely due to the slight downward bias caused by recombination and coalescence between admixed segments \citep[][see also Appendix \nameref{Appendix_3}]{liang_lengths_2014}. For ALD, this bias is much less severe, particularly for inference under the extended pulse model. For scenarios where the recombination map is misspecified,  $t_m$ is estimated to be only around half of its true value (Supplementary Figure \ref{fig:figResult2_sup}). However, we find that in some cases, the extended pulse model provides a better estimate of $t_m$ by estimating the pulses to be extremely long.

In Figure \ref{fig:figResult2}C we show examples of the estimated segment length and ALD distributions compared to the simulated data. For these log-plots, the slope of the curve corresponds to the estimate of $t_m$, and the deviation from linearity reflects the duration of gene flow. In all cases, we find that the expected decay is very close to linear, matching our finding that power to differentiate these old events is limited. We find that particularly when using a constant recombination map, all three summaries give a very close fit, and the segment length and ALD-decay distribution  closely follow their expectations, which is consistent with the generally  good parameter estimates under these conditions. In the case of a variable recombination map, we find that particularly inferred fragments perform poorly, which is reflected by a substantial downward bias of $t_m$ and $t_d$.






\begin{figure}
\centering
%\includegraphics[width=16cm,height=18cm,keepaspectratio]{ATE_Revisions_files/figure-latex/figResult2_all_together-1.pdf}
\includegraphics[width=16cm,height=18cm,keepaspectratio]{Fig_3_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:figResult2} Model Choice, Model Fit and Parameter Estimates. A) Log-likelihood ratios and RSS difference between the simple and extended pulse models for segment data and ALD, respectively. Simulation and inference were done using constant(purple) and an empirical (teal) recombination map. B) Estimates of $t_m$ and $t_d$. Solid black line indicate simulation values, the red dotted line adds a migration corrected ($t_m(1-\alpha)$). C) Model fit for a single simulation in each scenario. Estimated segment density or weighted LD (grey) is compared with the expected (turquoise) and fitted single pulse (yellow) and extended pulse (purple). }
\end{figure}


\subsection{Comparing Effect Sizes for Technical Covariates}\label{comparing effect sizes}

As we find that ALD performs as good or better than inferred segments (Figure \ref{fig:figResult2}), we focus on ALD for the remainder of this paper. Our next goal is to more carefully evaluate the relative importance of common assumptions made in the inference of admixture times, under both the simple and extended pulse model in the ALD framework on the bias and accuracy of estimates of $t_m$ under either model.

In particular, we use a Bayesian Generalized Linear Model (GLM) framework to contrast the effect of extended gene flow on admixture time inference with i) the effects of a simple/complex demographic history (Supplementary Figure \ref{fig:figS1}), ii) recombination map variation , iii)  the ALD ascertainment scheme,  iv) ,$d_0$, the minimum genetic distance between variants and v) the number of makers used to estimate the ALD curve (see Methods for details). For each modelling parameter and gene flow model, we use a simple model as the base case, and we study the impact of a more ``realistic'' alternative model.

In Figure \ref{fig:figGLM}, we present the estimated effect sizes for these six variables and four key interaction terms. To model bias, we fit a model to the standardized difference between the true and estimated mean admixture time, and to model accuracy (Methods, Supplementary Table \ref{tab:table_Supplements_ests_bias}, Supplementary Figure \ref{fig:figGLM_deviation}, Supplementary Table \ref{tab:table_Supplements_ests_deviation}). These effect sizes are estimated using simulations under all possible parameter combinations on a scenario with admixture happening 1,500 generations ago. (Supplementary Figures \ref{fig:figSGLM_data_SP} and \ref{fig:figSGLM_data_EP}).

As a baseline, for comparison, we define a standard model using a simple demography (Supplementary Figure \ref{fig:figS1}A) and a constant recombination rate.  This baseline model results in unbiased estimates of $t_m$ under the single pulse model with low deviation  of 0.08 (0.02 -- 0.14 $C.I._{95\%}$), and a slight upward bias 0.21 (0.15 -- 0.28) for the extended pulse model.

The effect of simulating an extended-pulse gene flow only results in a slight bias of -0.17; -0.21 -- -0.13) for the simple pulse and no bias for inference under the extended pulse model (-0.11; -0.15 -- 0.07). In contrast, uncertainty in the genetic map causes by far the largest downward bias (Simple pulse: -1.22, -1.29 -- -1.16; Extended Pulse: -0.74, -0.80 -- -0.67) with high deviation in the estimates (Supplementary Figure \ref{fig:figGLM_deviation}). The more complex demography results in an underestimate of $t_m$, presumably because of increased genetic drift, for both the simple pulse ( -0.27, -0.33 -- 0.22 and extended pulse models (-0.43, -0.49 -- -0.38). The remaining parameters largely only have very minor effects, the biggest of which is changing the minimum cutoff from 0.05cM to to 0.02 cM. 


\begin{figure}
\centering
%\includegraphics[width=12cm,height=16cm,keepaspectratio]{ATE_Revisions_files/figure-latex/figResult_3_GLM_SP_and_EP_bias-1.pdf}
\includegraphics[width=12cm,height=16cm,keepaspectratio]{Fig_4_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:figGLM} GLM effect sizes for the bias between simulated and estimated mean admixture time  and 95\% C.I. for the parameters between the simple and extended pulse model: gene flow (simple/extended), recombination rate (constant/varying), demography (simple/complex), minimal genetic distance (0.02/0.05 cM), SNPs used for ALD calculation (100 \% / 5 \%) and ascertainment scheme (LES/HES). Estimates are calculated across all possible combinations of parameters. Dotted horizontal line indicates unbiased admixture estimates.}
\end{figure}


\subsection{Application to Neandertal Data}
Our next aim is to apply our model on the case of Neandertal gene flow into Eurasians. We estimate the Neandertal admixture pulse from the 1000 genomes data \citep{the_1000_genomes_project_consortium_global_2015} and three high-coverage Neandertal genomes \citep{prufer_complete_2013, prufer_high-coverage_2017, mafessoni_high_coverage_2020} by fiting pulses with durations ranging from one generation up to 2,500 generations to the ALD decay curve (Figure \ref{fig:fig5}, Supplementary Table \ref{tab:table_Supplements_Application_to_Neandertal_data_Estimates}). Plotting these best-fit ALD curves shows the extremely slight difference predicted under these drastically different gene flow scenarios (Figure \ref{fig:fig5}A). The difference between scenarios becomes more apparent if we log-transform the y-axis (Figure \ref{fig:fig5}B), where we see that ongoing gene flow results in a heavier tail in the ALD distributions. However, these LD values are very close to zero, and are thus only very noisily estimated. 

For short gene flows (less than 1,000 generations), our estimates for $t_m$ are very similar and identical to the simple pulse, at around 1,682 (1,526 -- 1,839 $C.I._{95\%}$) generations. Extremely high values of $t_d$ result in slightly higher values of $t_m$ with overlapping compatibility intervals; but all predict that Neandertals would have survived until ~30kya, for which the archaeological evidence is extremely sparse \citep{hublin_last_2017}.  From the residual sum-of-squares, the models perform equally well, with longer extended pulses of gene flow achieving marginally better fits (Supplementary Table \ref{tab:table_Supplements_Application_to_Neandertal_data_RSS}). Therefore, we find that all scenarios are compatible with the observed data, and that there is little power to differentiate these cases from genetics alone.  


\begin{figure}
\centering
%\includegraphics[width=16cm,height=18cm,keepaspectratio]{ATE_Revisions_files/figure-latex/fig5_Real_data-1.pdf}
\includegraphics[width=16cm,height=18cm,keepaspectratio]{Fig_5_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:fig5} Neandertal Gene Flow in Modern Humans. We fit models with fixed $t_d$ from one to 2,500 generations of gene flow to the ALD-curve calculated from  CEU individuals on A) natural and B) log-scale.}
\end{figure}

\subsection{Sampling Closer to the Admixture Event}\label{estimating the Lomax-parameters under different conditions}

Since Neandertal gene flow happened long in the past, much of the signal has been lost, and we have shown that in this scenario, power to distinguish different scenarios is low.

However, we have also shown in Figure  \ref{fig:fig1_1} that inference is easier for more recent gene flow, a case that is relevant for many study systems. We investigate this in a series of simulations where the time between sampling and gene flow is smaller (Figure \ref{fig:Closer_sampling}). We use the simple demographic scenario with a constant-sized populations (Supplementary Figure \ref{fig:figS1}), and use ALD for inference using the optimised settings for the Neandertal case (ascertainment scheme = LES and  d0 = 0.05 cM).

In Figure \ref{fig:Closer_sampling}A and B, we show the accuracy of estimating $t_d$ and $t_m$ for increasingly longer pulses, sampled 50 generations after gene flow ended. The corresponding comparison of model fit is depicted in Supplementary Figure \ref{fig:Closer_sampling_supplement}. For these cases, we find that inference of $t_m$ under the simple pulse model works well for the shortest pulses, but becomes increasingly downward biased as $t_d$ increases.  Estimates of $t_m$ are less biased for inference under the extended 
pulse model, where we get accurate estimates particularly if recombination is constant. In the scenarios with a variable recombination rate, we find that for short, recent pulses, all corrections give good results, but for longer pulses, particularly assuming a constant recombination rate leads to a stronger bias. We also find that we are able to accurately infer the admixture duration, particularly if the recombination rate is constant.

In \ref{fig:Closer_sampling} C and D, we keep the pulse duration constant at $t_d=800$, but move it successively further into the past. For the first two cases of $t_m=450$ and $t_m=500$ where the pulse is recent, we again obtain good parameter estimates, but performance deteriorates for $t_m \geq 600$, which also results in low power to distinguish the simple and extended pulse model (Supplementary Figure \ref{fig:Closer_sampling_supplement}, lower panel). 

\begin{figure}
\centering
%\includegraphics[width=16cm,height=18cm,keepaspectratio]{ATE_Revisions_files/figure-latex/figCloser_Sampling_Supplement-1.pdf}
\includegraphics[width=16cm,height=18cm,keepaspectratio]{Fig_6_MBE-21-0164_R1_Peter_Iasi.pdf}
\caption{\label{fig:Closer_sampling} Parameter estimation using ALD from recent admixture pulses.  We show estimates of $t_m$ (panels A, C) and $t_d$ for a series of simulations with increasing $t_d$ and $t_m$ such that we sample 50 generations after gene flow (top) and a series of simulations with a increasingly older extended pulse (bottom).  All times are given in generations.}
\end{figure}


\section{Discussion}\label{discussion}
In this paper, we introduce a new population genetic model for dating extended pulses of gene flow. Our model has just two parameters, that can be interpreted as the mean time and duration of gene flow; and has simple closed form solutions for the segment length and ALD distributions. We show that both the instantaneous pulse and constant migration models are special cases of our model, where the duration is extremely short or long, respectively. We also demonstrate that the segment length distribution and ALD-decay can be directly transformed into each other; in particular the segment-length distribution is proportional to the second derivative of the ALD-decay curve. This makes our theory and models generally applicable beyond gene flow between Neandertals and humans. In fact, we find that we have little resolution for the parameter settings relevant for archaic gene flow, as the data resulting from simple and extended pulses long in the past is extremely similar. In contrast, we have much more power to estimate the duration of gene flow from events in the recent past, a scenario relevant for many hybridizing species. One limitation of our approach is that we assume that the overall amount of introduced material is low, and that we ignore the effects of genetic drift and selection.

Previous approaches to date Neandertal-human gene flow have focused almost entirely on the mean time of gene flow using a simple pulse model, for which reasonably tight credible intervals can be estimated \citep{sankararaman_date_2012, moorjani_genetic_2016}. Under this model, the credible intervals of this time are bounds of when gene flow between Neandertals and early modern humans could have happened. 

Our estimate of the $t_m$ for Neandertal gene flow of 1,682 generations corresponds to a mean time estimate of 49ky (assuming a generation time of 29 years \cite{moorjani_genetic_2016}), with bounds of 44-54ky. This is in almost perfect agreement with the previous result of \citep{moorjani_genetic_2016}, (41-54ky), which is based on largely the same method. However, here we show that models of extended gene flow with $t_d$ up to a thousand generations provide very similar fits to the data; and that marginally better fits are achieved with very long gene flows. However, these models all would have Neandertals survive until around 30kya, whereas archaeological evidence for Neandertals surviving beyond 40ky is increasingly sparse \citep{hublin_last_2017}, so that these models of extremely long gene flow might be rejected on these grounds. 

Our finding that the observed data is compatible with models involving hundreds of generations of gene flow means that while likely substantial amounts of gene flow happened around these mean times, gene flow might have also happened tens of thousands of years before or after. This is of great practical importance, as it makes linking genetic admixture date estimates with biogeographical events much more difficult \citep{sankararaman_date_2012,lazaridis_genomic_2016,douka_age_2019,jacobs_multiple_2019,vyas_analyses_2019}. 

The discovery of early modern human genomes dated to 40,000 - 45,000 ya with very recent Neandertal ancestors less than ten generations ago \citep{fu_genome_2014, hajdinjak_initial_2021} illustrates that gene flow likely happened over at least several thousand years.  In general, inference based on ancient genomes \citep{fu_genome_2014, fu_early_2015, moorjani_genetic_2016, hajdinjak_initial_2021} promises to resolve some of these dating issues, as inference is substantially easier when admixture is more recent, as the time difference between gene flow and sampling time is much lower (Figures \ref{fig:fig1_1}, \ref{fig:Closer_sampling}). However, using these genomes for dating leads to further hurdles, particularly pertaining to the spatial distribution of admixture events; whereas we can assume that the spatial structure present in initial upper paleolithic modern humans is largely homogenized in present-day people, the introgression signals observed in Bacho-Kiro and Oase could be partially private to these populations, and thus these populations may have a different admixture time distribution than present-day people.

The uncertainty over the duration of Neandertal gene flow also has some implications for selection on introgressed Neandertal haplotypes. Neandertal alleles have been suggested to be deleterious in modern human populations due to an increased mutation load \citep{harris_genetic_2016, juric_strength_2016}. Some details of these models may be affected if migration occurred over a longer time. For example, \cite{harris_genetic_2016} suggested that an initial pulse of gene flow of up to 10\% Neandertal ancestry might be necessary to explain current amounts of Neandertal ancestry, with very high variance in the first few generations after gene flow. More gradual gene flow could mean that such high admixture proportions were never reached, but rather a continuous migration-selection balance process persisted for the contact period, where deleterious Neandertal alleles continually entered the modern human populations, but were selected against immediately. 
However, in terms of the overall frequencies, there is likely little difference. For example, \cite{juric_strength_2016} showed using a two-locus model that the frequencies of Neandertal haplotypes alone cannot be used to distinguish different admixture histories.

In addition, we find that modelling and method assumptions have an impact on admixture time estimates that are of a similar or larger magnitude than the effect of assuming a one-generation pulse. In particular, recombination rate variation poses a practical limitation to the accuracy of admixture date estimates for old gene flow, and has to be very carefully considered when making inferences about admixture times. A possible reason is that both an extended pulse, as well as a non-homogeneous recombination map, lead to an admixture segment distribution that deviates from the expected exponential distribution. Throughout, we measure segment lengths and LD-decay distance in recombination units. Misspecification of the recombination rate will increase the variance in ALD or segment lengths, which might be confounded with a longer admixture pulse \citep{sankararaman_date_2012}. Therefore, population-specific fine-scale recombination maps are needed for accurate admixture time estimates, at least for admixture that happened more than a thousand generations ago. Estimates of more recent admixture appear to be more robust, perhaps because coarser-scale recombination maps are better estimated, differ less between populations \citep{hinch_landscape_2011} and the error relative to fragment length is substantially lower. 

To further refine admixture time estimates, time series data from more admixed early modern human and Neandertal genomes are needed. In particular, measures based on population differentiation  \citep[e.g][]{wall_higher_2013,browning_analysis_2018,villanea_multiple_2019} hold much promise to understand the different events that contributed to archaic ancestry in modern humans. While Neandertal ancestry in present-day people has been largely homogenized due to the substantial gene flow between populations, samples from both the Neandertal and early modern human populations immediately involved with the gene flow could refine when and where this gene flow happened. 


\section{Material \& Methods}\label{methods}

\subsection{Power Analysis under the Model}\label{power analysis}

To test the power to distinguish the simple from an extended pulse we simulated 100, 1000, 10,000 and 100,000 unique times $T_i$ form a Gamma distribution, with shape parameter $k+1$ and scale  $k/t_m$, setting $t_m$ to 1,500 generations.  Segment lengths $L_i$ are obtained by sampling for each $T_i$ from an exponential distribution  with rate parameter $T_i$ for present day samples and $T_i^{(closer)}= T_i - t_m - t_d/2 - 50$ for sampling 50 generations after the end of gene flow. We obtain maximum-likelihood estimates for the simple (Eq. \ref{eq:Likelihood_function_simple_pulse}) and extended pulse (Eq. \ref{eq:Likelihood_function_extended_pulse}) using the \texttt{optim} function implemented in R \citep{R_Core_Team_2019}.

\subsection{Coalescent Simulations}\label{coalescent simulations}

We further test our approach on coalescence simulations using  \texttt{msprime} 
\citep{kelleher_efficient_2016}. We focus on scenarios mimicking Neandertal admixture, and choose sample sizes to reflect those available from the 1000 Genomes data \citep{the_1000_genomes_project_consortium_global_2015}. For ALD simulations, we simulate 176 diploid
African individuals and 170 diploid non-Africans, corresponding to the
number of Yoruba (YRI) and Central Europeans from Utah (CEU). For inference based on segments we simulated 50 diploid non-Africans.
Since three high coverage Neandertal genomes are available \citep{prufer_complete_2013,prufer_high-coverage_2017,mafessoni_high_coverage_2020} we  simulate three diploid Neandertal genomes. 

The demographic parameters are based on
previous studies dating Neandertal admixture
\citep{sankararaman_date_2012,fu_genome_2014,moorjani_genetic_2016,skov_detecting_2018}. In
the ``simple'' demographic model (Supplementary Figure  \ref{fig:figS1} A), the effective
population size is assumed constant at $N_e=10,000$ for all populations, the
split time between modern humans and Neandertals is 10,000 generations
and the split between Africans and non-Africans is 2,550
generations. The migration rate from Neandertals into non-Africans
was set to zero before the split from Africans, to ensure that there is no Neandertal
ancestry in Africans.  For a more complex scenario of human population history, we followed the demographic model simulation setup of \cite{skov_detecting_2018}, but only simulated the Europeans. The split time of  non-Africans are kept the same as in the ALD simulations (2550 generations ago) (Supplementary Figure \ref{fig:figS1}B). 

For each individual we simulate 20
chromosomes with a length of 150 Mb each. The mutation rates are set to
 \(2*10^{-8}\) and \(1.2*10^{-8}\) per base per generation for the ``simple'' and ``complex'' models, respectively. The
recombination rates are set to \(1*10^{-8}\) per base pair per generation for the simple demography and \(1.2*10^{-8}\) per base pair per generation for the complex demographic model, unless specified otherwise. 

Since inferring archaic segments is slow, we use 25 replicates for scenarios where we compare segment-based and ALD-based inference, and use 100 replicates when we only perform ALD-based inference. 


\subsubsection{Simulating Admixture}\label{Simulating the expanded pulse}
We specify simulations under the extended pulse model using the mean admixture time $t_m$ and the duration $t_d$. We recover the simple pulse model by setting $t_d=1$, up to errors due to discrete generations. To obtain the migration rates in each generation, we use a discretized version of the migration density (Eq. \ref{eq:Gamma_density}), which we then scale to the approximate amount of Neandertal ancestry in non-Africans ($\alpha = 0.03$). 

\subsubsection{Recombination Maps}\label{recombination map}
Uncertainties in the recombination map were previously shown to influence admixture time estimates \citep{sankararaman_date_2012,fu_genome_2014,sankararaman_combined_2016}. To investigate the effect of more realistic
recombination rate variation, we perform simulations using empirical recombination maps. For the GLM, we use the  African-American map \citep{hinch_landscape_2011} for simulations and for the remaining simulations we use the HapMap phase 3 map \citep{HapMapConsortium_second_2007}. For simplicity, we use the same
recombination map (150 Mb of chromosome 1, excluding the first and last 10 Mb)
for all simulated chromosomes. When simulating under an empirical map, with the analysis assuming a constant rate (\emph{i.e.} no correction), we use the mean recombination rate from the respective map to calculate the genetic distance from the physical distance for each SNP. The mean recombination rate is
calculated from the 150 Mb map (\(1.017 \, \frac{cM}{Mb}\) AAMap,
\(0.992 \, \frac{cM}{Mb}\) HapMap). For inference, each segment is either assigned a lengths based on its physical length (``constant''), the African-American map or HapMap recombination map, depending on the inference scenario.

\subsection{Estimating Admixture Time from Simulated Segment Data}\label{Estimating admixture time from simulated segment data}

For estimating admixture time and duration from introgressed segments, we either used the simulated segment lengths directly,  or alternatively added an inference step using the HMM from \cite{skov_detecting_2018}. We only considered inferred segments with an average posterior probability of 0.9 or higher. Furthermore, we use an upper and lower cutoff for inferred segment length of 0.05 cM and 1.2 cM. We fit the simple (Eq. \ref{eq:Likelihood_function_simple_pulse}) and extended pulse (Eq. \ref{eq:Likelihood_function_extended_pulse}) using the \texttt{optim} function implemented using \texttt{R 4.0.3} (method="L-BFGS-B") with lower and upper constrains being $1,5000$ for $t_m$ and $2,1e8$ for $k$, respectively. 

\subsection{Estimating Admixture Time from ALD Data}\label{Estimating admixture time from ALD data}

\subsubsection{Ascertainment Scheme}\label{asceteinment scheme}
Since ALD for ancient admixture events can be quite similar to the genomic background, SNPs need to be ascertained to enrich for
Neandertal informative sites in the test population. This removes noise and
amplifies the ALD signal \citep{sankararaman_date_2012}. 
We evaluate the impact of the ascertainment scheme by contrasting two distinct schemes \citep{sankararaman_date_2012,fu_genome_2014}. The lower-enrichment ascertainment scheme (LES) only considers  sites that are fixed for the ancestral state in
Africans and polymorphic or fixed derived in Neandertals. The higher-enrichment
ascertainment scheme (HES) is more restrictive in that it further excludes all sites that are not polymorphic in non-Africans.

\subsubsection{ALD Calculation}\label{ALD calculation}

The pairwise weighted LD between the ascertained SNPs a certain genetic
distance \(d\) apart is calculated using ALDER
\citep{loh_inferring_2013}.  A minimal genetic distance \(d_0\) between
SNPs is set either to 0.02 cM and 0.05 cM. This minimal distance cutoff
removes extremely short-range LD, which might also be due to inheritance of segments from the ancestral population (incomplete lineage sorting ILS) and not gene flow. 

\subsubsection{Parameter Estimates}\label{Parameter estimate}

We estimate parameters by fitting the ALD-curve to equations \ref{eq:simple_pulse_tail_inf} and \ref{eq:extended_pulse_tail_inf}  using a non-linear least square approach implemented in the  \texttt{nls} function in \texttt{R 4.0.3} (algorithm="port") with lower and upper constrains being $1$ and $5000$ for $t_m$ and $1/1e10$ and $1/2$ for $1/k$, respectively. To achieve better conversion we pre-fit the functions using the estimates of the \texttt{DEoptim} optimization  \citep{ardia_deoptim_2016} as starting parameters for the \texttt{nls} function. To improve estimates for $t_d$, we run the fitting using 10 iterations to avoid local optima. We select the model with the lowest residual-sum-of-squares. 


\subsection{Modeling Parameter Effect Sizes}\label{modeling prameter effect sizes}

To estimate the effect size of the different parameters (Eq.
\ref{eq:GLM1}) we use a Bayesian Generalized Linear Model, where $E$ is the response, and $A, M, D, R, S$ and $G$ are binary predictors. 

The model can be written as 

\begin{equation}\label{eq:GLM1}
\begin{split}
E_i &\sim \text{Normal}(\mu_i,\sigma) \\
\mu_i &= \alpha + \beta_aA_i + \beta_mM_i + \beta_dD_i + \beta_rR_i + \beta_{s}S_i + \\ &\beta_gG_i + \beta_{sa}S_iA_i + \beta_{ma}M_iA_i + \beta_{dr}D_iR_i + \beta_{mr}M_iR_i \\
\alpha &\sim \text{Normal}(0,2) \\
\beta_a,\beta_m,\beta_d,\beta_r,\beta_{s},\beta_g,\beta_{sa}, \beta_{ma},\beta_{dr}, \beta_{mr} &\sim \text{Normal}(0,2) \\
\sigma &\sim \text{Exponential}(1)
\end{split}
\end{equation}

where the variables define
\begin{itemize} 
    \item ascertainment scheme: $A_i$ = LES/HES
    \item minimal genetic distance: $M_i$ = $0.02 cM/ 0.05 cM$
    \item demography: $D_i$ = simple/complex
    \item recombination rate: $R_i$ = constant/variable
    \item n SNPs used: $S_i$ = 100 \%/5 \%
    \item gene flow model: $G_i$ = simple pulse (SP)/extended pulse (EP)
\end{itemize}

We fit two models, a primary model  aimed at investigating the bias of our estimates, and a second model aimed at investigating the deviation. In the first case, the response variable $E_i$ is

$$E_i = \frac{t_{est} - t_{sim}}{\sigma_{t_{est}}}$$, 

and in the second case we use the absolute error 

$$E_i = \frac{\mid t_{est} - t_{sim} \mid}{\sigma_{t_{est}}}$$,where $\sigma$ is the standard deviation of $t_{est}$.  We also modeled the interaction between number of used SNPs and the ascertainment scheme ($\beta_{sa}$), minimal distance and ascertainment ($\beta_{ma}$), demography and recombination ($\beta_{dr}$) and minimal distance and recombination($\beta_{mr}$).

We perform simulations using all possible parameter combinations. For the effect of the amount of SNPs i.e. accuracy of the ALD estimates, we downsampled the data by randomly choosing 5 \% of the overall SNPs for ALD calculation.
We define a standard model having a constant recombination rate, simple demography and gene flow, LES ascertainment and $d_0 = 0.05$.
The genetic distance is assigned from the physical  position using the average recombination rate of the African-American genetic map (\emph{i.e.}, assuming the recombination rate is constant over the simulated chromosome given by this value) for simulations under a variable recombination rate.
For each of the possible sets of parameters, we simulate 100 replicates each and fit ALD decay curves. We excluded a small number of simulations for which either the simple pulse or extended pulse  curve could not be estimated (87 out of 6,400). 







We assume a Normal likelihood because it is the maximum entropy distribution in our case. We obtained the posterior probability using a Hamiltonian Monte Carlo MCMC algorithm, as implemented in STAN \citep{carpenter_stan_2017} using an R interface \citep{stan_development_team_rstan_2018,mcelreath_statistical_2020}. The Markov chains converged to the target distribution (Rhat = 1) and efficiently sampled from the posterior (Supplementary Table \ref{tab:table_Supplements_ests_bias} and \ref{tab:table_Supplements_ests_deviation}).  


\subsection{Estimating Neandertal Admixture Time}\label{Estimating admixture time from real data}
We estimate the Neandertal admixture time distribution using ALD from the 1000 Genomes data \citep{the_1000_genomes_project_consortium_global_2015}, together with the Altai, Vindija and Chagyrskaya high coverage Neandertals. We  include the 107 unrelated individuals from the YRI as representatives of unadmixed Africans and all CEU as admixed Europeans. We only consider biallelic sites, and determine the ancestral allele using  the Chimpanzee reference genome (panTro4). We used the CEU specific fine-scale recombination map \citep{spence_inference_2019} to convert the physical distance between sites into genetic distance. 

\section{Acknowledgments}

We thank Svante P\"a\"abo, Janet Kelso, Fabrizio Mafessoni, St\'{e}phane Peyr\'{e}gne, Laurits Skov, Divyaratan Popli, Alba Bossom Mesa, Arev S\"umer and Shai Carmi for helpful comments and discussions.
This work was supported by the Max Planck Society and the European Research Council (grant number: 694707) to Svante P\"a\"abo.

\section{Author Contributions}
\begin{itemize}
    \item Conceptualization (Design of study) -- BMP
    \item Software -- LNMI
    \item Methodology -- lead: BMP -- support: LNMI, HR
    \item Formal Analysis -- LNMI
    \item Visualization -- LNMI
    \item Data Curation -- LNMI
    \item Writing (original draft preparation) -- lead: LNMI -- support: BMP
    \item Writing (review and editing) -- input from all authors
    \item Supervision -- BMP
    %\item Project Administration -- BMP
\end{itemize}

\section{Data Availability}\label{Data Availability}

The simulation script, the analysis pipeline and the \emph{Extended Admixture Pulse} model is  available  on  GitHub: \url{https://github.com/LeonardoIasi/Extended_Admixture_Pulse}. No new data were generated for this study.

\section{Competing Interests}

The authors declare that they have no competing interests.


\appendix
\section{Appendix: Derivation and Approximation Details}
\setcounter{equation}{0}
\renewcommand{\theequation}{A\arabic{equation}}

\subsection{Formal Motivation for ALD}\referencetitle{A}\label{Appendix_1}
In the main text, we motivate the ALD decay using the intuitive argument of ``adding up'' ALD introduced at different generations in the past. Here we present a more formal derivation: In one generation, LD changes due to LD-decay through recombination, and through the introduction of new ALD through migration.

\begin{align}
    D(t+1) &= (1-r-m(t)) D(t) + \Delta_X \Delta_Y m(t)\nonumber\\
    &\approx (1-r) D(t) + A m(t)
\end{align}
where $A$ is a constant and $r$ the map distance between the pair of markers. The approximation in the second line is valid if we ignore the effect of new introgressed material replacing older introgressed material.

This leads to the following linear differential equation:
\begin{equation}
    \frac{dD}{dt} = -r D + A m(t)\text{.}
\end{equation}

It is straightforward to verify that equation \ref{eq:ld_general_bwd} is a solution to this equation:

\begin{align}
   D(t) &= A\int_{-\infty}^t m(s)\exp(-r(t-s)) ds \nonumber \\
   \frac{d D}{dt} &= \frac{d}{dt}\left[ A\int_{-\infty}^t m(s)\exp(-r(t-s)) ds \right]\nonumber\\
   &= A m(t) + \int_{-\infty}^t \left[\frac{d}{dt}A m(s)\exp(-r(t-s))\right]ds \nonumber\\
   &= A m(t) - r \int_{-\infty}^t A m(s)\exp(-r(t-s))ds \nonumber\\
   &= A m(t) -r D(t) 
  \end{align}
where the third line follows from Leibniz' integral rule.


\subsubsection{Replacement During Pulse}\label{Appendix_2}

The ``old''-LD also changes due to addition of new introgressed material, which is accomodated using

\begin{equation}
    \frac{dD}{dt} = -(r + m(t)) D + A m(t)\text{.}
\end{equation}

This equation has solution

\begin{align}
    D(t) = A\int_{-\infty}^t \exp\big[-r(t -s) - M(s,t)\big] m(s) ds
\end{align}

where $$M(s,t) = \int_s^t m(x)dx \text{,}$$

which can be interpreted as contributing how much LD has decayed from introgression time $s$ to the observation time $t$ due to replacement from new introgressed material.

This follows from 
\begin{align}
   D(t) &= A\int_{-\infty}^t m(s)\exp\left[-r(t-s) - \int_{s}^tm(x)dx\right] ds \nonumber \\
   \frac{d D}{dt} &= \frac{d}{dt}\left[ A\int_{-\infty}^t m(s)\exp\left(-r(t-s)-\int_{s}^tm(x)dx\right) ds \right]\nonumber\\
   &= A m(t) + \int_{-\infty}^t A m(s)\left[\frac{d}{dt}\exp\left(-r(t-s)\right)\exp\left(-\int_{s}^tm(x)dx\right)\right]ds \nonumber\\
&= A m(t) - (r + m(t)) \int_{-\infty}^t A m(s)\exp\left(-r(t-s)\right)\exp\left(-\int_{s}^tm(x)dx\right)ds \nonumber\\   
&=A m(t) - (r + m(t)) D
  \end{align}
  
where the derivative can be evaluated using the product rule:
\begin{align}
    \frac{d}{dt}\exp\left(-r(t-s)\right) & =  -r\exp(-r(t-s)) \nonumber\\
    \frac{d}{dt}\exp\left(-\int_{s}^tm(x)dx\right) &=
    -m(t)\exp\left(-\int_{s}^tm(x)dx\right)\nonumber
\end{align}

Changing the flow of time and setting $t=0$ as in the main text, times,  this results in
\begin{align}
    D(l) = A\int_0^\infty \exp[-lrt] \exp[-M_b(t)] m_b(t) dt
\end{align}

where again $m_b(t) = m(-t)$ and $$M_b(t) = \int_0^t m_b(x)dx\text{.}$$

This motivates the ``effective'' migration rate $$m_e(t) = m_b(t) \exp[-M_b(T)]\text{;}$$ for the 
case of constant migration, $M_b(t) = mt$ and $m_b(t)= m e^{-mt}$, which is an exponential density  \citep{pool_inference_2009}.


\subsection{Connection Between Admixture Segment Length Distribution and ALD Function}
\label{app:connection}
Here we describe how the admixture segment length distribution $P(l)$ and the ALD function $D(l)$ are interconnected (under the assumptions of this work) and in fact uniquely determine each other as claimed in the main text (Eq.~\ref{eq_fr1} and Eq.~\ref{eq_fr2}).

Following the models outlined in the main text, throughout we assume that admixture is rare and that consequently admixture segments do not interact. Using these assumptions, we first describe how the admixture segment length distribution uniquely determines the ALD function, and second how the ALD function in reverse uniquely determines the admixture segment length distribution.

\subsection{From the Admixture Segment Length Distribution to the ALD Function}
Without loss of generality, we assume that derived allele frequencies are 0 in the target and 1 in the admixture source population, otherwise the resulting ALD curve can be re-weighted with a constant factor $A$ as in Eq.~\ref{eq:ald_general}.

We start by noting that two-point ALD between two loci can be written as: 

\begin{equation*}
D = x_{11} - p \cdot q,
\end{equation*}
where $x_{11}$ denotes the frequency of $11$-haplotypes and $p$ and $q$ the allele frequencies at these two loci. Under the assumption that introgressed segments do not interact with each other, genome-wide excess $11$ haplotypes beyond random association ($p\cdot q$ for each pair of loci) originate from pairs of markers on the same introgressed segments. To get the genome-wide average $D$ we therefore have to sum over the contribution of $11$ haplotypes from introgressed segments of all lengths.

For all pairs of markers a map distance $l$ apart, only segments of length $x>l$ contribute pairs of markers at distance $l$. Let us first describe a single introgressed segment of length $x$. In the limit of long chromosome (of length $G$) and of high marker density, a fraction $(x-l)/G$ of all pairs of markers at distance $L$ fall both onto this segment. Then, denoting the expected number of segments of length $x$ as $E(x)$, we sum over all segment lengths $x>l$ to get:

\begin{equation}
D(l) = \frac{1}{G} \int_l^\infty E(x) (x-l) \diff x
\label{eq_hr01}
\end{equation}

The expected number of segments of a given length $E(x)$ can be directly derived from the segment length distribution via $\mathbb{E}[K]$, the total number of all introgressed segments:

\begin{equation}
E(x) = P(x) \mathbb{E}[K]
\label{eq_hr02}
\end{equation} 

Plugging \ref{eq_hr02} into Eq.~\ref{eq_hr01} yields:

\begin{equation}
D(l) = \frac{\mathbb{E}[K]}{G} \int_l^\infty P(x) (x-l) \diff x.
\label{func_rel1}
\end{equation}

Eq.~\ref{func_rel1} now allows one to directly calculate $D(l)$ from $P(l)$, which shows that $P(l)$ uniquely determines $D(l)$. 

\subsection{From the ALD Function to the Admixture Segment Length Distribution}
To derive the inverse relationship, we start by differentiating $D(l)$ twice with respect to l ($\frac{\diff}{\diff l}$). Using Eq.~\ref{func_rel1} and the Leibniz integral rule yields:

\begin{align*}
D'(l) &= -\frac{\mathbb{E}[K]}{G} \int_l^\infty P(x) \diff x \\
D''(l) &= \frac{\mathbb{E}[K]}{G} P(l). 
\end{align*}

Simple rearrangement and plugging in the re-weighting of LD with $A$ (that we omitted above) yields:
\begin{equation}
P(l) = \frac{1}{A}\frac{G}{\mathbb{E}[K]} D''(l)
\label{func_rel2}
\end{equation}

Thus, $D(l)$ uniquely determines $P(l)$.

\subsection{For the Continuous Admixture Model}
For the concrete case of the continuous admixture model we derived explicit formulas for both the ALD function and the admixture segment length distribution (Eq.~\ref{eq:standard_likelihood_definintion} and (Eq.~\ref{eq:ld_general_inverted}). We can validate the above derived functional relationships (Eq.~\ref{func_rel1}) directly and show that the more general derivation above holds for these specific formulas central for this work. 

To check the relationships, we start with reiterating the explicit formulas: (Eq.~\ref{eq:standard_likelihood_definintion}) and (Eq.~\ref{eq:ld_general_inverted}):

% W.l.o.g. assume $\Delta_x = \Delta_y = 1$.:

\begin{align}
P(l) &= \frac{G}{\mathbb{E}[K]}\int_{0}^{\infty} t^2 m(t) \exp(-lt) \diff t \label{eq_mt1} \\
    D(l) &= A\int_0^{\infty} m(t)\exp(-lt) \diff t \text{.} \label{eq_mt2}
\end{align}

First, plugging Eq.~\ref{eq_mt1} into the functional relationship Eq.~\ref{func_rel1} yields:

\begin{align*}
\frac{\mathbb{E}[K]}{G} \int_l^\infty P(x) (x-l)  &= \int_l^\infty \diff x \int_{0}^{\infty} \diff t\ t^2 m(t) \exp(-xt) (x-l)  \\
&=  \int_{0}^{\infty} \diff t\ t^2 m(t) \underbrace{\int_l^\infty \diff x \exp(-xt) (x-l)}_{\frac{1}{t^2}\exp(-lt)} \\
&= \int_{0}^{\infty} m(t) \exp(-lt) \diff t\
\end{align*}

If we reweight the right-hand side with the allele-frequency differences A, it becomes D(l) from Eq.~\ref{eq_mt2}, which finishes the validation that the first functional relationship Eq.~\ref{func_rel1} holds.


To verify the second functional relationship (Eq.~\ref{func_rel2}) we differentiate $D(l)$ twice and multiply with $\frac{1}{A}\frac{G}{\mathbb{E}[K]}$:

\begin{equation*}
\frac{1}{A}\frac{G}{\mathbb{E}[K]} D''(l) = 
\frac{G}{\mathbb{E}[K]} \int_0^{\infty} m(t) \frac{\diff^2}{\diff l^2} \exp(-lt) \diff t = 
\frac{G}{\mathbb{E}[K]}\int_0^{\infty} t^2 m(t) \exp(-lt) \diff t = P(l)
\end{equation*}

\subsection{Genetic Drift and Recombination}\label{Appendix_3}
Our model assumes that Neandertal segments in the human population always recombine with non-Neandertal haplotypes, and that the effect of genetic drift can be neglected. In this appendix, we discuss some possible extensions of the model to incorporate aspects of genetic drift and recombination between admixture segments.

\paragraph{Single Pulse Theory of Recombination Between Fragments}
Under our modelling assumptions, we show that $\mathbb{E}L_i = t_m^{-1}$, but this ignores recombination between introgressed material. Under a single pulse, \cite{liang_lengths_2014} showed that under the SMC model \citep{mcvean_approximating_2005}  the expectation is reduced as

\begin{equation}
    \mathbb{E}L_i = \left[t_m (1-\alpha) \right]^{-1} \text{,}
\end{equation}

and under the SMC' model \citep{marjoram_fast_2006} this is
$$
\mathbb{E}L_i = \left[2N(1-\alpha)(1-\exp\left(-\frac{t_m}{2N}\right)\right]^{-1} \text{.}
$$
Using a Taylor expansion around $N \to \infty$ and ignoring terms of the order $\frac{1}{N^2}$, this can be approximated as 
\begin{equation}
  \mathbb{E}L_i \approx  \left[t_m(1-\alpha)\left(1-\frac{t_m}{4N}\right)\right]^{-1},
\end{equation}
which makes the similarity to the SMC model more apparent.

This is compared to $\mathbb{E}L_i = \frac{1}{t_m}$ as we obtained from equation \ref{eq:Expected_l_simple_pulse}.
The $(1-\alpha)$-term models recombination between adjacent introgressed segments; and the $\left(1-\frac{t_m}{4N}\right)$ can be thought of reflecting genetic drift. 

The justification for both of these formulas is that they are geometric mixtures of exponential distributions \citep{liang_lengths_2014}, which are themselves exponential. Under the extended pulse model, the segment length distribution is no longer exponential, so the segments may have a more complicated mixture distribution. 

For the case of Neandertal admixture, assuming gene flow happened over a short duration, these equations can be used to estimate the error made from ignoring drift and recombination between Neandertal segments. As $\alpha \approx 0.03$, $t_m \approx 1600$, $N \approx 10,000$, and so the expected combined error of these two terms is on the order of $10\%$.




\subsubsection{Effect of Reduced ``Effective'' Recombination and Coalescence}

In the ALD framework, we can take further complications into account. For example, we can motivate 
\begin{itemize}
    \item an ``effective'' recombination rate $r(t) = r (1-\alpha(t))$ that takes into account as the admixture fraction increases, some recombination events will be between introgressed material, and we denote the total amount of introgressed material by time $t$ as $\alpha(t)$.
    \item the allele frequencies in the admixing populations may change, so that we replace the constant $A=\Delta_x\Delta_y$ by $A(t)=\Delta_x(t)\Delta_y(t)$.
    \item genetic drift will fix some haplotypes, which then can no longer decay. This happens at rate $\frac{1}{2N(t)}$
\end{itemize} 

Taken together, the analogous equation is

\begin{equation}
    \frac{dD}{dt} = -\left[r(1-\alpha(t)) + m(t) +\frac{1}{2N(t)} \right] D + A(t) m(t)\text{.}
\end{equation}

This equation is still a first-order non-homogeneous linear differential equation, so the solution will have the same form

\begin{equation}
    D(t) = \int_{-\infty}^t\exp[-F(s,t)]m(s)A(s)ds 
\end{equation}

where
\begin{align}
    F(s,t) &= \int_s^t \left[r(1-\alpha(x)) - m(x) -\frac{1}{2N(x)} \right] dx \\
    &= r(t-s)  - r \int_s^t \alpha(x)dx + \int_s^t m(x)dx + \int_s^t \frac{1}{2N(x)}dx\text{,}\nonumber
\end{align}

For example, if we assume $N$, $A(t)$ and $\alpha$ are all constant,  and migration is a simple pulse at $t_m$

\begin{align}
    D_t(l) &= A \int_0^t \delta_{t_m}(s)\exp\left(-\frac{s}{2N}\right)\exp\left(-l(1-\alpha)s\right) ds \nonumber\\
    D_t''(l) &\propto \exp(l t_m(1-\alpha)) \nonumber\\
\end{align}
\hypertarget{refs}{}

%\bibliography{References/MyLibraryATE}
\bibliography{MyLibraryATE2}

%\pagebreak
%\setcounter{figure}{0} %\renewcommand{\figurename}{Fig. S}
%\renewcommand{\tablename}{Tab. S}




\end{document}