\documentclass[11pt]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\newcommand*\diff{\mathop{}\!\mathrm{d}}
%\usepackage{fixltx2e} % provides \textsubscript
\usepackage{xr} % referencing external ducument
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fontshttps://de.overleaf.com/project/5e85b0680d0bed00011ea790
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Title: Limitations to the Human Neandertal Admixture dating Supplements},
            pdfauthor={Leonardo Nicola Martin Iasi (Max Planck Institute for Evolutionary Anthropology, MPI EVA), Dr.~Benjamin Marco Peter (MPI EVA, benjamin\_peter@eva.mpg.de)},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
 
\usepackage{natbib}
\bibliographystyle{References/my_abbrvnat}
\setcitestyle{authoryear,open={(},close={)}}

\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}

\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi



\usepackage{setspace}
\onehalfspacing
\usepackage[left]{lineno}
\linenumbers
\usepackage[none]{hyphenat}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{float}
\usepackage{xcolor}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}

\floatplacement{figure}{H}
\begin{document}

\begin{titlepage}


    \vspace*{1cm}
        
        
    \begin{center}       
        \large
        \vspace{1cm}
        An extended admixture pulse model reveals the limits to the dating of Human-Neandertal introgression
        
       \vspace{1.0cm}
        \large
        Iasi, Leonardo N. M. \textsuperscript{1,2} and Peter , Benjamin M. \textsuperscript{1,3} \\ 
        
        \vspace{1.0cm}
            \Huge
            \textbf{Supplement Material 2}
    \end{center} 

            

\end{titlepage}

\section{Connection between ALD function and admixture segment length distribution}

Here we describe how the admixture segment length distribution $P(l)$ and the ALD function $D(l)$ are interconnected and in fact determine each other.

Following the models outlined in the main text, throughout we assume that admixture is rare and that consequently admixture blocks do not interact. Using these assumptions, we first describe how the admixture segment length distribution uniquely determines the ALD function, and second how the LD function in reverse uniquely determines the admixture segment length distribution.

\subsection{From the admixture segment length distribution to the LD function}
Without loss of generality, we assume that derived allele frequencies are 0 in the target and 1 in the admixture source population, otherwise the resulting LD curve can be re-weighted with a constant factor as in Eq.~X. We use that two-point LD between two loci can be written as: 

\begin{equation}
D = x_{11} - p \cdot q,
\end{equation}
where $x_{11}$ denotes the frequency of $11$-haplotypes and $p$ and $q$ are the allele frequencies at these two loci. Under the assumption that introgressed blocks do not interact with each other, the genome-wide number of excess $11$ haplotypes beyond random association ($p\cdot q$ for each pair of loci) originates from pairs of markers on the same introgressed blocks. To get the genome-wide average $D$ we therefore have to simply sum over the contribution of $11$ haplotypes from all introgressed blocks.

For all pairs of markers a map distance $l$ apart, only blocks of length $x>l$ contribute pairs of markers at distance $l$.Let us assume a single introgressed segment of length $x$. In the limit of long chromosome (of length $G$) and of high marker density a fraction of $(x-l)/G$ of all pairs of markers at distance $L$ fall both onto this segment. Then, denoting the expected number of segments of length $x$ as $E(x)$, we sum over all segment lengths $x>l$ to get:

\begin{equation}
D(l) = \frac{1}{G} \int_l^\infty E(x) (x-l) \diff x
\label{eq_hr01}
\end{equation}

The expected number of segments of a given length $E(x)$ can be directly derived from the block length distribution when using $\mathbb{E}[K]$, the total number of all introgressed blocks:

\begin{equation}
E(x) = P(x) \mathbb{E}[K]
\label{eq_hr02}
\end{equation} 

Plugging \ref{eq_hr02} into Eq.~\ref{eq_hr01} yields:

\begin{equation}
D(l) = \frac{\mathbb{E}[K]}{G} \int_l^\infty P(x) (x-l) \diff x.
\label{func_rel1}
\end{equation}

Eq.~\ref{func_rel1} now allows one to directly calculate $D(l)$ from $P(l)$, showing that $P(l)$ uniquely determines $D(l)$. 

\subsection{From the LD function to the admixture segment length distribution}
To derive an inverse relationship, we start by deriving $D(l)$ two times with respect to l ($\frac{\diff}{\diff L}$). Using Eq.~\ref{func_rel1} and the Leibnitz rule yields:

\begin{align}
D'(l) &= -\frac{\mathbb{E}[K]}{G} \int_l^\infty P(x) \diff x \\
D''(l) &= \frac{\mathbb{E}[K]}{G} P(l). \label{eq_hr2}
\end{align}
This equation shows that $P(l)$ can be derived by differentiating $D(l)$ twice and thus $D(l)$ uniquely determines $P(l)$.

\subsection{For the continuous admixture model}
For the concrete case of the continuous admixture model the relationship the main text derived (Eq.~X and Eq.~Y). W.l.o.g. assume $\Delta_x = \Delta_y = 1$.:

\begin{align}
P(L) &= \frac{G}{n}\int_{0}^{\infty} t^2 m(t) \exp(-Lt) \diff t \label{eq_mt1} \\
    D(L) &= \int_0^{\infty} m(t)\exp(-Lt) \diff t \text{.} \label{eq_mt2}
\end{align}

 In this case the above derived functional relationships can be directly checked:  The the functional relationship Eq.~\ref{eq_hr2} holds by simple plugging in. Conversely, plugging Eq.~\ref{eq_mt1} into the functional relationship Eq.~\ref{func_rel1} yields Eq.~\ref{eq_mt2} using integration by parts.
\end{document}